{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import sys\n",
    "import subprocess\n",
    "import glob\n",
    "import json\n",
    "from pathlib import Path\n",
    "import pandas\n",
    "import pickle\n",
    "import pprint\n",
    "from Bio import SeqIO\n",
    "from Bio.Seq import Seq\n",
    "from Bio.Blast import NCBIXML\n",
    "from Bio.SeqRecord import SeqRecord\n",
    "from Bio.Blast.Applications import NcbiblastpCommandline\n",
    "from axe import axe\n",
    "from collections import defaultdict"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "RUN_BLAST = True"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "def parse_blast_results(blast_results : str) -> dict:\n",
    "    file_size = os.path.getsize(blast_results)\n",
    "    if file_size == 0:\n",
    "        return None\n",
    "    results = {}\n",
    "    with open(blast_results, \"r\") as f:\n",
    "        blast_records = NCBIXML.parse(f)\n",
    "        for record in blast_records:\n",
    "            results[record.query] = []\n",
    "            for alignment in record.alignments:\n",
    "                for hsp in alignment.hsps:\n",
    "                    results[record.query].append({'contig_id' : record.query,\n",
    "                                                  'contig_len' : record.query_length,\n",
    "                                                  'hit_id' : alignment.hit_id,  # Modify this line to use alignment.hit_id instead of alignment.title\n",
    "                                                  'hit_title' : alignment.title,\n",
    "                                                  'hit_def' : alignment.hit_def,\n",
    "                                                  'hit_expect' : hsp.expect,\n",
    "                                                  'hit_score' : hsp.score,\n",
    "                                                  'hit_query_start' : hsp.query_start,\n",
    "                                                  'hit_query_end' : hsp.query_end,\n",
    "                                                  'hit_subject_start' : hsp.sbjct_start,\n",
    "                                                  'hit_subject_end' : hsp.sbjct_end,\n",
    "                                                  'hit_idents' : hsp.identities,\n",
    "                                                  'hit_pos' : hsp.positives,\n",
    "                                                  'hit_gaps' : hsp.gaps,\n",
    "                                                  'hit_frame' : hsp.frame,\n",
    "                                                  'hit_query' : hsp.query,\n",
    "                                                  'hit_subject' : hsp.sbjct})\n",
    "    return results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "--- Logging error ---\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/mf019/miniconda3/lib/python3.12/logging/__init__.py\", line 1160, in emit\n",
      "    msg = self.format(record)\n",
      "          ^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/mf019/miniconda3/lib/python3.12/logging/__init__.py\", line 999, in format\n",
      "    return fmt.format(record)\n",
      "           ^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/mf019/miniconda3/lib/python3.12/site-packages/axe/axe.py\", line 74, in format\n",
      "    return formatter.format(record)\n",
      "           ^^^^^^^^^^^^^^^^\n",
      "AttributeError: 'tuple' object has no attribute 'format'\n",
      "Call stack:\n",
      "  File \"<frozen runpy>\", line 198, in _run_module_as_main\n",
      "  File \"<frozen runpy>\", line 88, in _run_code\n",
      "  File \"/home/mf019/miniconda3/lib/python3.12/site-packages/ipykernel_launcher.py\", line 18, in <module>\n",
      "    app.launch_new_instance()\n",
      "  File \"/home/mf019/miniconda3/lib/python3.12/site-packages/traitlets/config/application.py\", line 1075, in launch_instance\n",
      "    app.start()\n",
      "  File \"/home/mf019/miniconda3/lib/python3.12/site-packages/ipykernel/kernelapp.py\", line 739, in start\n",
      "    self.io_loop.start()\n",
      "  File \"/home/mf019/miniconda3/lib/python3.12/site-packages/tornado/platform/asyncio.py\", line 205, in start\n",
      "    self.asyncio_loop.run_forever()\n",
      "  File \"/home/mf019/miniconda3/lib/python3.12/asyncio/base_events.py\", line 639, in run_forever\n",
      "    self._run_once()\n",
      "  File \"/home/mf019/miniconda3/lib/python3.12/asyncio/base_events.py\", line 1985, in _run_once\n",
      "    handle._run()\n",
      "  File \"/home/mf019/miniconda3/lib/python3.12/asyncio/events.py\", line 88, in _run\n",
      "    self._context.run(self._callback, *self._args)\n",
      "  File \"/home/mf019/miniconda3/lib/python3.12/site-packages/ipykernel/kernelbase.py\", line 545, in dispatch_queue\n",
      "    await self.process_one()\n",
      "  File \"/home/mf019/miniconda3/lib/python3.12/site-packages/ipykernel/kernelbase.py\", line 534, in process_one\n",
      "    await dispatch(*args)\n",
      "  File \"/home/mf019/miniconda3/lib/python3.12/site-packages/ipykernel/kernelbase.py\", line 437, in dispatch_shell\n",
      "    await result\n",
      "  File \"/home/mf019/miniconda3/lib/python3.12/site-packages/ipykernel/ipkernel.py\", line 359, in execute_request\n",
      "    await super().execute_request(stream, ident, parent)\n",
      "  File \"/home/mf019/miniconda3/lib/python3.12/site-packages/ipykernel/kernelbase.py\", line 778, in execute_request\n",
      "    reply_content = await reply_content\n",
      "  File \"/home/mf019/miniconda3/lib/python3.12/site-packages/ipykernel/ipkernel.py\", line 446, in do_execute\n",
      "    res = shell.run_cell(\n",
      "  File \"/home/mf019/miniconda3/lib/python3.12/site-packages/ipykernel/zmqshell.py\", line 549, in run_cell\n",
      "    return super().run_cell(*args, **kwargs)\n",
      "  File \"/home/mf019/miniconda3/lib/python3.12/site-packages/IPython/core/interactiveshell.py\", line 3075, in run_cell\n",
      "    result = self._run_cell(\n",
      "  File \"/home/mf019/miniconda3/lib/python3.12/site-packages/IPython/core/interactiveshell.py\", line 3130, in _run_cell\n",
      "    result = runner(coro)\n",
      "  File \"/home/mf019/miniconda3/lib/python3.12/site-packages/IPython/core/async_helpers.py\", line 129, in _pseudo_sync_runner\n",
      "    coro.send(None)\n",
      "  File \"/home/mf019/miniconda3/lib/python3.12/site-packages/IPython/core/interactiveshell.py\", line 3334, in run_cell_async\n",
      "    has_raised = await self.run_ast_nodes(code_ast.body, cell_name,\n",
      "  File \"/home/mf019/miniconda3/lib/python3.12/site-packages/IPython/core/interactiveshell.py\", line 3517, in run_ast_nodes\n",
      "    if await self.run_code(code, result, async_=asy):\n",
      "  File \"/home/mf019/miniconda3/lib/python3.12/site-packages/IPython/core/interactiveshell.py\", line 3577, in run_code\n",
      "    exec(code_obj, self.user_global_ns, self.user_ns)\n",
      "  File \"/tmp/ipykernel_4102621/3510340624.py\", line 1, in <module>\n",
      "    mjf = axe.mjf(logfile_name='plasmid_identification_v3_parsing')\n",
      "  File \"/home/mf019/miniconda3/lib/python3.12/site-packages/axe/axe.py\", line 211, in __init__\n",
      "    self.logger.log(msg=f'FileHandler for {self.logfile} is now online at level: {self.logfile_level}', level=100)\n",
      "Message: 'FileHandler for plasmid_identification_v3_parsing___2024-04-26__1901__.log is now online at level: 10'\n",
      "Arguments: ()\n",
      "--- Logging error ---\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/mf019/miniconda3/lib/python3.12/logging/__init__.py\", line 1160, in emit\n",
      "    msg = self.format(record)\n",
      "          ^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/mf019/miniconda3/lib/python3.12/logging/__init__.py\", line 999, in format\n",
      "    return fmt.format(record)\n",
      "           ^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/mf019/miniconda3/lib/python3.12/site-packages/axe/axe.py\", line 74, in format\n",
      "    return formatter.format(record)\n",
      "           ^^^^^^^^^^^^^^^^\n",
      "AttributeError: 'tuple' object has no attribute 'format'\n",
      "Call stack:\n",
      "  File \"<frozen runpy>\", line 198, in _run_module_as_main\n",
      "  File \"<frozen runpy>\", line 88, in _run_code\n",
      "  File \"/home/mf019/miniconda3/lib/python3.12/site-packages/ipykernel_launcher.py\", line 18, in <module>\n",
      "    app.launch_new_instance()\n",
      "  File \"/home/mf019/miniconda3/lib/python3.12/site-packages/traitlets/config/application.py\", line 1075, in launch_instance\n",
      "    app.start()\n",
      "  File \"/home/mf019/miniconda3/lib/python3.12/site-packages/ipykernel/kernelapp.py\", line 739, in start\n",
      "    self.io_loop.start()\n",
      "  File \"/home/mf019/miniconda3/lib/python3.12/site-packages/tornado/platform/asyncio.py\", line 205, in start\n",
      "    self.asyncio_loop.run_forever()\n",
      "  File \"/home/mf019/miniconda3/lib/python3.12/asyncio/base_events.py\", line 639, in run_forever\n",
      "    self._run_once()\n",
      "  File \"/home/mf019/miniconda3/lib/python3.12/asyncio/base_events.py\", line 1985, in _run_once\n",
      "    handle._run()\n",
      "  File \"/home/mf019/miniconda3/lib/python3.12/asyncio/events.py\", line 88, in _run\n",
      "    self._context.run(self._callback, *self._args)\n",
      "  File \"/home/mf019/miniconda3/lib/python3.12/site-packages/ipykernel/kernelbase.py\", line 545, in dispatch_queue\n",
      "    await self.process_one()\n",
      "  File \"/home/mf019/miniconda3/lib/python3.12/site-packages/ipykernel/kernelbase.py\", line 534, in process_one\n",
      "    await dispatch(*args)\n",
      "  File \"/home/mf019/miniconda3/lib/python3.12/site-packages/ipykernel/kernelbase.py\", line 437, in dispatch_shell\n",
      "    await result\n",
      "  File \"/home/mf019/miniconda3/lib/python3.12/site-packages/ipykernel/ipkernel.py\", line 359, in execute_request\n",
      "    await super().execute_request(stream, ident, parent)\n",
      "  File \"/home/mf019/miniconda3/lib/python3.12/site-packages/ipykernel/kernelbase.py\", line 778, in execute_request\n",
      "    reply_content = await reply_content\n",
      "  File \"/home/mf019/miniconda3/lib/python3.12/site-packages/ipykernel/ipkernel.py\", line 446, in do_execute\n",
      "    res = shell.run_cell(\n",
      "  File \"/home/mf019/miniconda3/lib/python3.12/site-packages/ipykernel/zmqshell.py\", line 549, in run_cell\n",
      "    return super().run_cell(*args, **kwargs)\n",
      "  File \"/home/mf019/miniconda3/lib/python3.12/site-packages/IPython/core/interactiveshell.py\", line 3075, in run_cell\n",
      "    result = self._run_cell(\n",
      "  File \"/home/mf019/miniconda3/lib/python3.12/site-packages/IPython/core/interactiveshell.py\", line 3130, in _run_cell\n",
      "    result = runner(coro)\n",
      "  File \"/home/mf019/miniconda3/lib/python3.12/site-packages/IPython/core/async_helpers.py\", line 129, in _pseudo_sync_runner\n",
      "    coro.send(None)\n",
      "  File \"/home/mf019/miniconda3/lib/python3.12/site-packages/IPython/core/interactiveshell.py\", line 3334, in run_cell_async\n",
      "    has_raised = await self.run_ast_nodes(code_ast.body, cell_name,\n",
      "  File \"/home/mf019/miniconda3/lib/python3.12/site-packages/IPython/core/interactiveshell.py\", line 3517, in run_ast_nodes\n",
      "    if await self.run_code(code, result, async_=asy):\n",
      "  File \"/home/mf019/miniconda3/lib/python3.12/site-packages/IPython/core/interactiveshell.py\", line 3577, in run_code\n",
      "    exec(code_obj, self.user_global_ns, self.user_ns)\n",
      "  File \"/tmp/ipykernel_4102621/3510340624.py\", line 1, in <module>\n",
      "    mjf = axe.mjf(logfile_name='plasmid_identification_v3_parsing')\n",
      "  File \"/home/mf019/miniconda3/lib/python3.12/site-packages/axe/axe.py\", line 223, in __init__\n",
      "    self.logger.log(msg=f'Console StreamHandler is now online at level: {self.console_level}', level=100)\n",
      "Message: 'Console StreamHandler is now online at level: 10'\n",
      "Arguments: ()\n",
      "--- Logging error ---\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/mf019/miniconda3/lib/python3.12/logging/__init__.py\", line 1160, in emit\n",
      "    msg = self.format(record)\n",
      "          ^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/mf019/miniconda3/lib/python3.12/logging/__init__.py\", line 999, in format\n",
      "    return fmt.format(record)\n",
      "           ^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/mf019/miniconda3/lib/python3.12/site-packages/axe/axe.py\", line 74, in format\n",
      "    return formatter.format(record)\n",
      "           ^^^^^^^^^^^^^^^^\n",
      "AttributeError: 'tuple' object has no attribute 'format'\n",
      "Call stack:\n",
      "  File \"<frozen runpy>\", line 198, in _run_module_as_main\n",
      "  File \"<frozen runpy>\", line 88, in _run_code\n",
      "  File \"/home/mf019/miniconda3/lib/python3.12/site-packages/ipykernel_launcher.py\", line 18, in <module>\n",
      "    app.launch_new_instance()\n",
      "  File \"/home/mf019/miniconda3/lib/python3.12/site-packages/traitlets/config/application.py\", line 1075, in launch_instance\n",
      "    app.start()\n",
      "  File \"/home/mf019/miniconda3/lib/python3.12/site-packages/ipykernel/kernelapp.py\", line 739, in start\n",
      "    self.io_loop.start()\n",
      "  File \"/home/mf019/miniconda3/lib/python3.12/site-packages/tornado/platform/asyncio.py\", line 205, in start\n",
      "    self.asyncio_loop.run_forever()\n",
      "  File \"/home/mf019/miniconda3/lib/python3.12/asyncio/base_events.py\", line 639, in run_forever\n",
      "    self._run_once()\n",
      "  File \"/home/mf019/miniconda3/lib/python3.12/asyncio/base_events.py\", line 1985, in _run_once\n",
      "    handle._run()\n",
      "  File \"/home/mf019/miniconda3/lib/python3.12/asyncio/events.py\", line 88, in _run\n",
      "    self._context.run(self._callback, *self._args)\n",
      "  File \"/home/mf019/miniconda3/lib/python3.12/site-packages/ipykernel/kernelbase.py\", line 545, in dispatch_queue\n",
      "    await self.process_one()\n",
      "  File \"/home/mf019/miniconda3/lib/python3.12/site-packages/ipykernel/kernelbase.py\", line 534, in process_one\n",
      "    await dispatch(*args)\n",
      "  File \"/home/mf019/miniconda3/lib/python3.12/site-packages/ipykernel/kernelbase.py\", line 437, in dispatch_shell\n",
      "    await result\n",
      "  File \"/home/mf019/miniconda3/lib/python3.12/site-packages/ipykernel/ipkernel.py\", line 359, in execute_request\n",
      "    await super().execute_request(stream, ident, parent)\n",
      "  File \"/home/mf019/miniconda3/lib/python3.12/site-packages/ipykernel/kernelbase.py\", line 778, in execute_request\n",
      "    reply_content = await reply_content\n",
      "  File \"/home/mf019/miniconda3/lib/python3.12/site-packages/ipykernel/ipkernel.py\", line 446, in do_execute\n",
      "    res = shell.run_cell(\n",
      "  File \"/home/mf019/miniconda3/lib/python3.12/site-packages/ipykernel/zmqshell.py\", line 549, in run_cell\n",
      "    return super().run_cell(*args, **kwargs)\n",
      "  File \"/home/mf019/miniconda3/lib/python3.12/site-packages/IPython/core/interactiveshell.py\", line 3075, in run_cell\n",
      "    result = self._run_cell(\n",
      "  File \"/home/mf019/miniconda3/lib/python3.12/site-packages/IPython/core/interactiveshell.py\", line 3130, in _run_cell\n",
      "    result = runner(coro)\n",
      "  File \"/home/mf019/miniconda3/lib/python3.12/site-packages/IPython/core/async_helpers.py\", line 129, in _pseudo_sync_runner\n",
      "    coro.send(None)\n",
      "  File \"/home/mf019/miniconda3/lib/python3.12/site-packages/IPython/core/interactiveshell.py\", line 3334, in run_cell_async\n",
      "    has_raised = await self.run_ast_nodes(code_ast.body, cell_name,\n",
      "  File \"/home/mf019/miniconda3/lib/python3.12/site-packages/IPython/core/interactiveshell.py\", line 3517, in run_ast_nodes\n",
      "    if await self.run_code(code, result, async_=asy):\n",
      "  File \"/home/mf019/miniconda3/lib/python3.12/site-packages/IPython/core/interactiveshell.py\", line 3577, in run_code\n",
      "    exec(code_obj, self.user_global_ns, self.user_ns)\n",
      "  File \"/tmp/ipykernel_4102621/3510340624.py\", line 1, in <module>\n",
      "    mjf = axe.mjf(logfile_name='plasmid_identification_v3_parsing')\n",
      "  File \"/home/mf019/miniconda3/lib/python3.12/site-packages/axe/axe.py\", line 223, in __init__\n",
      "    self.logger.log(msg=f'Console StreamHandler is now online at level: {self.console_level}', level=100)\n",
      "Message: 'Console StreamHandler is now online at level: 10'\n",
      "Arguments: ()\n"
     ]
    }
   ],
   "source": [
    "mjf = axe.mjf(logfile_name='plasmid_identification_v3_parsing')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "testing_set_file = 'dbs/Minimal-PF32-testor-set-March2021.faa'\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [],
   "source": [
    "# get working directory\n",
    "init_cwd = os.getcwd()\n",
    "# Set working directory\n",
    "working_dir = '/home/mf019/borrelia_plasmid_classifier_v3'\n",
    "# path out the databases\n",
    "databases_dir = f'{working_dir}/dbs'\n",
    "whole_plasmid_db = f'{databases_dir}/plasmid_db/plasmid_db'\n",
    "pf32_db = f'{databases_dir}/pf32_db/pf32'\n",
    "# lets specify where parsing tables are to be found!\n",
    "parsing_tables_dir = f'{working_dir}/parsing_tables'\n",
    "# set assemblies directory # DO I WANT TO ARGV THIS?\n",
    "assemblies_dir = '/home/mf019/longread_GWAS/longread_analysis/paired_assemblies/paired_only' #f'{working_dir}/assemblies'\n",
    "shortread_dir  = f'{assemblies_dir}/shortread'\n",
    "longread_dir = f'{assemblies_dir}/longread'\n",
    "shortread_contigs  = f'{shortread_dir}/contigs'\n",
    "longread_contigs = f'{longread_dir}/contigs'\n",
    "shortread_annotations  = f'{shortread_dir}/annotation'\n",
    "longread_annotations = f'{longread_dir}/annotation'\n",
    "output_dir = f'{working_dir}/output'\n",
    "blast_results_dir = f'{output_dir}/blast_results'\n",
    "pf32_blast_results = f'{blast_results_dir}/pf32'\n",
    "plasmid_blast_results = f'{blast_results_dir}/whole_plasmid'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Working directory: /home/mf019/borrelia_plasmid_classifier_v3\n",
      "Databases directory: /home/mf019/borrelia_plasmid_classifier_v3/dbs\n",
      "Plasmid database: /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db/plasmid_db\n",
      "PF32 database: /home/mf019/borrelia_plasmid_classifier_v3/dbs/pf32_db/pf32\n",
      "Parsing tables directory: /home/mf019/borrelia_plasmid_classifier_v3/parsing_tables\n",
      "Assemblies directory: /home/mf019/longread_GWAS/longread_analysis/paired_assemblies/paired_only\n",
      "Shortread directory: /home/mf019/longread_GWAS/longread_analysis/paired_assemblies/paired_only/shortread\n",
      "Longread directory: /home/mf019/longread_GWAS/longread_analysis/paired_assemblies/paired_only/longread\n",
      "Shortread contigs directory: /home/mf019/longread_GWAS/longread_analysis/paired_assemblies/paired_only/shortread/contigs\n",
      "Shortread annotations directory: /home/mf019/longread_GWAS/longread_analysis/paired_assemblies/paired_only/shortread/annotation\n",
      "Longread contigs directory: /home/mf019/longread_GWAS/longread_analysis/paired_assemblies/paired_only/longread/contigs\n",
      "Longread annotations directory: /home/mf019/longread_GWAS/longread_analysis/paired_assemblies/paired_only/longread/annotation\n",
      "Output directory: /home/mf019/borrelia_plasmid_classifier_v3/output\n",
      "Whole contig blast results directory: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results\n",
      "PF32 Blast results: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/pf32\n"
     ]
    }
   ],
   "source": [
    "\n",
    "print(f'Working directory: {working_dir}')\n",
    "print(f'Databases directory: {databases_dir}')\n",
    "print(f'Plasmid database: {whole_plasmid_db}')\n",
    "print(f'PF32 database: {pf32_db}')\n",
    "print(f'Parsing tables directory: {parsing_tables_dir}')\n",
    "print(f'Assemblies directory: {assemblies_dir}')\n",
    "print(f'Shortread directory: {shortread_dir}')\n",
    "print(f'Longread directory: {longread_dir}')\n",
    "print(f'Shortread contigs directory: {shortread_contigs}')\n",
    "print(f'Shortread annotations directory: {shortread_annotations}')\n",
    "print(f'Longread contigs directory: {longread_contigs}')\n",
    "print(f'Longread annotations directory: {longread_annotations}')\n",
    "print(f'Output directory: {output_dir}')\n",
    "print(f'Whole contig blast results directory: {blast_results_dir}')\n",
    "print(f'PF32 Blast results: {pf32_blast_results}')\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Found 49 shortread gbff files\n",
      "Found 49 longread gbff files\n"
     ]
    }
   ],
   "source": [
    "# get annotations for each method\n",
    "sr_gbs = glob.glob(f'{shortread_annotations}/*/*.gbff')\n",
    "print(f'Found {len(sr_gbs)} shortread gbff files')\n",
    "lr_gbs = glob.glob(f'{longread_annotations}/*/*.gbff')\n",
    "print(f'Found {len(lr_gbs)} longread gbff files')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Processing assembly: URI87H\n",
      "contig000001\n",
      "contig000002\n",
      "contig000003\n",
      "contig000004\n",
      "contig000005\n",
      "contig000006\n",
      "contig000007\n",
      "contig000008\n",
      "contig000009\n",
      "contig000010\n",
      "contig000011\n",
      "contig000012\n",
      "contig000013\n",
      "contig000014\n",
      "contig000015\n",
      "contig000016\n",
      "contig000017\n",
      "contig000018\n",
      "contig000019\n",
      "contig000020\n",
      "contig000021\n",
      "contig000022\n",
      "contig000023\n",
      "contig000024\n",
      "contig000025\n",
      "contig000026\n",
      "contig000027\n",
      "contig000028\n",
      "contig000029\n",
      "contig000030\n",
      "contig000031\n",
      "contig000032\n",
      "contig000033\n",
      "contig000034\n",
      "contig000035\n",
      "contig000036\n",
      "contig000037\n",
      "contig000038\n",
      "contig000039\n",
      "contig000040\n",
      "contig000041\n",
      "contig000042\n",
      "contig000043\n",
      "contig000044\n",
      "contig000045\n",
      "contig000046\n",
      "contig000047\n",
      "contig000048\n",
      "contig000049\n",
      "contig000050\n",
      "contig000051\n",
      "contig000052\n",
      "contig000053\n",
      "contig000054\n",
      "contig000055\n",
      "contig000056\n",
      "contig000057\n",
      "contig000058\n",
      "contig000059\n",
      "contig000060\n",
      "contig000061\n",
      "contig000062\n",
      "contig000063\n",
      "contig000064\n",
      "contig000065\n",
      "contig000066\n",
      "longread:URI87['longread'] has 66 contigs\n",
      "longread: URI87['longread'] has 1466 annotated genes\n",
      "No shortread annotations found for URI87\n",
      "Processing assembly: URI34H\n",
      "contig000001\n",
      "contig000002\n",
      "contig000003\n",
      "contig000004\n",
      "contig000005\n",
      "contig000006\n",
      "contig000007\n",
      "contig000008\n",
      "contig000009\n",
      "contig000010\n",
      "contig000011\n",
      "contig000012\n",
      "contig000013\n",
      "contig000014\n",
      "contig000015\n",
      "contig000016\n",
      "contig000017\n",
      "contig000018\n",
      "contig000019\n",
      "contig000020\n",
      "contig000021\n",
      "contig000022\n",
      "contig000023\n",
      "contig000024\n",
      "contig000025\n",
      "contig000026\n",
      "contig000027\n",
      "contig000028\n",
      "contig000029\n",
      "contig000030\n",
      "contig000031\n",
      "contig000032\n",
      "contig000033\n",
      "contig000034\n",
      "contig000035\n",
      "contig000036\n",
      "contig000037\n",
      "contig000038\n",
      "contig000039\n",
      "contig000040\n",
      "contig000041\n",
      "contig000042\n",
      "contig000043\n",
      "contig000044\n",
      "contig000045\n",
      "contig000046\n",
      "contig000047\n",
      "contig000048\n",
      "contig000049\n",
      "contig000050\n",
      "contig000051\n",
      "contig000052\n",
      "contig000053\n",
      "contig000054\n",
      "contig000055\n",
      "contig000056\n",
      "contig000057\n",
      "contig000058\n",
      "contig000059\n",
      "contig000060\n",
      "contig000061\n",
      "contig000062\n",
      "contig000063\n",
      "contig000064\n",
      "contig000065\n",
      "contig000066\n",
      "contig000067\n",
      "contig000068\n",
      "contig000069\n",
      "contig000070\n",
      "contig000071\n",
      "contig000072\n",
      "contig000073\n",
      "contig000074\n",
      "contig000075\n",
      "contig000076\n",
      "contig000077\n",
      "contig000078\n",
      "contig000079\n",
      "longread:URI34['longread'] has 79 contigs\n",
      "longread: URI34['longread'] has 1295 annotated genes\n",
      "No shortread annotations found for URI34\n",
      "Processing assembly: URI88H\n",
      "contig000001\n",
      "contig000002\n",
      "contig000003\n",
      "contig000004\n",
      "contig000005\n",
      "contig000006\n",
      "contig000007\n",
      "contig000008\n",
      "contig000009\n",
      "contig000010\n",
      "contig000011\n",
      "contig000012\n",
      "contig000013\n",
      "contig000014\n",
      "contig000015\n",
      "contig000016\n",
      "contig000017\n",
      "contig000018\n",
      "contig000019\n",
      "contig000020\n",
      "contig000021\n",
      "contig000022\n",
      "contig000023\n",
      "contig000024\n",
      "contig000025\n",
      "contig000026\n",
      "contig000027\n",
      "contig000028\n",
      "contig000029\n",
      "contig000030\n",
      "contig000031\n",
      "contig000032\n",
      "contig000033\n",
      "longread:URI88['longread'] has 33 contigs\n",
      "longread: URI88['longread'] has 1502 annotated genes\n",
      "No shortread annotations found for URI88\n",
      "Processing assembly: URI33H\n",
      "contig000001\n",
      "contig000002\n",
      "contig000003\n",
      "contig000004\n",
      "contig000005\n",
      "contig000006\n",
      "contig000007\n",
      "contig000008\n",
      "contig000009\n",
      "contig000010\n",
      "contig000011\n",
      "contig000012\n",
      "contig000013\n",
      "contig000014\n",
      "contig000015\n",
      "contig000016\n",
      "contig000017\n",
      "contig000018\n",
      "contig000019\n",
      "contig000020\n",
      "contig000021\n",
      "contig000022\n",
      "contig000023\n",
      "contig000024\n",
      "contig000025\n",
      "contig000026\n",
      "contig000027\n",
      "contig000028\n",
      "contig000029\n",
      "contig000030\n",
      "contig000031\n",
      "contig000032\n",
      "contig000033\n",
      "contig000034\n",
      "contig000035\n",
      "contig000036\n",
      "contig000037\n",
      "contig000038\n",
      "contig000039\n",
      "contig000040\n",
      "contig000041\n",
      "contig000042\n",
      "contig000043\n",
      "contig000044\n",
      "longread:URI33['longread'] has 44 contigs\n",
      "longread: URI33['longread'] has 1272 annotated genes\n",
      "No shortread annotations found for URI33\n",
      "Processing assembly: UCT110H\n",
      "contig000001\n",
      "contig000002\n",
      "contig000003\n",
      "contig000004\n",
      "contig000005\n",
      "contig000006\n",
      "contig000007\n",
      "contig000008\n",
      "contig000009\n",
      "contig000010\n",
      "contig000011\n",
      "contig000012\n",
      "contig000013\n",
      "contig000014\n",
      "contig000015\n",
      "contig000016\n",
      "contig000017\n",
      "contig000018\n",
      "contig000019\n",
      "contig000020\n",
      "contig000021\n",
      "contig000022\n",
      "contig000023\n",
      "contig000024\n",
      "contig000025\n",
      "contig000026\n",
      "contig000027\n",
      "contig000028\n",
      "contig000029\n",
      "contig000030\n",
      "contig000031\n",
      "contig000032\n",
      "contig000033\n",
      "contig000034\n",
      "contig000035\n",
      "contig000036\n",
      "contig000037\n",
      "contig000038\n",
      "contig000039\n",
      "contig000040\n",
      "contig000041\n",
      "contig000042\n",
      "contig000043\n",
      "contig000044\n",
      "contig000045\n",
      "contig000046\n",
      "contig000047\n",
      "contig000048\n",
      "contig000049\n",
      "contig000050\n",
      "contig000051\n",
      "contig000052\n",
      "contig000053\n",
      "contig000054\n",
      "contig000055\n",
      "contig000056\n",
      "contig000057\n",
      "contig000058\n",
      "contig000059\n",
      "contig000060\n",
      "contig000061\n",
      "contig000062\n",
      "contig000063\n",
      "contig000064\n",
      "contig000065\n",
      "contig000066\n",
      "contig000067\n",
      "contig000068\n",
      "contig000069\n",
      "contig000070\n",
      "contig000071\n",
      "contig000072\n",
      "contig000073\n",
      "contig000074\n",
      "contig000075\n",
      "contig000076\n",
      "contig000077\n",
      "contig000078\n",
      "contig000079\n",
      "contig000080\n",
      "contig000081\n",
      "contig000082\n",
      "contig000083\n",
      "contig000084\n",
      "contig000085\n",
      "contig000086\n",
      "contig000087\n",
      "contig000088\n",
      "contig000089\n",
      "contig000090\n",
      "contig000091\n",
      "contig000092\n",
      "contig000093\n",
      "contig000094\n",
      "contig000095\n",
      "contig000096\n",
      "contig000097\n",
      "contig000098\n",
      "contig000099\n",
      "contig000100\n",
      "contig000101\n",
      "contig000102\n",
      "contig000103\n",
      "contig000104\n",
      "contig000105\n",
      "contig000106\n",
      "contig000107\n",
      "contig000108\n",
      "contig000109\n",
      "contig000110\n",
      "contig000111\n",
      "contig000112\n",
      "contig000113\n",
      "contig000114\n",
      "longread:UCT110['longread'] has 114 contigs\n",
      "longread: UCT110['longread'] has 1343 annotated genes\n",
      "No shortread annotations found for UCT110\n",
      "Processing assembly: URI39H\n",
      "contig000001\n",
      "contig000002\n",
      "contig000003\n",
      "contig000004\n",
      "contig000005\n",
      "contig000006\n",
      "contig000007\n",
      "contig000008\n",
      "contig000009\n",
      "contig000010\n",
      "contig000011\n",
      "contig000012\n",
      "contig000013\n",
      "contig000014\n",
      "contig000015\n",
      "contig000016\n",
      "contig000017\n",
      "contig000018\n",
      "contig000019\n",
      "contig000020\n",
      "contig000021\n",
      "contig000022\n",
      "contig000023\n",
      "contig000024\n",
      "contig000025\n",
      "contig000026\n",
      "contig000027\n",
      "contig000028\n",
      "longread:URI39['longread'] has 28 contigs\n",
      "longread: URI39['longread'] has 1542 annotated genes\n",
      "No shortread annotations found for URI39\n",
      "Processing assembly: URI91H\n",
      "contig000001\n",
      "contig000002\n",
      "contig000003\n",
      "contig000004\n",
      "contig000005\n",
      "contig000006\n",
      "contig000007\n",
      "contig000008\n",
      "contig000009\n",
      "contig000010\n",
      "contig000011\n",
      "contig000012\n",
      "contig000013\n",
      "contig000014\n",
      "contig000015\n",
      "contig000016\n",
      "contig000017\n",
      "contig000018\n",
      "contig000019\n",
      "contig000020\n",
      "contig000021\n",
      "contig000022\n",
      "contig000023\n",
      "contig000024\n",
      "contig000025\n",
      "contig000026\n",
      "longread:URI91['longread'] has 26 contigs\n",
      "longread: URI91['longread'] has 1574 annotated genes\n",
      "No shortread annotations found for URI91\n",
      "Processing assembly: UCT35H\n",
      "contig000001\n",
      "contig000002\n",
      "contig000003\n",
      "contig000004\n",
      "contig000005\n",
      "contig000006\n",
      "contig000007\n",
      "contig000008\n",
      "contig000009\n",
      "contig000010\n",
      "contig000011\n",
      "contig000012\n",
      "contig000013\n",
      "contig000014\n",
      "contig000015\n",
      "contig000016\n",
      "contig000017\n",
      "contig000018\n",
      "contig000019\n",
      "contig000020\n",
      "contig000021\n",
      "contig000022\n",
      "contig000023\n",
      "contig000024\n",
      "contig000025\n",
      "contig000026\n",
      "contig000027\n",
      "contig000028\n",
      "contig000029\n",
      "contig000030\n",
      "contig000031\n",
      "contig000032\n",
      "contig000033\n",
      "longread:UCT35['longread'] has 33 contigs\n",
      "longread: UCT35['longread'] has 1278 annotated genes\n",
      "No shortread annotations found for UCT35\n",
      "Processing assembly: UWI247P\n",
      "MC104_chr\n",
      "MC104_cp26\n",
      "MC104_cp32-1\n",
      "MC104_cp32-11\n",
      "MC104_cp32-3\n",
      "MC104_cp32-4\n",
      "MC104_cp32-6\n",
      "MC104_cp32-7\n",
      "MC104_lp17\n",
      "MC104_lp25\n",
      "MC104_lp28-1\n",
      "MC104_lp28-3\n",
      "MC104_lp28-4\n",
      "MC104_lp28-5\n",
      "MC104_lp36\n",
      "MC104_lp38\n",
      "MC104_lp54\n",
      "longread:UWI247['longread'] has 17 contigs\n",
      "longread: UWI247['longread'] has 1300 annotated genes\n",
      "No shortread annotations found for UWI247\n",
      "Processing assembly: URI120H\n",
      "contig000001\n",
      "contig000002\n",
      "contig000003\n",
      "contig000004\n",
      "contig000005\n",
      "contig000006\n",
      "contig000007\n",
      "contig000008\n",
      "contig000009\n",
      "contig000010\n",
      "contig000011\n",
      "contig000012\n",
      "contig000013\n",
      "contig000014\n",
      "contig000015\n",
      "contig000016\n",
      "contig000017\n",
      "contig000018\n",
      "contig000019\n",
      "contig000020\n",
      "contig000021\n",
      "contig000022\n",
      "contig000023\n",
      "contig000024\n",
      "contig000025\n",
      "contig000026\n",
      "contig000027\n",
      "contig000028\n",
      "contig000029\n",
      "contig000030\n",
      "contig000031\n",
      "contig000032\n",
      "contig000033\n",
      "contig000034\n",
      "contig000035\n",
      "contig000036\n",
      "contig000037\n",
      "contig000038\n",
      "contig000039\n",
      "contig000040\n",
      "contig000041\n",
      "contig000042\n",
      "contig000043\n",
      "contig000044\n",
      "contig000045\n",
      "contig000046\n",
      "contig000047\n",
      "contig000048\n",
      "contig000049\n",
      "contig000050\n",
      "contig000051\n",
      "contig000052\n",
      "contig000053\n",
      "contig000054\n",
      "contig000055\n",
      "contig000056\n",
      "contig000057\n",
      "contig000058\n",
      "contig000059\n",
      "contig000060\n",
      "contig000061\n",
      "contig000062\n",
      "contig000063\n",
      "contig000064\n",
      "contig000065\n",
      "contig000066\n",
      "contig000067\n",
      "contig000068\n",
      "contig000069\n",
      "contig000070\n",
      "contig000071\n",
      "contig000072\n",
      "contig000073\n",
      "contig000074\n",
      "contig000075\n",
      "contig000076\n",
      "contig000077\n",
      "contig000078\n",
      "contig000079\n",
      "contig000080\n",
      "contig000081\n",
      "contig000082\n",
      "contig000083\n",
      "contig000084\n",
      "contig000085\n",
      "contig000086\n",
      "contig000087\n",
      "contig000088\n",
      "contig000089\n",
      "contig000090\n",
      "contig000091\n",
      "contig000092\n",
      "contig000093\n",
      "contig000094\n",
      "contig000095\n",
      "contig000096\n",
      "contig000097\n",
      "contig000098\n",
      "contig000099\n",
      "contig000100\n",
      "contig000101\n",
      "contig000102\n",
      "contig000103\n",
      "contig000104\n",
      "contig000105\n",
      "longread:URI120['longread'] has 105 contigs\n",
      "longread: URI120['longread'] has 1544 annotated genes\n",
      "No shortread annotations found for URI120\n",
      "Processing assembly: URI107H\n",
      "contig000001\n",
      "contig000002\n",
      "contig000003\n",
      "contig000004\n",
      "contig000005\n",
      "contig000006\n",
      "contig000007\n",
      "contig000008\n",
      "contig000009\n",
      "contig000010\n",
      "contig000011\n",
      "contig000012\n",
      "contig000013\n",
      "contig000014\n",
      "contig000015\n",
      "contig000016\n",
      "contig000017\n",
      "contig000018\n",
      "contig000019\n",
      "contig000020\n",
      "contig000021\n",
      "contig000022\n",
      "contig000023\n",
      "contig000024\n",
      "contig000025\n",
      "contig000026\n",
      "contig000027\n",
      "contig000028\n",
      "contig000029\n",
      "contig000030\n",
      "contig000031\n",
      "contig000032\n",
      "contig000033\n",
      "contig000034\n",
      "contig000035\n",
      "contig000036\n",
      "contig000037\n",
      "contig000038\n",
      "contig000039\n",
      "contig000040\n",
      "contig000041\n",
      "contig000042\n",
      "contig000043\n",
      "contig000044\n",
      "contig000045\n",
      "contig000046\n",
      "contig000047\n",
      "contig000048\n",
      "contig000049\n",
      "longread:URI107['longread'] has 49 contigs\n",
      "longread: URI107['longread'] has 1479 annotated genes\n",
      "No shortread annotations found for URI107\n",
      "Processing assembly: UWI263P\n",
      "MC123_chr\n",
      "MC123_cp26\n",
      "MC123_cp32-11\n",
      "MC123_cp32-12\n",
      "MC123_cp32-4\n",
      "MC123_cp32-5\n",
      "MC123_cp32-6\n",
      "MC123_cp32-7\n",
      "MC123_cp32-9\n",
      "MC123_cp9\n",
      "MC123_lp17\n",
      "MC123_lp25\n",
      "MC123_lp28-1\n",
      "MC123_lp28-3\n",
      "MC123_lp28-4\n",
      "MC123_lp28-5\n",
      "MC123_lp36\n",
      "MC123_lp38\n",
      "MC123_lp54\n",
      "longread:UWI263['longread'] has 19 contigs\n",
      "longread: UWI263['longread'] has 1372 annotated genes\n",
      "No shortread annotations found for UWI263\n",
      "Processing assembly: URI89H\n",
      "contig000001\n",
      "contig000002\n",
      "contig000003\n",
      "contig000004\n",
      "contig000005\n",
      "contig000006\n",
      "contig000007\n",
      "contig000008\n",
      "contig000009\n",
      "contig000010\n",
      "contig000011\n",
      "contig000012\n",
      "contig000013\n",
      "contig000014\n",
      "contig000015\n",
      "contig000016\n",
      "contig000017\n",
      "contig000018\n",
      "contig000019\n",
      "contig000020\n",
      "contig000021\n",
      "contig000022\n",
      "contig000023\n",
      "contig000024\n",
      "contig000025\n",
      "contig000026\n",
      "contig000027\n",
      "contig000028\n",
      "contig000029\n",
      "contig000030\n",
      "contig000031\n",
      "contig000032\n",
      "contig000033\n",
      "contig000034\n",
      "contig000035\n",
      "contig000036\n",
      "contig000037\n",
      "contig000038\n",
      "contig000039\n",
      "contig000040\n",
      "contig000041\n",
      "contig000042\n",
      "contig000043\n",
      "contig000044\n",
      "contig000045\n",
      "contig000046\n",
      "contig000047\n",
      "contig000048\n",
      "contig000049\n",
      "contig000050\n",
      "contig000051\n",
      "contig000052\n",
      "contig000053\n",
      "contig000054\n",
      "contig000055\n",
      "contig000056\n",
      "contig000057\n",
      "contig000058\n",
      "contig000059\n",
      "contig000060\n",
      "contig000061\n",
      "contig000062\n",
      "contig000063\n",
      "contig000064\n",
      "contig000065\n",
      "contig000066\n",
      "contig000067\n",
      "contig000068\n",
      "contig000069\n",
      "contig000070\n",
      "contig000071\n",
      "contig000072\n",
      "contig000073\n",
      "contig000074\n",
      "contig000075\n",
      "contig000076\n",
      "contig000077\n",
      "contig000078\n",
      "contig000079\n",
      "contig000080\n",
      "contig000081\n",
      "contig000082\n",
      "contig000083\n",
      "contig000084\n",
      "contig000085\n",
      "longread:URI89['longread'] has 85 contigs\n",
      "longread: URI89['longread'] has 1526 annotated genes\n",
      "No shortread annotations found for URI89\n",
      "Processing assembly: URI42H\n",
      "contig000001\n",
      "contig000002\n",
      "contig000003\n",
      "contig000004\n",
      "contig000005\n",
      "contig000006\n",
      "contig000007\n",
      "contig000008\n",
      "contig000009\n",
      "contig000010\n",
      "contig000011\n",
      "contig000012\n",
      "contig000013\n",
      "contig000014\n",
      "contig000015\n",
      "contig000016\n",
      "contig000017\n",
      "contig000018\n",
      "contig000019\n",
      "contig000020\n",
      "contig000021\n",
      "contig000022\n",
      "contig000023\n",
      "contig000024\n",
      "contig000025\n",
      "contig000026\n",
      "contig000027\n",
      "contig000028\n",
      "contig000029\n",
      "contig000030\n",
      "contig000031\n",
      "contig000032\n",
      "contig000033\n",
      "contig000034\n",
      "contig000035\n",
      "contig000036\n",
      "contig000037\n",
      "contig000038\n",
      "contig000039\n",
      "contig000040\n",
      "contig000041\n",
      "contig000042\n",
      "contig000043\n",
      "contig000044\n",
      "contig000045\n",
      "contig000046\n",
      "contig000047\n",
      "contig000048\n",
      "contig000049\n",
      "contig000050\n",
      "contig000051\n",
      "contig000052\n",
      "contig000053\n",
      "contig000054\n",
      "contig000055\n",
      "contig000056\n",
      "contig000057\n",
      "contig000058\n",
      "contig000059\n",
      "contig000060\n",
      "contig000061\n",
      "contig000062\n",
      "contig000063\n",
      "contig000064\n",
      "contig000065\n",
      "contig000066\n",
      "contig000067\n",
      "contig000068\n",
      "contig000069\n",
      "contig000070\n",
      "contig000071\n",
      "contig000072\n",
      "contig000073\n",
      "contig000074\n",
      "longread:URI42['longread'] has 74 contigs\n",
      "longread: URI42['longread'] has 1422 annotated genes\n",
      "No shortread annotations found for URI42\n",
      "Processing assembly: URI44H\n",
      "contig000001\n",
      "contig000002\n",
      "contig000003\n",
      "contig000004\n",
      "contig000005\n",
      "contig000006\n",
      "contig000007\n",
      "contig000008\n",
      "contig000009\n",
      "contig000010\n",
      "contig000011\n",
      "contig000012\n",
      "contig000013\n",
      "contig000014\n",
      "contig000015\n",
      "contig000016\n",
      "contig000017\n",
      "contig000018\n",
      "contig000019\n",
      "contig000020\n",
      "contig000021\n",
      "contig000022\n",
      "contig000023\n",
      "contig000024\n",
      "contig000025\n",
      "contig000026\n",
      "contig000027\n",
      "contig000028\n",
      "contig000029\n",
      "contig000030\n",
      "contig000031\n",
      "contig000032\n",
      "contig000033\n",
      "contig000034\n",
      "contig000035\n",
      "contig000036\n",
      "contig000037\n",
      "contig000038\n",
      "contig000039\n",
      "contig000040\n",
      "contig000041\n",
      "contig000042\n",
      "contig000043\n",
      "contig000044\n",
      "contig000045\n",
      "contig000046\n",
      "contig000047\n",
      "contig000048\n",
      "contig000049\n",
      "contig000050\n",
      "contig000051\n",
      "contig000052\n",
      "contig000053\n",
      "contig000054\n",
      "contig000055\n",
      "contig000056\n",
      "contig000057\n",
      "contig000058\n",
      "contig000059\n",
      "contig000060\n",
      "contig000061\n",
      "contig000062\n",
      "contig000063\n",
      "contig000064\n",
      "contig000065\n",
      "contig000066\n",
      "contig000067\n",
      "contig000068\n",
      "contig000069\n",
      "contig000070\n",
      "contig000071\n",
      "contig000072\n",
      "contig000073\n",
      "contig000074\n",
      "contig000075\n",
      "longread:URI44['longread'] has 75 contigs\n",
      "longread: URI44['longread'] has 1341 annotated genes\n",
      "No shortread annotations found for URI44\n",
      "Processing assembly: UCT109H\n",
      "contig000001\n",
      "contig000002\n",
      "contig000003\n",
      "contig000004\n",
      "contig000005\n",
      "contig000006\n",
      "contig000007\n",
      "contig000008\n",
      "contig000009\n",
      "contig000010\n",
      "contig000011\n",
      "contig000012\n",
      "contig000013\n",
      "contig000014\n",
      "contig000015\n",
      "contig000016\n",
      "contig000017\n",
      "contig000018\n",
      "contig000019\n",
      "contig000020\n",
      "contig000021\n",
      "contig000022\n",
      "contig000023\n",
      "contig000024\n",
      "contig000025\n",
      "contig000026\n",
      "contig000027\n",
      "contig000028\n",
      "longread:UCT109['longread'] has 28 contigs\n",
      "longread: UCT109['longread'] has 1438 annotated genes\n",
      "No shortread annotations found for UCT109\n",
      "Processing assembly: URI40H\n",
      "contig000001\n",
      "contig000002\n",
      "contig000003\n",
      "contig000004\n",
      "contig000005\n",
      "contig000006\n",
      "contig000007\n",
      "contig000008\n",
      "contig000009\n",
      "contig000010\n",
      "contig000011\n",
      "contig000012\n",
      "contig000013\n",
      "contig000014\n",
      "contig000015\n",
      "contig000016\n",
      "contig000017\n",
      "contig000018\n",
      "contig000019\n",
      "contig000020\n",
      "contig000021\n",
      "contig000022\n",
      "contig000023\n",
      "contig000024\n",
      "contig000025\n",
      "contig000026\n",
      "contig000027\n",
      "contig000028\n",
      "contig000029\n",
      "contig000030\n",
      "contig000031\n",
      "contig000032\n",
      "contig000033\n",
      "contig000034\n",
      "contig000035\n",
      "contig000036\n",
      "contig000037\n",
      "contig000038\n",
      "contig000039\n",
      "contig000040\n",
      "contig000041\n",
      "contig000042\n",
      "contig000043\n",
      "contig000044\n",
      "contig000045\n",
      "contig000046\n",
      "contig000047\n",
      "contig000048\n",
      "contig000049\n",
      "contig000050\n",
      "contig000051\n",
      "contig000052\n",
      "contig000053\n",
      "contig000054\n",
      "contig000055\n",
      "contig000056\n",
      "contig000057\n",
      "contig000058\n",
      "contig000059\n",
      "contig000060\n",
      "contig000061\n",
      "contig000062\n",
      "contig000063\n",
      "contig000064\n",
      "contig000065\n",
      "contig000066\n",
      "contig000067\n",
      "contig000068\n",
      "contig000069\n",
      "contig000070\n",
      "contig000071\n",
      "contig000072\n",
      "contig000073\n",
      "contig000074\n",
      "contig000075\n",
      "contig000076\n",
      "contig000077\n",
      "contig000078\n",
      "contig000079\n",
      "contig000080\n",
      "contig000081\n",
      "contig000082\n",
      "longread:URI40['longread'] has 82 contigs\n",
      "longread: URI40['longread'] has 1470 annotated genes\n",
      "No shortread annotations found for URI40\n",
      "Processing assembly: URI117H\n",
      "contig000001\n",
      "contig000002\n",
      "contig000003\n",
      "contig000004\n",
      "contig000005\n",
      "contig000006\n",
      "contig000007\n",
      "contig000008\n",
      "contig000009\n",
      "contig000010\n",
      "contig000011\n",
      "contig000012\n",
      "contig000013\n",
      "contig000014\n",
      "contig000015\n",
      "contig000016\n",
      "contig000017\n",
      "contig000018\n",
      "contig000019\n",
      "contig000020\n",
      "contig000021\n",
      "contig000022\n",
      "contig000023\n",
      "contig000024\n",
      "contig000025\n",
      "contig000026\n",
      "contig000027\n",
      "contig000028\n",
      "contig000029\n",
      "contig000030\n",
      "contig000031\n",
      "contig000032\n",
      "contig000033\n",
      "contig000034\n",
      "contig000035\n",
      "contig000036\n",
      "contig000037\n",
      "contig000038\n",
      "contig000039\n",
      "contig000040\n",
      "contig000041\n",
      "contig000042\n",
      "contig000043\n",
      "contig000044\n",
      "contig000045\n",
      "contig000046\n",
      "contig000047\n",
      "contig000048\n",
      "contig000049\n",
      "contig000050\n",
      "contig000051\n",
      "contig000052\n",
      "contig000053\n",
      "contig000054\n",
      "contig000055\n",
      "contig000056\n",
      "contig000057\n",
      "contig000058\n",
      "contig000059\n",
      "contig000060\n",
      "contig000061\n",
      "contig000062\n",
      "contig000063\n",
      "contig000064\n",
      "contig000065\n",
      "contig000066\n",
      "contig000067\n",
      "contig000068\n",
      "contig000069\n",
      "contig000070\n",
      "contig000071\n",
      "contig000072\n",
      "contig000073\n",
      "contig000074\n",
      "contig000075\n",
      "contig000076\n",
      "contig000077\n",
      "contig000078\n",
      "contig000079\n",
      "contig000080\n",
      "contig000081\n",
      "contig000082\n",
      "contig000083\n",
      "contig000084\n",
      "contig000085\n",
      "contig000086\n",
      "contig000087\n",
      "contig000088\n",
      "contig000089\n",
      "contig000090\n",
      "contig000091\n",
      "contig000092\n",
      "contig000093\n",
      "contig000094\n",
      "contig000095\n",
      "contig000096\n",
      "contig000097\n",
      "contig000098\n",
      "contig000099\n",
      "contig000100\n",
      "contig000101\n",
      "contig000102\n",
      "contig000103\n",
      "contig000104\n",
      "contig000105\n",
      "contig000106\n",
      "contig000107\n",
      "contig000108\n",
      "contig000109\n",
      "contig000110\n",
      "contig000111\n",
      "contig000112\n",
      "contig000113\n",
      "contig000114\n",
      "contig000115\n",
      "contig000116\n",
      "contig000117\n",
      "contig000118\n",
      "contig000119\n",
      "contig000120\n",
      "contig000121\n",
      "contig000122\n",
      "contig000123\n",
      "contig000124\n",
      "contig000125\n",
      "contig000126\n",
      "contig000127\n",
      "contig000128\n",
      "contig000129\n",
      "contig000130\n",
      "contig000131\n",
      "contig000132\n",
      "contig000133\n",
      "contig000134\n",
      "contig000135\n",
      "contig000136\n",
      "contig000137\n",
      "contig000138\n",
      "contig000139\n",
      "contig000140\n",
      "contig000141\n",
      "contig000142\n",
      "contig000143\n",
      "contig000144\n",
      "contig000145\n",
      "contig000146\n",
      "contig000147\n",
      "contig000148\n",
      "contig000149\n",
      "contig000150\n",
      "contig000151\n",
      "contig000152\n",
      "contig000153\n",
      "contig000154\n",
      "contig000155\n",
      "contig000156\n",
      "contig000157\n",
      "contig000158\n",
      "contig000159\n",
      "contig000160\n",
      "longread:URI117['longread'] has 160 contigs\n",
      "longread: URI117['longread'] has 1299 annotated genes\n",
      "No shortread annotations found for URI117\n",
      "Processing assembly: URI47H\n",
      "contig000001\n",
      "contig000002\n",
      "contig000003\n",
      "contig000004\n",
      "contig000005\n",
      "contig000006\n",
      "contig000007\n",
      "contig000008\n",
      "contig000009\n",
      "contig000010\n",
      "contig000011\n",
      "contig000012\n",
      "contig000013\n",
      "contig000014\n",
      "contig000015\n",
      "contig000016\n",
      "contig000017\n",
      "contig000018\n",
      "contig000019\n",
      "contig000020\n",
      "contig000021\n",
      "contig000022\n",
      "contig000023\n",
      "contig000024\n",
      "contig000025\n",
      "contig000026\n",
      "contig000027\n",
      "contig000028\n",
      "contig000029\n",
      "longread:URI47['longread'] has 29 contigs\n",
      "longread: URI47['longread'] has 1477 annotated genes\n",
      "No shortread annotations found for URI47\n",
      "Processing assembly: URI86H\n",
      "contig000001\n",
      "contig000002\n",
      "contig000003\n",
      "contig000004\n",
      "contig000005\n",
      "contig000006\n",
      "contig000007\n",
      "contig000008\n",
      "contig000009\n",
      "contig000010\n",
      "contig000011\n",
      "contig000012\n",
      "contig000013\n",
      "contig000014\n",
      "contig000015\n",
      "contig000016\n",
      "contig000017\n",
      "contig000018\n",
      "contig000019\n",
      "contig000020\n",
      "contig000021\n",
      "contig000022\n",
      "contig000023\n",
      "contig000024\n",
      "contig000025\n",
      "contig000026\n",
      "contig000027\n",
      "contig000028\n",
      "contig000029\n",
      "contig000030\n",
      "contig000031\n",
      "contig000032\n",
      "contig000033\n",
      "contig000034\n",
      "contig000035\n",
      "contig000036\n",
      "contig000037\n",
      "contig000038\n",
      "contig000039\n",
      "contig000040\n",
      "contig000041\n",
      "contig000042\n",
      "contig000043\n",
      "contig000044\n",
      "contig000045\n",
      "contig000046\n",
      "contig000047\n",
      "contig000048\n",
      "contig000049\n",
      "contig000050\n",
      "contig000051\n",
      "contig000052\n",
      "contig000053\n",
      "contig000054\n",
      "contig000055\n",
      "contig000056\n",
      "contig000057\n",
      "contig000058\n",
      "contig000059\n",
      "contig000060\n",
      "contig000061\n",
      "contig000062\n",
      "contig000063\n",
      "contig000064\n",
      "contig000065\n",
      "contig000066\n",
      "contig000067\n",
      "contig000068\n",
      "contig000069\n",
      "contig000070\n",
      "contig000071\n",
      "contig000072\n",
      "contig000073\n",
      "contig000074\n",
      "contig000075\n",
      "contig000076\n",
      "contig000077\n",
      "contig000078\n",
      "contig000079\n",
      "contig000080\n",
      "contig000081\n",
      "contig000082\n",
      "contig000083\n",
      "contig000084\n",
      "contig000085\n",
      "contig000086\n",
      "contig000087\n",
      "contig000088\n",
      "longread:URI86['longread'] has 88 contigs\n",
      "longread: URI86['longread'] has 1329 annotated genes\n",
      "No shortread annotations found for URI86\n",
      "Processing assembly: URI36H\n",
      "contig000001\n",
      "contig000002\n",
      "contig000003\n",
      "contig000004\n",
      "contig000005\n",
      "contig000006\n",
      "contig000007\n",
      "contig000008\n",
      "contig000009\n",
      "contig000010\n",
      "contig000011\n",
      "contig000012\n",
      "contig000013\n",
      "contig000014\n",
      "contig000015\n",
      "contig000016\n",
      "contig000017\n",
      "contig000018\n",
      "contig000019\n",
      "contig000020\n",
      "contig000021\n",
      "contig000022\n",
      "contig000023\n",
      "contig000024\n",
      "contig000025\n",
      "contig000026\n",
      "contig000027\n",
      "contig000028\n",
      "contig000029\n",
      "contig000030\n",
      "contig000031\n",
      "contig000032\n",
      "contig000033\n",
      "contig000034\n",
      "contig000035\n",
      "contig000036\n",
      "contig000037\n",
      "contig000038\n",
      "contig000039\n",
      "contig000040\n",
      "contig000041\n",
      "contig000042\n",
      "contig000043\n",
      "contig000044\n",
      "contig000045\n",
      "contig000046\n",
      "contig000047\n",
      "contig000048\n",
      "contig000049\n",
      "contig000050\n",
      "contig000051\n",
      "contig000052\n",
      "contig000053\n",
      "longread:URI36['longread'] has 53 contigs\n",
      "longread: URI36['longread'] has 1423 annotated genes\n",
      "No shortread annotations found for URI36\n",
      "Processing assembly: UNY208P\n",
      "MR641_chr\n",
      "MR641_cp26\n",
      "MR641_cp32-1\n",
      "MR641_cp32-11\n",
      "MR641_cp32-12\n",
      "MR641_cp32-4\n",
      "MR641_cp32-5\n",
      "MR641_cp32-7\n",
      "MR641_cp32-8\n",
      "MR641_cp32-x\n",
      "MR641_lp17\n",
      "MR641_lp25\n",
      "MR641_lp28-1\n",
      "MR641_lp28-3\n",
      "MR641_lp28-4\n",
      "MR641_lp28-5\n",
      "MR641_lp28-6\n",
      "MR641_lp36\n",
      "MR641_lp54\n",
      "longread:UNY208['longread'] has 19 contigs\n",
      "longread: UNY208['longread'] has 1376 annotated genes\n",
      "No shortread annotations found for UNY208\n",
      "Processing assembly: ESI26H\n",
      "contig000001\n",
      "contig000002\n",
      "contig000003\n",
      "contig000004\n",
      "contig000005\n",
      "contig000006\n",
      "contig000007\n",
      "contig000008\n",
      "contig000009\n",
      "contig000010\n",
      "contig000011\n",
      "contig000012\n",
      "contig000013\n",
      "contig000014\n",
      "contig000015\n",
      "contig000016\n",
      "contig000017\n",
      "contig000018\n",
      "contig000019\n",
      "contig000020\n",
      "contig000021\n",
      "contig000022\n",
      "contig000023\n",
      "contig000024\n",
      "contig000025\n",
      "contig000026\n",
      "contig000027\n",
      "contig000028\n",
      "contig000029\n",
      "contig000030\n",
      "contig000031\n",
      "contig000032\n",
      "contig000033\n",
      "contig000034\n",
      "contig000035\n",
      "contig000036\n",
      "contig000037\n",
      "contig000038\n",
      "contig000039\n",
      "contig000040\n",
      "contig000041\n",
      "contig000042\n",
      "contig000043\n",
      "contig000044\n",
      "contig000045\n",
      "contig000046\n",
      "contig000047\n",
      "contig000048\n",
      "contig000049\n",
      "contig000050\n",
      "contig000051\n",
      "contig000052\n",
      "contig000053\n",
      "contig000054\n",
      "contig000055\n",
      "contig000056\n",
      "contig000057\n",
      "contig000058\n",
      "contig000059\n",
      "contig000060\n",
      "contig000061\n",
      "contig000062\n",
      "contig000063\n",
      "contig000064\n",
      "longread:ESI26['longread'] has 64 contigs\n",
      "longread: ESI26['longread'] has 1213 annotated genes\n",
      "No shortread annotations found for ESI26\n",
      "Processing assembly: UCT31H\n",
      "contig000001\n",
      "contig000002\n",
      "contig000003\n",
      "contig000004\n",
      "contig000005\n",
      "contig000006\n",
      "contig000007\n",
      "contig000008\n",
      "contig000009\n",
      "contig000010\n",
      "contig000011\n",
      "contig000012\n",
      "contig000013\n",
      "contig000014\n",
      "contig000015\n",
      "contig000016\n",
      "contig000017\n",
      "contig000018\n",
      "contig000019\n",
      "contig000020\n",
      "contig000021\n",
      "contig000022\n",
      "contig000023\n",
      "contig000024\n",
      "contig000025\n",
      "contig000026\n",
      "contig000027\n",
      "contig000028\n",
      "contig000029\n",
      "contig000030\n",
      "contig000031\n",
      "contig000032\n",
      "longread:UCT31['longread'] has 32 contigs\n",
      "longread: UCT31['longread'] has 1514 annotated genes\n",
      "No shortread annotations found for UCT31\n",
      "Processing assembly: URI56H\n",
      "contig000001\n",
      "contig000002\n",
      "contig000003\n",
      "contig000004\n",
      "contig000005\n",
      "contig000006\n",
      "contig000007\n",
      "contig000008\n",
      "contig000009\n",
      "contig000010\n",
      "contig000011\n",
      "contig000012\n",
      "contig000013\n",
      "contig000014\n",
      "contig000015\n",
      "contig000016\n",
      "contig000017\n",
      "contig000018\n",
      "contig000019\n",
      "contig000020\n",
      "contig000021\n",
      "contig000022\n",
      "contig000023\n",
      "contig000024\n",
      "contig000025\n",
      "contig000026\n",
      "contig000027\n",
      "contig000028\n",
      "contig000029\n",
      "contig000030\n",
      "contig000031\n",
      "contig000032\n",
      "contig000033\n",
      "contig000034\n",
      "contig000035\n",
      "contig000036\n",
      "contig000037\n",
      "contig000038\n",
      "contig000039\n",
      "contig000040\n",
      "contig000041\n",
      "contig000042\n",
      "contig000043\n",
      "contig000044\n",
      "contig000045\n",
      "contig000046\n",
      "contig000047\n",
      "contig000048\n",
      "contig000049\n",
      "contig000050\n",
      "contig000051\n",
      "contig000052\n",
      "contig000053\n",
      "contig000054\n",
      "contig000055\n",
      "contig000056\n",
      "contig000057\n",
      "contig000058\n",
      "contig000059\n",
      "contig000060\n",
      "contig000061\n",
      "contig000062\n",
      "contig000063\n",
      "contig000064\n",
      "contig000065\n",
      "contig000066\n",
      "contig000067\n",
      "contig000068\n",
      "contig000069\n",
      "contig000070\n",
      "contig000071\n",
      "contig000072\n",
      "contig000073\n",
      "contig000074\n",
      "contig000075\n",
      "contig000076\n",
      "contig000077\n",
      "contig000078\n",
      "contig000079\n",
      "contig000080\n",
      "contig000081\n",
      "contig000082\n",
      "contig000083\n",
      "contig000084\n",
      "contig000085\n",
      "contig000086\n",
      "contig000087\n",
      "contig000088\n",
      "contig000089\n",
      "contig000090\n",
      "contig000091\n",
      "contig000092\n",
      "contig000093\n",
      "contig000094\n",
      "contig000095\n",
      "contig000096\n",
      "contig000097\n",
      "contig000098\n",
      "contig000099\n",
      "contig000100\n",
      "contig000101\n",
      "contig000102\n",
      "contig000103\n",
      "contig000104\n",
      "contig000105\n",
      "contig000106\n",
      "contig000107\n",
      "contig000108\n",
      "contig000109\n",
      "contig000110\n",
      "contig000111\n",
      "contig000112\n",
      "contig000113\n",
      "contig000114\n",
      "contig000115\n",
      "contig000116\n",
      "contig000117\n",
      "contig000118\n",
      "contig000119\n",
      "contig000120\n",
      "contig000121\n",
      "contig000122\n",
      "contig000123\n",
      "contig000124\n",
      "contig000125\n",
      "contig000126\n",
      "contig000127\n",
      "contig000128\n",
      "contig000129\n",
      "contig000130\n",
      "contig000131\n",
      "contig000132\n",
      "contig000133\n",
      "contig000134\n",
      "contig000135\n",
      "contig000136\n",
      "contig000137\n",
      "contig000138\n",
      "contig000139\n",
      "contig000140\n",
      "contig000141\n",
      "contig000142\n",
      "contig000143\n",
      "contig000144\n",
      "contig000145\n",
      "contig000146\n",
      "contig000147\n",
      "contig000148\n",
      "contig000149\n",
      "contig000150\n",
      "contig000151\n",
      "contig000152\n",
      "contig000153\n",
      "contig000154\n",
      "contig000155\n",
      "contig000156\n",
      "contig000157\n",
      "contig000158\n",
      "contig000159\n",
      "contig000160\n",
      "contig000161\n",
      "contig000162\n",
      "contig000163\n",
      "contig000164\n",
      "contig000165\n",
      "contig000166\n",
      "contig000167\n",
      "contig000168\n",
      "contig000169\n",
      "contig000170\n",
      "contig000171\n",
      "contig000172\n",
      "contig000173\n",
      "contig000174\n",
      "contig000175\n",
      "contig000176\n",
      "contig000177\n",
      "contig000178\n",
      "contig000179\n",
      "contig000180\n",
      "contig000181\n",
      "contig000182\n",
      "contig000183\n",
      "contig000184\n",
      "contig000185\n",
      "contig000186\n",
      "contig000187\n",
      "contig000188\n",
      "contig000189\n",
      "contig000190\n",
      "contig000191\n",
      "contig000192\n",
      "contig000193\n",
      "contig000194\n",
      "contig000195\n",
      "contig000196\n",
      "contig000197\n",
      "contig000198\n",
      "contig000199\n",
      "contig000200\n",
      "contig000201\n",
      "contig000202\n",
      "contig000203\n",
      "contig000204\n",
      "contig000205\n",
      "contig000206\n",
      "contig000207\n",
      "contig000208\n",
      "contig000209\n",
      "contig000210\n",
      "contig000211\n",
      "contig000212\n",
      "contig000213\n",
      "contig000214\n",
      "contig000215\n",
      "contig000216\n",
      "contig000217\n",
      "contig000218\n",
      "contig000219\n",
      "contig000220\n",
      "contig000221\n",
      "contig000222\n",
      "contig000223\n",
      "contig000224\n",
      "contig000225\n",
      "contig000226\n",
      "contig000227\n",
      "contig000228\n",
      "contig000229\n",
      "contig000230\n",
      "longread:URI56['longread'] has 230 contigs\n",
      "longread: URI56['longread'] has 1187 annotated genes\n",
      "No shortread annotations found for URI56\n",
      "Processing assembly: UCT30H\n",
      "contig000001\n",
      "contig000002\n",
      "contig000003\n",
      "contig000004\n",
      "contig000005\n",
      "contig000006\n",
      "contig000007\n",
      "contig000008\n",
      "contig000009\n",
      "contig000010\n",
      "contig000011\n",
      "contig000012\n",
      "contig000013\n",
      "contig000014\n",
      "contig000015\n",
      "contig000016\n",
      "contig000017\n",
      "contig000018\n",
      "contig000019\n",
      "contig000020\n",
      "contig000021\n",
      "contig000022\n",
      "contig000023\n",
      "contig000024\n",
      "contig000025\n",
      "contig000026\n",
      "contig000027\n",
      "contig000028\n",
      "contig000029\n",
      "contig000030\n",
      "contig000031\n",
      "contig000032\n",
      "contig000033\n",
      "contig000034\n",
      "contig000035\n",
      "contig000036\n",
      "contig000037\n",
      "contig000038\n",
      "contig000039\n",
      "contig000040\n",
      "contig000041\n",
      "contig000042\n",
      "contig000043\n",
      "contig000044\n",
      "contig000045\n",
      "contig000046\n",
      "contig000047\n",
      "contig000048\n",
      "contig000049\n",
      "contig000050\n",
      "contig000051\n",
      "contig000052\n",
      "contig000053\n",
      "contig000054\n",
      "contig000055\n",
      "contig000056\n",
      "contig000057\n",
      "contig000058\n",
      "contig000059\n",
      "contig000060\n",
      "contig000061\n",
      "contig000062\n",
      "contig000063\n",
      "contig000064\n",
      "contig000065\n",
      "contig000066\n",
      "contig000067\n",
      "contig000068\n",
      "contig000069\n",
      "contig000070\n",
      "contig000071\n",
      "contig000072\n",
      "contig000073\n",
      "contig000074\n",
      "contig000075\n",
      "contig000076\n",
      "contig000077\n",
      "contig000078\n",
      "contig000079\n",
      "contig000080\n",
      "contig000081\n",
      "contig000082\n",
      "contig000083\n",
      "contig000084\n",
      "contig000085\n",
      "contig000086\n",
      "contig000087\n",
      "contig000088\n",
      "contig000089\n",
      "longread:UCT30['longread'] has 89 contigs\n",
      "longread: UCT30['longread'] has 1385 annotated genes\n",
      "No shortread annotations found for UCT30\n",
      "Processing assembly: URI103H\n",
      "contig000001\n",
      "contig000002\n",
      "contig000003\n",
      "contig000004\n",
      "contig000005\n",
      "contig000006\n",
      "contig000007\n",
      "contig000008\n",
      "contig000009\n",
      "contig000010\n",
      "contig000011\n",
      "contig000012\n",
      "contig000013\n",
      "contig000014\n",
      "contig000015\n",
      "contig000016\n",
      "contig000017\n",
      "contig000018\n",
      "contig000019\n",
      "contig000020\n",
      "contig000021\n",
      "contig000022\n",
      "contig000023\n",
      "contig000024\n",
      "contig000025\n",
      "contig000026\n",
      "contig000027\n",
      "contig000028\n",
      "contig000029\n",
      "contig000030\n",
      "contig000031\n",
      "contig000032\n",
      "contig000033\n",
      "contig000034\n",
      "contig000035\n",
      "contig000036\n",
      "contig000037\n",
      "contig000038\n",
      "contig000039\n",
      "contig000040\n",
      "contig000041\n",
      "contig000042\n",
      "contig000043\n",
      "contig000044\n",
      "contig000045\n",
      "contig000046\n",
      "contig000047\n",
      "contig000048\n",
      "contig000049\n",
      "contig000050\n",
      "contig000051\n",
      "contig000052\n",
      "contig000053\n",
      "contig000054\n",
      "contig000055\n",
      "contig000056\n",
      "longread:URI103['longread'] has 56 contigs\n",
      "longread: URI103['longread'] has 1424 annotated genes\n",
      "No shortread annotations found for URI103\n",
      "Processing assembly: UCT29H\n",
      "contig000001\n",
      "contig000002\n",
      "contig000003\n",
      "contig000004\n",
      "contig000005\n",
      "contig000006\n",
      "contig000007\n",
      "contig000008\n",
      "contig000009\n",
      "contig000010\n",
      "contig000011\n",
      "contig000012\n",
      "contig000013\n",
      "contig000014\n",
      "contig000015\n",
      "contig000016\n",
      "contig000017\n",
      "contig000018\n",
      "contig000019\n",
      "contig000020\n",
      "contig000021\n",
      "contig000022\n",
      "contig000023\n",
      "contig000024\n",
      "longread:UCT29['longread'] has 24 contigs\n",
      "longread: UCT29['longread'] has 1547 annotated genes\n",
      "No shortread annotations found for UCT29\n",
      "Processing assembly: URI112H\n",
      "contig000001\n",
      "contig000002\n",
      "contig000003\n",
      "contig000004\n",
      "contig000005\n",
      "contig000006\n",
      "contig000007\n",
      "contig000008\n",
      "contig000009\n",
      "contig000010\n",
      "contig000011\n",
      "contig000012\n",
      "contig000013\n",
      "contig000014\n",
      "contig000015\n",
      "contig000016\n",
      "contig000017\n",
      "contig000018\n",
      "contig000019\n",
      "contig000020\n",
      "contig000021\n",
      "contig000022\n",
      "contig000023\n",
      "contig000024\n",
      "contig000025\n",
      "contig000026\n",
      "contig000027\n",
      "contig000028\n",
      "contig000029\n",
      "contig000030\n",
      "contig000031\n",
      "contig000032\n",
      "contig000033\n",
      "contig000034\n",
      "contig000035\n",
      "contig000036\n",
      "contig000037\n",
      "contig000038\n",
      "contig000039\n",
      "contig000040\n",
      "contig000041\n",
      "contig000042\n",
      "contig000043\n",
      "contig000044\n",
      "contig000045\n",
      "contig000046\n",
      "contig000047\n",
      "contig000048\n",
      "contig000049\n",
      "contig000050\n",
      "contig000051\n",
      "contig000052\n",
      "contig000053\n",
      "contig000054\n",
      "contig000055\n",
      "contig000056\n",
      "contig000057\n",
      "contig000058\n",
      "contig000059\n",
      "contig000060\n",
      "contig000061\n",
      "contig000062\n",
      "contig000063\n",
      "contig000064\n",
      "contig000065\n",
      "contig000066\n",
      "contig000067\n",
      "contig000068\n",
      "contig000069\n",
      "contig000070\n",
      "contig000071\n",
      "contig000072\n",
      "contig000073\n",
      "contig000074\n",
      "contig000075\n",
      "contig000076\n",
      "contig000077\n",
      "contig000078\n",
      "contig000079\n",
      "contig000080\n",
      "contig000081\n",
      "contig000082\n",
      "contig000083\n",
      "contig000084\n",
      "contig000085\n",
      "longread:URI112['longread'] has 85 contigs\n",
      "longread: URI112['longread'] has 1366 annotated genes\n",
      "No shortread annotations found for URI112\n",
      "Processing assembly: UWI248P\n",
      "MC105_chr\n",
      "MC105_cp26\n",
      "MC105_cp32-1\n",
      "MC105_cp32-11\n",
      "MC105_cp32-12\n",
      "MC105_cp32-3\n",
      "MC105_cp32-4\n",
      "MC105_cp32-6\n",
      "MC105_cp32-7\n",
      "MC105_lp17\n",
      "MC105_lp25\n",
      "MC105_lp28-1\n",
      "MC105_lp28-3\n",
      "MC105_lp28-4\n",
      "MC105_lp28-5\n",
      "MC105_lp36\n",
      "MC105_lp38\n",
      "MC105_lp54\n",
      "longread:UWI248['longread'] has 18 contigs\n",
      "longread: UWI248['longread'] has 1337 annotated genes\n",
      "No shortread annotations found for UWI248\n",
      "Processing assembly: UNY203P\n",
      "MR616_chr\n",
      "MR616_cp26\n",
      "MR616_cp32-10\n",
      "MR616_cp32-12\n",
      "MR616_cp32-4\n",
      "MR616_cp32-5\n",
      "MR616_cp32-6\n",
      "MR616_cp32-7\n",
      "MR616_cp9\n",
      "MR616_lp17\n",
      "MR616_lp28-3\n",
      "MR616_lp28-5\n",
      "MR616_lp28-6\n",
      "MR616_lp36\n",
      "MR616_lp38\n",
      "MR616_lp5\n",
      "MR616_lp54\n",
      "longread:UNY203['longread'] has 17 contigs\n",
      "longread: UNY203['longread'] has 1289 annotated genes\n",
      "No shortread annotations found for UNY203\n",
      "Processing assembly: UCT96H\n",
      "contig000001\n",
      "contig000002\n",
      "contig000003\n",
      "contig000004\n",
      "contig000005\n",
      "contig000006\n",
      "contig000007\n",
      "contig000008\n",
      "contig000009\n",
      "contig000010\n",
      "contig000011\n",
      "contig000012\n",
      "contig000013\n",
      "contig000014\n",
      "contig000015\n",
      "contig000016\n",
      "contig000017\n",
      "contig000018\n",
      "contig000019\n",
      "contig000020\n",
      "contig000021\n",
      "contig000022\n",
      "contig000023\n",
      "contig000024\n",
      "contig000025\n",
      "contig000026\n",
      "contig000027\n",
      "contig000028\n",
      "contig000029\n",
      "contig000030\n",
      "contig000031\n",
      "contig000032\n",
      "contig000033\n",
      "contig000034\n",
      "contig000035\n",
      "longread:UCT96['longread'] has 35 contigs\n",
      "longread: UCT96['longread'] has 1451 annotated genes\n",
      "No shortread annotations found for UCT96\n",
      "Processing assembly: UCT32H\n",
      "contig000001\n",
      "contig000002\n",
      "contig000003\n",
      "contig000004\n",
      "contig000005\n",
      "contig000006\n",
      "contig000007\n",
      "contig000008\n",
      "contig000009\n",
      "contig000010\n",
      "contig000011\n",
      "contig000012\n",
      "contig000013\n",
      "contig000014\n",
      "contig000015\n",
      "contig000016\n",
      "contig000017\n",
      "contig000018\n",
      "contig000019\n",
      "contig000020\n",
      "contig000021\n",
      "contig000022\n",
      "contig000023\n",
      "contig000024\n",
      "contig000025\n",
      "contig000026\n",
      "contig000027\n",
      "contig000028\n",
      "contig000029\n",
      "contig000030\n",
      "contig000031\n",
      "contig000032\n",
      "contig000033\n",
      "contig000034\n",
      "contig000035\n",
      "contig000036\n",
      "contig000037\n",
      "contig000038\n",
      "contig000039\n",
      "contig000040\n",
      "contig000041\n",
      "contig000042\n",
      "contig000043\n",
      "contig000044\n",
      "contig000045\n",
      "contig000046\n",
      "contig000047\n",
      "contig000048\n",
      "contig000049\n",
      "contig000050\n",
      "contig000051\n",
      "contig000052\n",
      "contig000053\n",
      "contig000054\n",
      "contig000055\n",
      "contig000056\n",
      "contig000057\n",
      "longread:UCT32['longread'] has 57 contigs\n",
      "longread: UCT32['longread'] has 1459 annotated genes\n",
      "No shortread annotations found for UCT32\n",
      "Processing assembly: UNY193P\n",
      "B477_chr\n",
      "B477_cp26\n",
      "B477_cp32-10\n",
      "B477_cp32-12\n",
      "B477_cp32-5\n",
      "B477_cp32-6\n",
      "B477_cp32-7\n",
      "B477_cp9\n",
      "B477_lp17\n",
      "B477_lp25\n",
      "B477_lp28-3\n",
      "B477_lp28-4\n",
      "B477_lp28-6\n",
      "B477_lp28-7\n",
      "B477_lp28-x\n",
      "B477_lp36\n",
      "B477_lp38\n",
      "B477_lp54\n",
      "B477_lp5\n",
      "longread:UNY193['longread'] has 19 contigs\n",
      "longread: UNY193['longread'] has 1358 annotated genes\n",
      "No shortread annotations found for UNY193\n",
      "Processing assembly: UCT113H\n",
      "contig000001\n",
      "contig000002\n",
      "contig000003\n",
      "contig000004\n",
      "contig000005\n",
      "contig000006\n",
      "contig000007\n",
      "contig000008\n",
      "contig000009\n",
      "contig000010\n",
      "contig000011\n",
      "contig000012\n",
      "contig000013\n",
      "contig000014\n",
      "contig000015\n",
      "contig000016\n",
      "contig000017\n",
      "contig000018\n",
      "contig000019\n",
      "contig000020\n",
      "contig000021\n",
      "contig000022\n",
      "contig000023\n",
      "contig000024\n",
      "contig000025\n",
      "contig000026\n",
      "contig000027\n",
      "contig000028\n",
      "contig000029\n",
      "contig000030\n",
      "contig000031\n",
      "contig000032\n",
      "contig000033\n",
      "contig000034\n",
      "contig000035\n",
      "contig000036\n",
      "contig000037\n",
      "contig000038\n",
      "contig000039\n",
      "contig000040\n",
      "contig000041\n",
      "contig000042\n",
      "contig000043\n",
      "contig000044\n",
      "contig000045\n",
      "contig000046\n",
      "contig000047\n",
      "contig000048\n",
      "contig000049\n",
      "contig000050\n",
      "contig000051\n",
      "contig000052\n",
      "contig000053\n",
      "contig000054\n",
      "contig000055\n",
      "contig000056\n",
      "contig000057\n",
      "contig000058\n",
      "contig000059\n",
      "contig000060\n",
      "contig000061\n",
      "contig000062\n",
      "contig000063\n",
      "contig000064\n",
      "contig000065\n",
      "contig000066\n",
      "contig000067\n",
      "contig000068\n",
      "contig000069\n",
      "contig000070\n",
      "contig000071\n",
      "contig000072\n",
      "contig000073\n",
      "contig000074\n",
      "contig000075\n",
      "contig000076\n",
      "contig000077\n",
      "contig000078\n",
      "contig000079\n",
      "contig000080\n",
      "contig000081\n",
      "contig000082\n",
      "contig000083\n",
      "contig000084\n",
      "contig000085\n",
      "contig000086\n",
      "contig000087\n",
      "contig000088\n",
      "contig000089\n",
      "contig000090\n",
      "contig000091\n",
      "contig000092\n",
      "contig000093\n",
      "contig000094\n",
      "contig000095\n",
      "contig000096\n",
      "contig000097\n",
      "contig000098\n",
      "contig000099\n",
      "contig000100\n",
      "contig000101\n",
      "contig000102\n",
      "contig000103\n",
      "contig000104\n",
      "contig000105\n",
      "contig000106\n",
      "contig000107\n",
      "contig000108\n",
      "contig000109\n",
      "contig000110\n",
      "contig000111\n",
      "contig000112\n",
      "contig000113\n",
      "contig000114\n",
      "contig000115\n",
      "contig000116\n",
      "contig000117\n",
      "contig000118\n",
      "contig000119\n",
      "contig000120\n",
      "contig000121\n",
      "contig000122\n",
      "contig000123\n",
      "contig000124\n",
      "contig000125\n",
      "contig000126\n",
      "contig000127\n",
      "contig000128\n",
      "contig000129\n",
      "contig000130\n",
      "contig000131\n",
      "contig000132\n",
      "contig000133\n",
      "contig000134\n",
      "contig000135\n",
      "contig000136\n",
      "contig000137\n",
      "contig000138\n",
      "contig000139\n",
      "contig000140\n",
      "contig000141\n",
      "contig000142\n",
      "contig000143\n",
      "contig000144\n",
      "contig000145\n",
      "contig000146\n",
      "contig000147\n",
      "contig000148\n",
      "contig000149\n",
      "contig000150\n",
      "contig000151\n",
      "contig000152\n",
      "contig000153\n",
      "contig000154\n",
      "contig000155\n",
      "contig000156\n",
      "contig000157\n",
      "contig000158\n",
      "contig000159\n",
      "contig000160\n",
      "contig000161\n",
      "contig000162\n",
      "contig000163\n",
      "contig000164\n",
      "contig000165\n",
      "contig000166\n",
      "contig000167\n",
      "contig000168\n",
      "contig000169\n",
      "contig000170\n",
      "contig000171\n",
      "contig000172\n",
      "contig000173\n",
      "contig000174\n",
      "contig000175\n",
      "contig000176\n",
      "contig000177\n",
      "contig000178\n",
      "contig000179\n",
      "contig000180\n",
      "contig000181\n",
      "contig000182\n",
      "contig000183\n",
      "contig000184\n",
      "contig000185\n",
      "contig000186\n",
      "contig000187\n",
      "contig000188\n",
      "contig000189\n",
      "contig000190\n",
      "contig000191\n",
      "contig000192\n",
      "contig000193\n",
      "contig000194\n",
      "contig000195\n",
      "contig000196\n",
      "contig000197\n",
      "contig000198\n",
      "contig000199\n",
      "contig000200\n",
      "contig000201\n",
      "contig000202\n",
      "contig000203\n",
      "contig000204\n",
      "contig000205\n",
      "contig000206\n",
      "contig000207\n",
      "contig000208\n",
      "contig000209\n",
      "contig000210\n",
      "contig000211\n",
      "contig000212\n",
      "contig000213\n",
      "contig000214\n",
      "contig000215\n",
      "contig000216\n",
      "contig000217\n",
      "contig000218\n",
      "contig000219\n",
      "contig000220\n",
      "contig000221\n",
      "contig000222\n",
      "contig000223\n",
      "contig000224\n",
      "contig000225\n",
      "contig000226\n",
      "contig000227\n",
      "contig000228\n",
      "contig000229\n",
      "contig000230\n",
      "contig000231\n",
      "contig000232\n",
      "contig000233\n",
      "contig000234\n",
      "contig000235\n",
      "contig000236\n",
      "contig000237\n",
      "contig000238\n",
      "contig000239\n",
      "contig000240\n",
      "contig000241\n",
      "contig000242\n",
      "contig000243\n",
      "contig000244\n",
      "contig000245\n",
      "contig000246\n",
      "contig000247\n",
      "contig000248\n",
      "contig000249\n",
      "contig000250\n",
      "contig000251\n",
      "contig000252\n",
      "contig000253\n",
      "longread:UCT113['longread'] has 253 contigs\n",
      "longread: UCT113['longread'] has 1189 annotated genes\n",
      "No shortread annotations found for UCT113\n",
      "Processing assembly: URI93H\n",
      "contig000001\n",
      "contig000002\n",
      "contig000003\n",
      "contig000004\n",
      "contig000005\n",
      "contig000006\n",
      "contig000007\n",
      "contig000008\n",
      "contig000009\n",
      "contig000010\n",
      "contig000011\n",
      "contig000012\n",
      "contig000013\n",
      "contig000014\n",
      "contig000015\n",
      "contig000016\n",
      "contig000017\n",
      "contig000018\n",
      "contig000019\n",
      "contig000020\n",
      "contig000021\n",
      "contig000022\n",
      "contig000023\n",
      "contig000024\n",
      "contig000025\n",
      "contig000026\n",
      "contig000027\n",
      "contig000028\n",
      "contig000029\n",
      "longread:URI93['longread'] has 29 contigs\n",
      "longread: URI93['longread'] has 1426 annotated genes\n",
      "No shortread annotations found for URI93\n",
      "Processing assembly: UNY169P\n",
      "B348_chr\n",
      "B348_cp26\n",
      "B348_cp32-10\n",
      "B348_cp32-12\n",
      "B348_cp32-4\n",
      "B348_cp32-5\n",
      "B348_cp32-7\n",
      "B348_cp32-9\n",
      "B348_cp9\n",
      "B348_lp17\n",
      "B348_lp25\n",
      "B348_lp28-2\n",
      "B348_lp28-3\n",
      "B348_lp28-4\n",
      "B348_lp36\n",
      "B348_lp38\n",
      "B348_lp54\n",
      "longread:UNY169['longread'] has 17 contigs\n",
      "longread: UNY169['longread'] has 1303 annotated genes\n",
      "No shortread annotations found for UNY169\n",
      "Processing assembly: UWI283P\n",
      "MC149_chr\n",
      "MC149_cp26\n",
      "MC149_cp32-1\n",
      "MC149_cp32-12\n",
      "MC149_cp32-2\n",
      "MC149_cp32-4\n",
      "MC149_cp32-5\n",
      "MC149_cp32-9\n",
      "MC149_lp17\n",
      "MC149_lp25\n",
      "MC149_lp28-2\n",
      "MC149_lp28-3\n",
      "MC149_lp28-4\n",
      "MC149_lp28-5\n",
      "MC149_lp36\n",
      "MC149_lp54\n",
      "longread:UWI283['longread'] has 16 contigs\n",
      "longread: UWI283['longread'] has 1277 annotated genes\n",
      "No shortread annotations found for UWI283\n",
      "Processing assembly: URI102H\n",
      "contig000001\n",
      "contig000002\n",
      "contig000003\n",
      "contig000004\n",
      "contig000005\n",
      "contig000006\n",
      "contig000007\n",
      "contig000008\n",
      "contig000009\n",
      "contig000010\n",
      "contig000011\n",
      "contig000012\n",
      "contig000013\n",
      "contig000014\n",
      "contig000015\n",
      "contig000016\n",
      "contig000017\n",
      "contig000018\n",
      "contig000019\n",
      "contig000020\n",
      "contig000021\n",
      "contig000022\n",
      "contig000023\n",
      "contig000024\n",
      "contig000025\n",
      "contig000026\n",
      "contig000027\n",
      "contig000028\n",
      "contig000029\n",
      "contig000030\n",
      "contig000031\n",
      "contig000032\n",
      "contig000033\n",
      "contig000034\n",
      "contig000035\n",
      "contig000036\n",
      "contig000037\n",
      "contig000038\n",
      "contig000039\n",
      "contig000040\n",
      "contig000041\n",
      "contig000042\n",
      "contig000043\n",
      "contig000044\n",
      "longread:URI102['longread'] has 44 contigs\n",
      "longread: URI102['longread'] has 1392 annotated genes\n",
      "No shortread annotations found for URI102\n",
      "Processing assembly: URI41H\n",
      "contig000001\n",
      "contig000002\n",
      "contig000003\n",
      "contig000004\n",
      "contig000005\n",
      "contig000006\n",
      "contig000007\n",
      "contig000008\n",
      "contig000009\n",
      "contig000010\n",
      "contig000011\n",
      "contig000012\n",
      "contig000013\n",
      "contig000014\n",
      "contig000015\n",
      "contig000016\n",
      "contig000017\n",
      "contig000018\n",
      "contig000019\n",
      "contig000020\n",
      "contig000021\n",
      "contig000022\n",
      "contig000023\n",
      "contig000024\n",
      "contig000025\n",
      "contig000026\n",
      "contig000027\n",
      "contig000028\n",
      "contig000029\n",
      "contig000030\n",
      "contig000031\n",
      "contig000032\n",
      "contig000033\n",
      "contig000034\n",
      "contig000035\n",
      "contig000036\n",
      "contig000037\n",
      "contig000038\n",
      "contig000039\n",
      "contig000040\n",
      "contig000041\n",
      "contig000042\n",
      "contig000043\n",
      "contig000044\n",
      "contig000045\n",
      "contig000046\n",
      "contig000047\n",
      "contig000048\n",
      "contig000049\n",
      "contig000050\n",
      "contig000051\n",
      "contig000052\n",
      "contig000053\n",
      "contig000054\n",
      "contig000055\n",
      "contig000056\n",
      "contig000057\n",
      "contig000058\n",
      "contig000059\n",
      "contig000060\n",
      "contig000061\n",
      "contig000062\n",
      "contig000063\n",
      "contig000064\n",
      "contig000065\n",
      "contig000066\n",
      "contig000067\n",
      "contig000068\n",
      "contig000069\n",
      "contig000070\n",
      "contig000071\n",
      "contig000072\n",
      "contig000073\n",
      "contig000074\n",
      "contig000075\n",
      "contig000076\n",
      "contig000077\n",
      "contig000078\n",
      "contig000079\n",
      "contig000080\n",
      "contig000081\n",
      "contig000082\n",
      "contig000083\n",
      "contig000084\n",
      "longread:URI41['longread'] has 84 contigs\n",
      "longread: URI41['longread'] has 1437 annotated genes\n",
      "No shortread annotations found for URI41\n",
      "Processing assembly: UNY172P\n",
      "B356_chr\n",
      "B356_cp26\n",
      "B356_cp32-12\n",
      "B356_cp32-4\n",
      "B356_cp32-5+1\n",
      "B356_cp32-5\n",
      "B356_cp32-7\n",
      "B356_cp32-9\n",
      "B356_lp17\n",
      "B356_lp25\n",
      "B356_lp28-2\n",
      "B356_lp28-3\n",
      "B356_lp28-4\n",
      "B356_lp36\n",
      "B356_lp38\n",
      "B356_lp54\n",
      "longread:UNY172['longread'] has 16 contigs\n",
      "longread: UNY172['longread'] has 1298 annotated genes\n",
      "No shortread annotations found for UNY172\n",
      "Processing assembly: UNY149P\n",
      "B247_chr\n",
      "B247_cp26\n",
      "B247_cp32-11\n",
      "B247_cp32-3\n",
      "B247_cp32-4\n",
      "B247_cp32-5\n",
      "B247_cp32-6\n",
      "B247_cp32-7\n",
      "B247_cp32-9\n",
      "B247_lp17\n",
      "B247_lp25\n",
      "B247_lp28-1\n",
      "B247_lp28-3\n",
      "B247_lp28-4\n",
      "B247_lp28-5\n",
      "B247_lp28-6\n",
      "B247_lp36\n",
      "B247_lp38\n",
      "B247_lp54\n",
      "longread:UNY149['longread'] has 19 contigs\n",
      "longread: UNY149['longread'] has 1389 annotated genes\n",
      "No shortread annotations found for UNY149\n",
      "Processing assembly: UCT92H\n",
      "contig000001\n",
      "contig000002\n",
      "contig000003\n",
      "contig000004\n",
      "contig000005\n",
      "contig000006\n",
      "contig000007\n",
      "contig000008\n",
      "contig000009\n",
      "contig000010\n",
      "contig000011\n",
      "contig000012\n",
      "contig000013\n",
      "contig000014\n",
      "contig000015\n",
      "contig000016\n",
      "contig000017\n",
      "contig000018\n",
      "contig000019\n",
      "contig000020\n",
      "contig000021\n",
      "contig000022\n",
      "contig000023\n",
      "contig000024\n",
      "contig000025\n",
      "contig000026\n",
      "contig000027\n",
      "contig000028\n",
      "contig000029\n",
      "contig000030\n",
      "contig000031\n",
      "contig000032\n",
      "contig000033\n",
      "contig000034\n",
      "contig000035\n",
      "contig000036\n",
      "contig000037\n",
      "contig000038\n",
      "contig000039\n",
      "contig000040\n",
      "contig000041\n",
      "contig000042\n",
      "contig000043\n",
      "contig000044\n",
      "contig000045\n",
      "contig000046\n",
      "contig000047\n",
      "contig000048\n",
      "contig000049\n",
      "contig000050\n",
      "contig000051\n",
      "contig000052\n",
      "contig000053\n",
      "contig000054\n",
      "contig000055\n",
      "contig000056\n",
      "contig000057\n",
      "contig000058\n",
      "contig000059\n",
      "contig000060\n",
      "contig000061\n",
      "contig000062\n",
      "longread:UCT92['longread'] has 62 contigs\n",
      "longread: UCT92['longread'] has 1512 annotated genes\n",
      "No shortread annotations found for UCT92\n",
      "Processing assembly: URI118H\n",
      "contig000001\n",
      "contig000002\n",
      "contig000003\n",
      "contig000004\n",
      "contig000005\n",
      "contig000006\n",
      "contig000007\n",
      "contig000008\n",
      "contig000009\n",
      "contig000010\n",
      "contig000011\n",
      "contig000012\n",
      "contig000013\n",
      "contig000014\n",
      "contig000015\n",
      "contig000016\n",
      "contig000017\n",
      "contig000018\n",
      "contig000019\n",
      "contig000020\n",
      "contig000021\n",
      "contig000022\n",
      "contig000023\n",
      "contig000024\n",
      "contig000025\n",
      "contig000026\n",
      "contig000027\n",
      "contig000028\n",
      "contig000029\n",
      "contig000030\n",
      "contig000031\n",
      "contig000032\n",
      "contig000033\n",
      "contig000034\n",
      "contig000035\n",
      "contig000036\n",
      "contig000037\n",
      "contig000038\n",
      "contig000039\n",
      "contig000040\n",
      "contig000041\n",
      "contig000042\n",
      "contig000043\n",
      "contig000044\n",
      "contig000045\n",
      "contig000046\n",
      "contig000047\n",
      "contig000048\n",
      "contig000049\n",
      "contig000050\n",
      "contig000051\n",
      "contig000052\n",
      "contig000053\n",
      "contig000054\n",
      "contig000055\n",
      "contig000056\n",
      "contig000057\n",
      "contig000058\n",
      "contig000059\n",
      "contig000060\n",
      "contig000061\n",
      "contig000062\n",
      "contig000063\n",
      "contig000064\n",
      "contig000065\n",
      "contig000066\n",
      "contig000067\n",
      "contig000068\n",
      "contig000069\n",
      "contig000070\n",
      "contig000071\n",
      "contig000072\n",
      "contig000073\n",
      "contig000074\n",
      "contig000075\n",
      "contig000076\n",
      "contig000077\n",
      "contig000078\n",
      "contig000079\n",
      "contig000080\n",
      "contig000081\n",
      "contig000082\n",
      "contig000083\n",
      "contig000084\n",
      "contig000085\n",
      "contig000086\n",
      "contig000087\n",
      "contig000088\n",
      "contig000089\n",
      "contig000090\n",
      "contig000091\n",
      "contig000092\n",
      "contig000093\n",
      "contig000094\n",
      "contig000095\n",
      "contig000096\n",
      "contig000097\n",
      "contig000098\n",
      "contig000099\n",
      "contig000100\n",
      "contig000101\n",
      "contig000102\n",
      "contig000103\n",
      "contig000104\n",
      "contig000105\n",
      "contig000106\n",
      "contig000107\n",
      "contig000108\n",
      "contig000109\n",
      "longread:URI118['longread'] has 109 contigs\n",
      "longread: URI118['longread'] has 1317 annotated genes\n",
      "No shortread annotations found for URI118\n",
      "Processing assembly: UCT50H\n",
      "contig000001\n",
      "contig000002\n",
      "contig000003\n",
      "contig000004\n",
      "contig000005\n",
      "contig000006\n",
      "contig000007\n",
      "contig000008\n",
      "contig000009\n",
      "contig000010\n",
      "contig000011\n",
      "contig000012\n",
      "contig000013\n",
      "contig000014\n",
      "contig000015\n",
      "contig000016\n",
      "contig000017\n",
      "contig000018\n",
      "contig000019\n",
      "contig000020\n",
      "contig000021\n",
      "contig000022\n",
      "contig000023\n",
      "contig000024\n",
      "contig000025\n",
      "contig000026\n",
      "contig000027\n",
      "contig000028\n",
      "contig000029\n",
      "contig000030\n",
      "contig000031\n",
      "contig000032\n",
      "contig000033\n",
      "contig000034\n",
      "contig000035\n",
      "contig000036\n",
      "contig000037\n",
      "contig000038\n",
      "contig000039\n",
      "contig000040\n",
      "contig000041\n",
      "contig000042\n",
      "contig000043\n",
      "contig000044\n",
      "contig000045\n",
      "contig000046\n",
      "contig000047\n",
      "contig000048\n",
      "contig000049\n",
      "contig000050\n",
      "contig000051\n",
      "contig000052\n",
      "contig000053\n",
      "contig000054\n",
      "contig000055\n",
      "contig000056\n",
      "contig000057\n",
      "contig000058\n",
      "contig000059\n",
      "contig000060\n",
      "contig000061\n",
      "contig000062\n",
      "contig000063\n",
      "contig000064\n",
      "contig000065\n",
      "contig000066\n",
      "contig000067\n",
      "contig000068\n",
      "contig000069\n",
      "contig000070\n",
      "contig000071\n",
      "contig000072\n",
      "contig000073\n",
      "contig000074\n",
      "contig000075\n",
      "contig000076\n",
      "contig000077\n",
      "contig000078\n",
      "contig000079\n",
      "contig000080\n",
      "contig000081\n",
      "contig000082\n",
      "contig000083\n",
      "contig000084\n",
      "contig000085\n",
      "contig000086\n",
      "contig000087\n",
      "contig000088\n",
      "contig000089\n",
      "contig000090\n",
      "longread:UCT50['longread'] has 90 contigs\n",
      "longread: UCT50['longread'] has 1284 annotated genes\n",
      "No shortread annotations found for UCT50\n",
      "Processing assembly: URI101H\n",
      "contig000001\n",
      "contig000002\n",
      "contig000003\n",
      "contig000004\n",
      "contig000005\n",
      "contig000006\n",
      "contig000007\n",
      "contig000008\n",
      "contig000009\n",
      "contig000010\n",
      "contig000011\n",
      "contig000012\n",
      "contig000013\n",
      "contig000014\n",
      "contig000015\n",
      "contig000016\n",
      "contig000017\n",
      "contig000018\n",
      "contig000019\n",
      "contig000020\n",
      "contig000021\n",
      "contig000022\n",
      "contig000023\n",
      "contig000024\n",
      "contig000025\n",
      "contig000026\n",
      "contig000027\n",
      "contig000028\n",
      "contig000029\n",
      "contig000030\n",
      "contig000031\n",
      "contig000032\n",
      "longread:URI101['longread'] has 32 contigs\n",
      "longread: URI101['longread'] has 1513 annotated genes\n",
      "No shortread annotations found for URI101\n",
      "Processing assembly: URI46H\n",
      "contig000001\n",
      "contig000002\n",
      "contig000003\n",
      "contig000004\n",
      "contig000005\n",
      "contig000006\n",
      "contig000007\n",
      "contig000008\n",
      "contig000009\n",
      "contig000010\n",
      "contig000011\n",
      "contig000012\n",
      "contig000013\n",
      "contig000014\n",
      "contig000015\n",
      "contig000016\n",
      "contig000017\n",
      "contig000018\n",
      "contig000019\n",
      "contig000020\n",
      "contig000021\n",
      "contig000022\n",
      "contig000023\n",
      "contig000024\n",
      "contig000025\n",
      "contig000026\n",
      "contig000027\n",
      "contig000028\n",
      "contig000029\n",
      "contig000030\n",
      "contig000031\n",
      "contig000032\n",
      "contig000033\n",
      "contig000034\n",
      "contig000035\n",
      "contig000036\n",
      "contig000037\n",
      "contig000038\n",
      "contig000039\n",
      "contig000040\n",
      "contig000041\n",
      "contig000042\n",
      "contig000043\n",
      "contig000044\n",
      "contig000045\n",
      "contig000046\n",
      "contig000047\n",
      "contig000048\n",
      "contig000049\n",
      "contig000050\n",
      "contig000051\n",
      "contig000052\n",
      "contig000053\n",
      "contig000054\n",
      "contig000055\n",
      "contig000056\n",
      "contig000057\n",
      "contig000058\n",
      "contig000059\n",
      "contig000060\n",
      "contig000061\n",
      "contig000062\n",
      "contig000063\n",
      "contig000064\n",
      "contig000065\n",
      "contig000066\n",
      "contig000067\n",
      "contig000068\n",
      "contig000069\n",
      "contig000070\n",
      "contig000071\n",
      "contig000072\n",
      "contig000073\n",
      "contig000074\n",
      "contig000075\n",
      "contig000076\n",
      "contig000077\n",
      "contig000078\n",
      "contig000079\n",
      "contig000080\n",
      "contig000081\n",
      "contig000082\n",
      "contig000083\n",
      "contig000084\n",
      "contig000085\n",
      "contig000086\n",
      "contig000087\n",
      "contig000088\n",
      "contig000089\n",
      "contig000090\n",
      "contig000091\n",
      "contig000092\n",
      "contig000093\n",
      "contig000094\n",
      "contig000095\n",
      "longread:URI46['longread'] has 95 contigs\n",
      "longread: URI46['longread'] has 1363 annotated genes\n",
      "No shortread annotations found for URI46\n",
      "Processing assembly: URI48H\n",
      "contig000001\n",
      "contig000002\n",
      "contig000003\n",
      "contig000004\n",
      "contig000005\n",
      "contig000006\n",
      "contig000007\n",
      "contig000008\n",
      "contig000009\n",
      "contig000010\n",
      "contig000011\n",
      "contig000012\n",
      "contig000013\n",
      "contig000014\n",
      "contig000015\n",
      "contig000016\n",
      "contig000017\n",
      "contig000018\n",
      "contig000019\n",
      "contig000020\n",
      "contig000021\n",
      "contig000022\n",
      "longread:URI48['longread'] has 22 contigs\n",
      "longread: URI48['longread'] has 1323 annotated genes\n",
      "No shortread annotations found for URI48\n",
      "Processing assembly: URI111H\n",
      "contig000001\n",
      "contig000002\n",
      "contig000003\n",
      "contig000004\n",
      "contig000005\n",
      "contig000006\n",
      "contig000007\n",
      "contig000008\n",
      "contig000009\n",
      "contig000010\n",
      "contig000011\n",
      "contig000012\n",
      "contig000013\n",
      "contig000014\n",
      "contig000015\n",
      "contig000016\n",
      "contig000017\n",
      "contig000018\n",
      "contig000019\n",
      "contig000020\n",
      "contig000021\n",
      "contig000022\n",
      "contig000023\n",
      "longread:URI111['longread'] has 23 contigs\n",
      "longread: URI111['longread'] has 1513 annotated genes\n",
      "No shortread annotations found for URI111\n",
      "dictionary construction complete! pickling!\n",
      "dictionary pickled to: /home/mf019/borrelia_plasmid_classifier_v3/output/assembly_dict_v3.pkl\n"
     ]
    }
   ],
   "source": [
    "\n",
    "if not os.path.exists(f'{output_dir}/assembly_dict.pkl'):\n",
    "    assembly_dict = defaultdict(dict)\n",
    "    for assembly in lr_gbs: # iterate through each longread assembly\n",
    "        assembly_name = Path(assembly).stem # get the name of the assembly\n",
    "        print(f'Processing assembly: {assembly_name}')\n",
    "        if assembly_name.endswith(('H','P')):\n",
    "            assembly_name = assembly_name[:-1] # chop last character off for dict key\n",
    "        assembly_dict[assembly_name] = defaultdict(dict)\n",
    "        assembly_dict[assembly_name]['longread'] = defaultdict(dict)\n",
    "        with open(assembly, 'r') as f: # get longread contigs and annotations\n",
    "            num_contigs = 0\n",
    "            num_genes = 0\n",
    "            for contig in SeqIO.parse(handle=f, format='genbank'):\n",
    "                num_contigs += 1\n",
    "                contig_id = contig.name\n",
    "                print(contig_id)\n",
    "                assembly_dict[assembly_name]['longread'][contig_id]['seq'] = contig.seq\n",
    "                assembly_dict[assembly_name]['longread'][contig_id]['annotations'] = contig.annotations\n",
    "                assembly_dict[assembly_name]['longread'][contig_id]['features'] = contig.features\n",
    "                if len(contig.features) > 0:\n",
    "                    for feature in contig.features:\n",
    "                        if feature.type == 'CDS':\n",
    "                            num_genes += 1\n",
    "            print(f'longread:{assembly_name}[\\'longread\\'] has {num_contigs} contigs')\n",
    "            print(f'longread: {assembly_name}[\\'longread\\'] has {num_genes} annotated genes')\n",
    "\n",
    "        # now we need to get the shortread contigs and annotations if they exist.\n",
    "        shortread_file = f'{shortread_annotations}/{assembly_name}.gbff'\n",
    "        if shortread_file in sr_gbs:\n",
    "            print(f'Found shortread annotations for {assembly_name}')\n",
    "            assembly_dict[assembly_name]['shortread'] = defaultdict(dict)\n",
    "            with open(shortread_file, 'r') as fi:\n",
    "                num_genes = 0\n",
    "                num_contigs = 0\n",
    "                for scontig in SeqIO.parse(handle=fi, format='genbank'):\n",
    "                    num_contigs += 1\n",
    "                    scontig_id = scontig.id\n",
    "                    #print(scontig_id)\n",
    "                    assembly_dict[assembly_name]['shortread'][scontig_id]['seq'] = scontig.seq\n",
    "                    assembly_dict[assembly_name]['shortread'][scontig_id]['annotations'] = scontig.annotations\n",
    "                    assembly_dict[assembly_name]['shortread'][scontig_id]['features'] = scontig.features\n",
    "                    if len(scontig.features) > 0:\n",
    "                        for feature in scontig.features:\n",
    "                            if feature.type == 'CDS':\n",
    "                                num_genes += 1\n",
    "                    # now we need to blast this contig against the pf32 and whole plasmid databases\n",
    "                    # we will need to save these results in a dictionary\n",
    "            print(f'shortread:{assembly_name} has {num_contigs} contigs')\n",
    "            print(f'shortread: {assembly_name} has {num_genes} annotated genes')\n",
    "        else:\n",
    "            assembly_dict[assembly_name]['shortread'] = 'NOT AVAILABLE'\n",
    "            print(f'No shortread annotations found for {assembly_name}')\n",
    "    print('dictionary construction complete! pickling!')\n",
    "    pickle.dump(assembly_dict, open(f'{output_dir}/assembly_dict_v3.pkl', 'wb'))\n",
    "    print(f'dictionary pickled to: {output_dir}/assembly_dict_v3.pkl')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "No shortread annotations found for URI87\n",
      "setting up blast commands for URI87\n",
      "/home/mf019/longread_GWAS/longread_analysis/paired_assemblies/paired_only/longread/contigs\n",
      "Running blast command:\n",
      "blastn -query /tmp/URI87.fasta -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/whole_plasmid/URI87_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 5\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/whole_plasmid/URI87_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "Running blast command:\n",
      "blastx -query /tmp/URI87.fasta -task \"blastx\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/pf32_db/pf32  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/pf32/URI87_pf32.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 5\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/pf32/URI87_pf32.xml\n",
      "No shortread annotations found for URI34\n",
      "setting up blast commands for URI34\n",
      "/home/mf019/longread_GWAS/longread_analysis/paired_assemblies/paired_only/longread/contigs\n",
      "Running blast command:\n",
      "blastn -query /tmp/URI34.fasta -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/whole_plasmid/URI34_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 5\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/whole_plasmid/URI34_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "Running blast command:\n",
      "blastx -query /tmp/URI34.fasta -task \"blastx\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/pf32_db/pf32  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/pf32/URI34_pf32.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 5\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/pf32/URI34_pf32.xml\n",
      "No shortread annotations found for URI88\n",
      "setting up blast commands for URI88\n",
      "/home/mf019/longread_GWAS/longread_analysis/paired_assemblies/paired_only/longread/contigs\n",
      "Running blast command:\n",
      "blastn -query /tmp/URI88.fasta -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/whole_plasmid/URI88_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 5\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/whole_plasmid/URI88_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "Running blast command:\n",
      "blastx -query /tmp/URI88.fasta -task \"blastx\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/pf32_db/pf32  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/pf32/URI88_pf32.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 5\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/pf32/URI88_pf32.xml\n",
      "No shortread annotations found for URI33\n",
      "setting up blast commands for URI33\n",
      "/home/mf019/longread_GWAS/longread_analysis/paired_assemblies/paired_only/longread/contigs\n",
      "Running blast command:\n",
      "blastn -query /tmp/URI33.fasta -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/whole_plasmid/URI33_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 5\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/whole_plasmid/URI33_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "Running blast command:\n",
      "blastx -query /tmp/URI33.fasta -task \"blastx\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/pf32_db/pf32  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/pf32/URI33_pf32.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 5\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/pf32/URI33_pf32.xml\n",
      "No shortread annotations found for UCT110\n",
      "setting up blast commands for UCT110\n",
      "/home/mf019/longread_GWAS/longread_analysis/paired_assemblies/paired_only/longread/contigs\n",
      "Running blast command:\n",
      "blastn -query /tmp/UCT110.fasta -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/whole_plasmid/UCT110_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 5\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/whole_plasmid/UCT110_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "Running blast command:\n",
      "blastx -query /tmp/UCT110.fasta -task \"blastx\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/pf32_db/pf32  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/pf32/UCT110_pf32.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 5\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/pf32/UCT110_pf32.xml\n",
      "No shortread annotations found for URI39\n",
      "setting up blast commands for URI39\n",
      "/home/mf019/longread_GWAS/longread_analysis/paired_assemblies/paired_only/longread/contigs\n",
      "Running blast command:\n",
      "blastn -query /tmp/URI39.fasta -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/whole_plasmid/URI39_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 5\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/whole_plasmid/URI39_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "Running blast command:\n",
      "blastx -query /tmp/URI39.fasta -task \"blastx\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/pf32_db/pf32  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/pf32/URI39_pf32.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 5\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/pf32/URI39_pf32.xml\n",
      "No shortread annotations found for URI91\n",
      "setting up blast commands for URI91\n",
      "/home/mf019/longread_GWAS/longread_analysis/paired_assemblies/paired_only/longread/contigs\n",
      "Running blast command:\n",
      "blastn -query /tmp/URI91.fasta -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/whole_plasmid/URI91_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 5\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/whole_plasmid/URI91_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "Running blast command:\n",
      "blastx -query /tmp/URI91.fasta -task \"blastx\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/pf32_db/pf32  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/pf32/URI91_pf32.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 5\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/pf32/URI91_pf32.xml\n",
      "No shortread annotations found for UCT35\n",
      "setting up blast commands for UCT35\n",
      "/home/mf019/longread_GWAS/longread_analysis/paired_assemblies/paired_only/longread/contigs\n",
      "Running blast command:\n",
      "blastn -query /tmp/UCT35.fasta -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/whole_plasmid/UCT35_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 5\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/whole_plasmid/UCT35_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "Running blast command:\n",
      "blastx -query /tmp/UCT35.fasta -task \"blastx\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/pf32_db/pf32  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/pf32/UCT35_pf32.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 5\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/pf32/UCT35_pf32.xml\n",
      "No shortread annotations found for UWI247\n",
      "setting up blast commands for UWI247\n",
      "/home/mf019/longread_GWAS/longread_analysis/paired_assemblies/paired_only/longread/contigs\n",
      "Running blast command:\n",
      "blastn -query /tmp/UWI247.fasta -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/whole_plasmid/UWI247_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 5\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/whole_plasmid/UWI247_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "Running blast command:\n",
      "blastx -query /tmp/UWI247.fasta -task \"blastx\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/pf32_db/pf32  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/pf32/UWI247_pf32.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 5\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/pf32/UWI247_pf32.xml\n",
      "No shortread annotations found for URI120\n",
      "setting up blast commands for URI120\n",
      "/home/mf019/longread_GWAS/longread_analysis/paired_assemblies/paired_only/longread/contigs\n",
      "Running blast command:\n",
      "blastn -query /tmp/URI120.fasta -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/whole_plasmid/URI120_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 5\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/whole_plasmid/URI120_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "Running blast command:\n",
      "blastx -query /tmp/URI120.fasta -task \"blastx\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/pf32_db/pf32  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/pf32/URI120_pf32.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 5\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/pf32/URI120_pf32.xml\n",
      "No shortread annotations found for URI107\n",
      "setting up blast commands for URI107\n",
      "/home/mf019/longread_GWAS/longread_analysis/paired_assemblies/paired_only/longread/contigs\n",
      "Running blast command:\n",
      "blastn -query /tmp/URI107.fasta -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/whole_plasmid/URI107_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 5\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/whole_plasmid/URI107_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "Running blast command:\n",
      "blastx -query /tmp/URI107.fasta -task \"blastx\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/pf32_db/pf32  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/pf32/URI107_pf32.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 5\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/pf32/URI107_pf32.xml\n",
      "No shortread annotations found for UWI263\n",
      "setting up blast commands for UWI263\n",
      "/home/mf019/longread_GWAS/longread_analysis/paired_assemblies/paired_only/longread/contigs\n",
      "Running blast command:\n",
      "blastn -query /tmp/UWI263.fasta -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/whole_plasmid/UWI263_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 5\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/whole_plasmid/UWI263_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "Running blast command:\n",
      "blastx -query /tmp/UWI263.fasta -task \"blastx\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/pf32_db/pf32  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/pf32/UWI263_pf32.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 5\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/pf32/UWI263_pf32.xml\n",
      "No shortread annotations found for URI89\n",
      "setting up blast commands for URI89\n",
      "/home/mf019/longread_GWAS/longread_analysis/paired_assemblies/paired_only/longread/contigs\n",
      "Running blast command:\n",
      "blastn -query /tmp/URI89.fasta -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/whole_plasmid/URI89_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 5\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/whole_plasmid/URI89_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "Running blast command:\n",
      "blastx -query /tmp/URI89.fasta -task \"blastx\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/pf32_db/pf32  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/pf32/URI89_pf32.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 5\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/pf32/URI89_pf32.xml\n",
      "No shortread annotations found for URI42\n",
      "setting up blast commands for URI42\n",
      "/home/mf019/longread_GWAS/longread_analysis/paired_assemblies/paired_only/longread/contigs\n",
      "Running blast command:\n",
      "blastn -query /tmp/URI42.fasta -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/whole_plasmid/URI42_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 5\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/whole_plasmid/URI42_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "Running blast command:\n",
      "blastx -query /tmp/URI42.fasta -task \"blastx\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/pf32_db/pf32  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/pf32/URI42_pf32.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 5\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/pf32/URI42_pf32.xml\n",
      "No shortread annotations found for URI44\n",
      "setting up blast commands for URI44\n",
      "/home/mf019/longread_GWAS/longread_analysis/paired_assemblies/paired_only/longread/contigs\n",
      "Running blast command:\n",
      "blastn -query /tmp/URI44.fasta -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/whole_plasmid/URI44_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 5\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/whole_plasmid/URI44_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "Running blast command:\n",
      "blastx -query /tmp/URI44.fasta -task \"blastx\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/pf32_db/pf32  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/pf32/URI44_pf32.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 5\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/pf32/URI44_pf32.xml\n",
      "No shortread annotations found for UCT109\n",
      "setting up blast commands for UCT109\n",
      "/home/mf019/longread_GWAS/longread_analysis/paired_assemblies/paired_only/longread/contigs\n",
      "Running blast command:\n",
      "blastn -query /tmp/UCT109.fasta -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/whole_plasmid/UCT109_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 5\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/whole_plasmid/UCT109_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "Running blast command:\n",
      "blastx -query /tmp/UCT109.fasta -task \"blastx\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/pf32_db/pf32  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/pf32/UCT109_pf32.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 5\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/pf32/UCT109_pf32.xml\n",
      "No shortread annotations found for URI40\n",
      "setting up blast commands for URI40\n",
      "/home/mf019/longread_GWAS/longread_analysis/paired_assemblies/paired_only/longread/contigs\n",
      "Running blast command:\n",
      "blastn -query /tmp/URI40.fasta -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/whole_plasmid/URI40_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 5\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/whole_plasmid/URI40_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "Running blast command:\n",
      "blastx -query /tmp/URI40.fasta -task \"blastx\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/pf32_db/pf32  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/pf32/URI40_pf32.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 5\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/pf32/URI40_pf32.xml\n",
      "No shortread annotations found for URI117\n",
      "setting up blast commands for URI117\n",
      "/home/mf019/longread_GWAS/longread_analysis/paired_assemblies/paired_only/longread/contigs\n",
      "Running blast command:\n",
      "blastn -query /tmp/URI117.fasta -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/whole_plasmid/URI117_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 5\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/whole_plasmid/URI117_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "Running blast command:\n",
      "blastx -query /tmp/URI117.fasta -task \"blastx\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/pf32_db/pf32  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/pf32/URI117_pf32.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 5\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/pf32/URI117_pf32.xml\n",
      "No shortread annotations found for URI47\n",
      "setting up blast commands for URI47\n",
      "/home/mf019/longread_GWAS/longread_analysis/paired_assemblies/paired_only/longread/contigs\n",
      "Running blast command:\n",
      "blastn -query /tmp/URI47.fasta -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/whole_plasmid/URI47_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 5\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/whole_plasmid/URI47_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "Running blast command:\n",
      "blastx -query /tmp/URI47.fasta -task \"blastx\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/pf32_db/pf32  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/pf32/URI47_pf32.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 5\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/pf32/URI47_pf32.xml\n",
      "No shortread annotations found for URI86\n",
      "setting up blast commands for URI86\n",
      "/home/mf019/longread_GWAS/longread_analysis/paired_assemblies/paired_only/longread/contigs\n",
      "Running blast command:\n",
      "blastn -query /tmp/URI86.fasta -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/whole_plasmid/URI86_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 5\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/whole_plasmid/URI86_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "Running blast command:\n",
      "blastx -query /tmp/URI86.fasta -task \"blastx\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/pf32_db/pf32  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/pf32/URI86_pf32.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 5\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/pf32/URI86_pf32.xml\n",
      "No shortread annotations found for URI36\n",
      "setting up blast commands for URI36\n",
      "/home/mf019/longread_GWAS/longread_analysis/paired_assemblies/paired_only/longread/contigs\n",
      "Running blast command:\n",
      "blastn -query /tmp/URI36.fasta -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/whole_plasmid/URI36_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 5\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/whole_plasmid/URI36_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "Running blast command:\n",
      "blastx -query /tmp/URI36.fasta -task \"blastx\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/pf32_db/pf32  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/pf32/URI36_pf32.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 5\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/pf32/URI36_pf32.xml\n",
      "No shortread annotations found for UNY208\n",
      "setting up blast commands for UNY208\n",
      "/home/mf019/longread_GWAS/longread_analysis/paired_assemblies/paired_only/longread/contigs\n",
      "Running blast command:\n",
      "blastn -query /tmp/UNY208.fasta -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/whole_plasmid/UNY208_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 5\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/whole_plasmid/UNY208_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "Running blast command:\n",
      "blastx -query /tmp/UNY208.fasta -task \"blastx\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/pf32_db/pf32  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/pf32/UNY208_pf32.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 5\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/pf32/UNY208_pf32.xml\n",
      "No shortread annotations found for ESI26\n",
      "setting up blast commands for ESI26\n",
      "/home/mf019/longread_GWAS/longread_analysis/paired_assemblies/paired_only/longread/contigs\n",
      "Running blast command:\n",
      "blastn -query /tmp/ESI26.fasta -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/whole_plasmid/ESI26_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 5\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/whole_plasmid/ESI26_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "Running blast command:\n",
      "blastx -query /tmp/ESI26.fasta -task \"blastx\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/pf32_db/pf32  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/pf32/ESI26_pf32.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 5\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/pf32/ESI26_pf32.xml\n",
      "No shortread annotations found for UCT31\n",
      "setting up blast commands for UCT31\n",
      "/home/mf019/longread_GWAS/longread_analysis/paired_assemblies/paired_only/longread/contigs\n",
      "Running blast command:\n",
      "blastn -query /tmp/UCT31.fasta -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/whole_plasmid/UCT31_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 5\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/whole_plasmid/UCT31_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "Running blast command:\n",
      "blastx -query /tmp/UCT31.fasta -task \"blastx\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/pf32_db/pf32  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/pf32/UCT31_pf32.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 5\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/pf32/UCT31_pf32.xml\n",
      "No shortread annotations found for URI56\n",
      "setting up blast commands for URI56\n",
      "/home/mf019/longread_GWAS/longread_analysis/paired_assemblies/paired_only/longread/contigs\n",
      "Running blast command:\n",
      "blastn -query /tmp/URI56.fasta -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/whole_plasmid/URI56_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 5\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/whole_plasmid/URI56_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "Running blast command:\n",
      "blastx -query /tmp/URI56.fasta -task \"blastx\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/pf32_db/pf32  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/pf32/URI56_pf32.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 5\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/pf32/URI56_pf32.xml\n",
      "No shortread annotations found for UCT30\n",
      "setting up blast commands for UCT30\n",
      "/home/mf019/longread_GWAS/longread_analysis/paired_assemblies/paired_only/longread/contigs\n",
      "Running blast command:\n",
      "blastn -query /tmp/UCT30.fasta -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/whole_plasmid/UCT30_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 5\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/whole_plasmid/UCT30_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "Running blast command:\n",
      "blastx -query /tmp/UCT30.fasta -task \"blastx\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/pf32_db/pf32  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/pf32/UCT30_pf32.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 5\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/pf32/UCT30_pf32.xml\n",
      "No shortread annotations found for URI103\n",
      "setting up blast commands for URI103\n",
      "/home/mf019/longread_GWAS/longread_analysis/paired_assemblies/paired_only/longread/contigs\n",
      "Running blast command:\n",
      "blastn -query /tmp/URI103.fasta -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/whole_plasmid/URI103_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 5\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/whole_plasmid/URI103_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "Running blast command:\n",
      "blastx -query /tmp/URI103.fasta -task \"blastx\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/pf32_db/pf32  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/pf32/URI103_pf32.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 5\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/pf32/URI103_pf32.xml\n",
      "No shortread annotations found for UCT29\n",
      "setting up blast commands for UCT29\n",
      "/home/mf019/longread_GWAS/longread_analysis/paired_assemblies/paired_only/longread/contigs\n",
      "Running blast command:\n",
      "blastn -query /tmp/UCT29.fasta -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/whole_plasmid/UCT29_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 5\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/whole_plasmid/UCT29_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "Running blast command:\n",
      "blastx -query /tmp/UCT29.fasta -task \"blastx\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/pf32_db/pf32  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/pf32/UCT29_pf32.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 5\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/pf32/UCT29_pf32.xml\n",
      "No shortread annotations found for URI112\n",
      "setting up blast commands for URI112\n",
      "/home/mf019/longread_GWAS/longread_analysis/paired_assemblies/paired_only/longread/contigs\n",
      "Running blast command:\n",
      "blastn -query /tmp/URI112.fasta -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/whole_plasmid/URI112_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 5\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/whole_plasmid/URI112_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "Running blast command:\n",
      "blastx -query /tmp/URI112.fasta -task \"blastx\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/pf32_db/pf32  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/pf32/URI112_pf32.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 5\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/pf32/URI112_pf32.xml\n",
      "No shortread annotations found for UWI248\n",
      "setting up blast commands for UWI248\n",
      "/home/mf019/longread_GWAS/longread_analysis/paired_assemblies/paired_only/longread/contigs\n",
      "Running blast command:\n",
      "blastn -query /tmp/UWI248.fasta -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/whole_plasmid/UWI248_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 5\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/whole_plasmid/UWI248_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "Running blast command:\n",
      "blastx -query /tmp/UWI248.fasta -task \"blastx\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/pf32_db/pf32  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/pf32/UWI248_pf32.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 5\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/pf32/UWI248_pf32.xml\n",
      "No shortread annotations found for UNY203\n",
      "setting up blast commands for UNY203\n",
      "/home/mf019/longread_GWAS/longread_analysis/paired_assemblies/paired_only/longread/contigs\n",
      "Running blast command:\n",
      "blastn -query /tmp/UNY203.fasta -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/whole_plasmid/UNY203_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 5\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/whole_plasmid/UNY203_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "Running blast command:\n",
      "blastx -query /tmp/UNY203.fasta -task \"blastx\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/pf32_db/pf32  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/pf32/UNY203_pf32.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 5\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/pf32/UNY203_pf32.xml\n",
      "No shortread annotations found for UCT96\n",
      "setting up blast commands for UCT96\n",
      "/home/mf019/longread_GWAS/longread_analysis/paired_assemblies/paired_only/longread/contigs\n",
      "Running blast command:\n",
      "blastn -query /tmp/UCT96.fasta -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/whole_plasmid/UCT96_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 5\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/whole_plasmid/UCT96_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "Running blast command:\n",
      "blastx -query /tmp/UCT96.fasta -task \"blastx\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/pf32_db/pf32  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/pf32/UCT96_pf32.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 5\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/pf32/UCT96_pf32.xml\n",
      "No shortread annotations found for UCT32\n",
      "setting up blast commands for UCT32\n",
      "/home/mf019/longread_GWAS/longread_analysis/paired_assemblies/paired_only/longread/contigs\n",
      "Running blast command:\n",
      "blastn -query /tmp/UCT32.fasta -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/whole_plasmid/UCT32_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 5\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/whole_plasmid/UCT32_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "Running blast command:\n",
      "blastx -query /tmp/UCT32.fasta -task \"blastx\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/pf32_db/pf32  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/pf32/UCT32_pf32.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 5\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/pf32/UCT32_pf32.xml\n",
      "No shortread annotations found for UNY193\n",
      "setting up blast commands for UNY193\n",
      "/home/mf019/longread_GWAS/longread_analysis/paired_assemblies/paired_only/longread/contigs\n",
      "Running blast command:\n",
      "blastn -query /tmp/UNY193.fasta -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/whole_plasmid/UNY193_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 5\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/whole_plasmid/UNY193_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "Running blast command:\n",
      "blastx -query /tmp/UNY193.fasta -task \"blastx\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/pf32_db/pf32  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/pf32/UNY193_pf32.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 5\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/pf32/UNY193_pf32.xml\n",
      "No shortread annotations found for UCT113\n",
      "setting up blast commands for UCT113\n",
      "/home/mf019/longread_GWAS/longread_analysis/paired_assemblies/paired_only/longread/contigs\n",
      "Running blast command:\n",
      "blastn -query /tmp/UCT113.fasta -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/whole_plasmid/UCT113_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 5\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/whole_plasmid/UCT113_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "Running blast command:\n",
      "blastx -query /tmp/UCT113.fasta -task \"blastx\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/pf32_db/pf32  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/pf32/UCT113_pf32.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 5\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/pf32/UCT113_pf32.xml\n",
      "No shortread annotations found for URI93\n",
      "setting up blast commands for URI93\n",
      "/home/mf019/longread_GWAS/longread_analysis/paired_assemblies/paired_only/longread/contigs\n",
      "Running blast command:\n",
      "blastn -query /tmp/URI93.fasta -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/whole_plasmid/URI93_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 5\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/whole_plasmid/URI93_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "Running blast command:\n",
      "blastx -query /tmp/URI93.fasta -task \"blastx\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/pf32_db/pf32  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/pf32/URI93_pf32.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 5\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/pf32/URI93_pf32.xml\n",
      "No shortread annotations found for UNY169\n",
      "setting up blast commands for UNY169\n",
      "/home/mf019/longread_GWAS/longread_analysis/paired_assemblies/paired_only/longread/contigs\n",
      "Running blast command:\n",
      "blastn -query /tmp/UNY169.fasta -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/whole_plasmid/UNY169_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 5\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/whole_plasmid/UNY169_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "Running blast command:\n",
      "blastx -query /tmp/UNY169.fasta -task \"blastx\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/pf32_db/pf32  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/pf32/UNY169_pf32.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 5\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/pf32/UNY169_pf32.xml\n",
      "No shortread annotations found for UWI283\n",
      "setting up blast commands for UWI283\n",
      "/home/mf019/longread_GWAS/longread_analysis/paired_assemblies/paired_only/longread/contigs\n",
      "Running blast command:\n",
      "blastn -query /tmp/UWI283.fasta -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/whole_plasmid/UWI283_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 5\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/whole_plasmid/UWI283_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "Running blast command:\n",
      "blastx -query /tmp/UWI283.fasta -task \"blastx\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/pf32_db/pf32  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/pf32/UWI283_pf32.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 5\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/pf32/UWI283_pf32.xml\n",
      "No shortread annotations found for URI102\n",
      "setting up blast commands for URI102\n",
      "/home/mf019/longread_GWAS/longread_analysis/paired_assemblies/paired_only/longread/contigs\n",
      "Running blast command:\n",
      "blastn -query /tmp/URI102.fasta -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/whole_plasmid/URI102_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 5\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/whole_plasmid/URI102_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "Running blast command:\n",
      "blastx -query /tmp/URI102.fasta -task \"blastx\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/pf32_db/pf32  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/pf32/URI102_pf32.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 5\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/pf32/URI102_pf32.xml\n",
      "No shortread annotations found for URI41\n",
      "setting up blast commands for URI41\n",
      "/home/mf019/longread_GWAS/longread_analysis/paired_assemblies/paired_only/longread/contigs\n",
      "Running blast command:\n",
      "blastn -query /tmp/URI41.fasta -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/whole_plasmid/URI41_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 5\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/whole_plasmid/URI41_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "Running blast command:\n",
      "blastx -query /tmp/URI41.fasta -task \"blastx\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/pf32_db/pf32  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/pf32/URI41_pf32.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 5\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/pf32/URI41_pf32.xml\n",
      "No shortread annotations found for UNY172\n",
      "setting up blast commands for UNY172\n",
      "/home/mf019/longread_GWAS/longread_analysis/paired_assemblies/paired_only/longread/contigs\n",
      "Running blast command:\n",
      "blastn -query /tmp/UNY172.fasta -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/whole_plasmid/UNY172_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 5\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/whole_plasmid/UNY172_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "Running blast command:\n",
      "blastx -query /tmp/UNY172.fasta -task \"blastx\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/pf32_db/pf32  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/pf32/UNY172_pf32.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 5\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/pf32/UNY172_pf32.xml\n",
      "No shortread annotations found for UNY149\n",
      "setting up blast commands for UNY149\n",
      "/home/mf019/longread_GWAS/longread_analysis/paired_assemblies/paired_only/longread/contigs\n",
      "Running blast command:\n",
      "blastn -query /tmp/UNY149.fasta -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/whole_plasmid/UNY149_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 5\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/whole_plasmid/UNY149_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "Running blast command:\n",
      "blastx -query /tmp/UNY149.fasta -task \"blastx\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/pf32_db/pf32  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/pf32/UNY149_pf32.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 5\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/pf32/UNY149_pf32.xml\n",
      "No shortread annotations found for UCT92\n",
      "setting up blast commands for UCT92\n",
      "/home/mf019/longread_GWAS/longread_analysis/paired_assemblies/paired_only/longread/contigs\n",
      "Running blast command:\n",
      "blastn -query /tmp/UCT92.fasta -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/whole_plasmid/UCT92_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 5\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/whole_plasmid/UCT92_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "Running blast command:\n",
      "blastx -query /tmp/UCT92.fasta -task \"blastx\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/pf32_db/pf32  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/pf32/UCT92_pf32.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 5\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/pf32/UCT92_pf32.xml\n",
      "No shortread annotations found for URI118\n",
      "setting up blast commands for URI118\n",
      "/home/mf019/longread_GWAS/longread_analysis/paired_assemblies/paired_only/longread/contigs\n",
      "Running blast command:\n",
      "blastn -query /tmp/URI118.fasta -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/whole_plasmid/URI118_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 5\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/whole_plasmid/URI118_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "Running blast command:\n",
      "blastx -query /tmp/URI118.fasta -task \"blastx\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/pf32_db/pf32  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/pf32/URI118_pf32.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 5\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/pf32/URI118_pf32.xml\n",
      "No shortread annotations found for UCT50\n",
      "setting up blast commands for UCT50\n",
      "/home/mf019/longread_GWAS/longread_analysis/paired_assemblies/paired_only/longread/contigs\n",
      "Running blast command:\n",
      "blastn -query /tmp/UCT50.fasta -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/whole_plasmid/UCT50_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 5\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/whole_plasmid/UCT50_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "Running blast command:\n",
      "blastx -query /tmp/UCT50.fasta -task \"blastx\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/pf32_db/pf32  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/pf32/UCT50_pf32.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 5\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/pf32/UCT50_pf32.xml\n",
      "No shortread annotations found for URI101\n",
      "setting up blast commands for URI101\n",
      "/home/mf019/longread_GWAS/longread_analysis/paired_assemblies/paired_only/longread/contigs\n",
      "Running blast command:\n",
      "blastn -query /tmp/URI101.fasta -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/whole_plasmid/URI101_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 5\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/whole_plasmid/URI101_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "Running blast command:\n",
      "blastx -query /tmp/URI101.fasta -task \"blastx\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/pf32_db/pf32  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/pf32/URI101_pf32.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 5\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/pf32/URI101_pf32.xml\n",
      "No shortread annotations found for URI46\n",
      "setting up blast commands for URI46\n",
      "/home/mf019/longread_GWAS/longread_analysis/paired_assemblies/paired_only/longread/contigs\n",
      "Running blast command:\n",
      "blastn -query /tmp/URI46.fasta -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/whole_plasmid/URI46_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 5\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/whole_plasmid/URI46_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "Running blast command:\n",
      "blastx -query /tmp/URI46.fasta -task \"blastx\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/pf32_db/pf32  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/pf32/URI46_pf32.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 5\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/pf32/URI46_pf32.xml\n",
      "No shortread annotations found for URI48\n",
      "setting up blast commands for URI48\n",
      "/home/mf019/longread_GWAS/longread_analysis/paired_assemblies/paired_only/longread/contigs\n",
      "Running blast command:\n",
      "blastn -query /tmp/URI48.fasta -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/whole_plasmid/URI48_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 5\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/whole_plasmid/URI48_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "Running blast command:\n",
      "blastx -query /tmp/URI48.fasta -task \"blastx\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/pf32_db/pf32  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/pf32/URI48_pf32.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 5\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/pf32/URI48_pf32.xml\n",
      "No shortread annotations found for URI111\n",
      "setting up blast commands for URI111\n",
      "/home/mf019/longread_GWAS/longread_analysis/paired_assemblies/paired_only/longread/contigs\n",
      "Running blast command:\n",
      "blastn -query /tmp/URI111.fasta -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/whole_plasmid/URI111_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 5\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/whole_plasmid/URI111_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "Running blast command:\n",
      "blastx -query /tmp/URI111.fasta -task \"blastx\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/pf32_db/pf32  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/pf32/URI111_pf32.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 5\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results/pf32/URI111_pf32.xml\n",
      "saddle up, we are done here!\n"
     ]
    }
   ],
   "source": [
    "if RUN_BLAST is True:\n",
    "    for isolate in assembly_dict:\n",
    "        input_files = [] # empty list to store input files\n",
    "        lr_data = assembly_dict[isolate]['longread']\n",
    "        sr_data = assembly_dict[isolate]['shortread']\n",
    "        lr_id = isolate # since we're operating on lr first\n",
    "        input_files.append(f'{longread_contigs}/{lr_id}.fasta') # add lr contigs to list\n",
    "        if sr_data == 'NOT AVAILABLE':\n",
    "            print(f'No shortread annotations found for {isolate}')\n",
    "            sr_id = None\n",
    "        else:\n",
    "            sr_id = isolate[:-1:]\n",
    "            input_files.append(f'{shortread_contigs}/{sr_id}.fasta')\n",
    "        print(f'setting up blast commands for {isolate}')\n",
    "        for file in input_files:\n",
    "            name = Path(file).stem\n",
    "            path = os.path.dirname(file)\n",
    "            print(path)\n",
    "            bind_path = f'/tmp/{name}.fasta'\n",
    "            wp_output_file = f'{plasmid_blast_results}/{name}_whole_plasmid.xml'\n",
    "            wp_blast_cmd = f'blastn -query {bind_path} -task \"blastn\" '\n",
    "            wp_blast_cmd += f'-db {whole_plasmid_db} '\n",
    "            wp_blast_cmd += f' -out {wp_output_file} '\n",
    "            wp_blast_cmd += \"-evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 5\"\n",
    "            print(f'Running blast command:\\n{wp_blast_cmd}')\n",
    "            #subprocess.run(f'singularity exec --bind {path}:/tmp containers/blast.simg {wp_blast_cmd}', shell=True, check=False) # run it\n",
    "            print(f'whole plasmid blast results written to: {wp_output_file}')\n",
    "            print(f'setting up blast command for pf32 blast')\n",
    "            pf_output_file = f'{pf32_blast_results}/{name}_pf32.xml'\n",
    "            pf_blast_cmd = f'blastx -query {bind_path} -task \"blastx\" '\n",
    "            pf_blast_cmd += f'-db {pf32_db} '\n",
    "            pf_blast_cmd += f' -out {pf_output_file} '\n",
    "            pf_blast_cmd += \"-evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 5\"\n",
    "            print(f'Running blast command:\\n{pf_blast_cmd}')\n",
    "            #subprocess.run(f'singularity exec --bind {path}:/tmp containers/blast.simg {pf_blast_cmd}', shell=True, check=False) # run it, but don't check for errors to avoid stopping the script\n",
    "            print(f'pf32 blast results written to: {pf_output_file}')\n",
    "    print('saddle up, we are done here!')\n",
    "else:\n",
    "    print('Blast already run, skipping blast')\n",
    "    "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## ~~Ok so now let's strip out the pfam32s and blast those against the testor set from Ira!~~ will implement later after parsing "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# let's define some possible annotated names!\n",
    "#genbanks = glob.glob(f'{assembly_annotations}/*.gbff')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# lets make a dictionary containing our contigs, parse the genbank and pull the top level details for this.\n",
    "# I forgot that I already did this below but this can hang around in case I need it (I probably will...)\n",
    "#assembly_dict = {}\n",
    "#for file in genbanks:\n",
    "#    print(file.split('/')[-1])\n",
    "#    sample_id = file.split(\"/\")[-1].split(\".\")[0]\n",
    "#    assembly_dict[sample_id] = {}\n",
    "#    with open(file, \"r\") as f:\n",
    "#        record = SeqIO.parse(f, \"genbank\")\n",
    "#        for rec in record:\n",
    "#            contig_id = rec.id\n",
    "#            assembly_dict[sample_id][\"contigs\"] = {contig_id: {}}\n",
    "#            contig_sequence = rec.seq\n",
    "#            contig_len = len(rec.seq)\n",
    "#            num_feats = len(rec.features)\n",
    "#            assembly_dict[sample_id][\"contigs\"][contig_id] = {\"contig_seq\" : contig_sequence, \"contig_len\": contig_len, \"feature_count\": num_feats}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#pfam32_names = [\"parA\",'ParA','pf32','partition'] # this needs considerable WORK, # I may just blast one of the testors and see what possible names come back.\n",
    "#parA_genes = {}\n",
    "#for file in genbanks:\n",
    "#    print(file)\n",
    "#    sample_id = file.split(\"/\")[-1].split(\".\")[0]\n",
    "#    parA_genes[sample_id] = []\n",
    "#    with open(file, \"r\") as f:\n",
    "#        record = SeqIO.parse(f, \"genbank\")\n",
    "#        for rec in record:\n",
    "#            #print(rec.id)\n",
    "#            for feature in rec.features:\n",
    "#                if feature.type == \"CDS\":\n",
    "#                    product = feature.qualifiers['product']\n",
    "#                    if any(pfam32_names) in product[0]:\n",
    "#                        feature.qualifiers['product']\n",
    "#                        nucseq = feature.extract(rec.seq)\n",
    "#                        if len(nucseq) % 3 != 0:\n",
    "#                            print(f\"Sequence length not divisible by 3 for {rec.id}---{sample_id}\")\n",
    "#                            nucseq += \"N\"\n",
    "#                            print(\"adding N!\")\n",
    "#                        else:\n",
    "#                            protseq = nucseq.translate(table=11, to_stop=True)\n",
    "#                            gene = SeqIO.SeqRecord(protseq, id=f\"{sample_id}-{rec.id}-ParA\", name=product[0], description=f\"{rec.id}---{sample_id}\", dbxrefs=None)\n",
    "#                            parA_genes[sample_id].append(gene)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "## now write out to file for blastin!\n",
    "#with open('parA_genes_query_v2.faa', 'w') as out:\n",
    "#    for sample in parA_genes:\n",
    "#        for gene in parA_genes[sample]:\n",
    "#            print(gene)\n",
    "#            SeqIO.write(gene, out, \"fasta\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Let's parse our blast hits now!"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "pf32_results = glob.glob(f'{pf32_blast_results}/*.xml')\n",
    "wp_results = glob.glob(f'{plasmid_blast_results}/*.xml')\n",
    "mjf.logdebug(f'Found {len(pf32_results)} pf32 blast results')\n",
    "mjf.logdebug(f'Found {len(wp_results)} whole plasmid blast results')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "for result in pf32_results:\n",
    "    print(result)\n",
    "    name = Path(result).stem.split(\"_\")[0]\n",
    "    if name.endswith(('H','P')):\n",
    "        print(f'input is longread: {name}')\n",
    "        name = name[:-1:] # chop the char for dict lookup.\n",
    "        current_asm = assembly_dict[name]['longread']\n",
    "    else:\n",
    "        print(f'input is shortread: {name}')\n",
    "        current_asm = assembly_dict[name]['shortread']\n",
    "    pf32_hits = parse_blast_results(result)\n",
    "    for hit in pf32_hits:\n",
    "        for k, v in enumerate(pf32_hits[hit]):\n",
    "            curr_contig = current_asm[v[\"contig_id\"]]\n",
    "            curr_contig['pf32_hits'] = v"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "for result in wp_results:\n",
    "    print(result)\n",
    "    name = Path(result).stem.split(\"_\")[0]\n",
    "    if name.endswith(('H','P')):\n",
    "        print(f'input is longread: {name}')\n",
    "        name = name[:-1:] # chop the char for dict lookup.\n",
    "        current_asm = assembly_dict[name]['longread']\n",
    "    else:\n",
    "        print(f'input is shortread: {name}')\n",
    "        current_asm = assembly_dict[name]['shortread']\n",
    "    wp_hits = parse_blast_results(result)\n",
    "    for hit in wp_hits:\n",
    "        for k, v in enumerate(wp_hits[hit]):\n",
    "            curr_contig = current_asm[v[\"contig_id\"]]\n",
    "            curr_contig['wp_hits'] = v\n",
    "            #print(f'Contig: {v[\"contig_id\"]}')\n",
    "            #print(f'Contig Length: {v[\"contig_len\"]}')\n",
    "            #print(f'Hit: {v[\"hit_title\"]}')\n",
    "            #print(f'Hit_Def: {v[\"hit_def\"]}')\n",
    "            #print(f'Expect: {v[\"hit_expect\"]}')\n",
    "            #print(f'Score: {v[\"hit_score\"]}')\n",
    "            #print(f'Query Start: {v[\"hit_query_start\"]}')\n",
    "            #print(f'Query End: {v[\"hit_query_end\"]}')\n",
    "            #print(f'Subject Start: {v[\"hit_subject_start\"]}')\n",
    "            #print(f'Subject End: {v[\"hit_subject_end\"]}')\n",
    "            #print(f'Identities: {v[\"hit_idents\"]}')\n",
    "            #print(f'Positives: {v[\"hit_pos\"]}')\n",
    "            #print(f'Gaps: {v[\"hit_gaps\"]}')\n",
    "            #print(f'Frame: {v[\"hit_frame\"]}')\n",
    "            #print(f'Query:   {v[\"hit_query\"]}')\n",
    "            #print(f'Subject: {v[\"hit_subject\"]}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "pickle.dump(assembly_dict, open(f'{output_dir}/assembly_dict_v3_parsed.pkl', 'wb'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "for isolate in assembly_dict:\n",
    "    for method in assembly_dict[isolate]:\n",
    "        for contig in assembly_dict[isolate][method]:\n",
    "            print(f'Isolate: {isolate}')\n",
    "            print(f'Method: {method}')\n",
    "            print(f'Contig: {assembly_dict[isolate][method][contig].items()}')\n",
    "            print(f'pf32_hits: {[hit[\"contig_id\"] for hit in assembly_dict[isolate][method][contig][\"pf32_hits\"]]}')\n",
    "            print(f'wp_hits: {[hit[\"contig_id\"] for hit in assembly_dict[isolate][method][contig][\"wp_hits\"]]}')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
