{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import sys\n",
    "import subprocess\n",
    "import glob\n",
    "import json\n",
    "import pandas\n",
    "import pickle\n",
    "import pprint\n",
    "from axe import axe # This package needs HELLA work.\n",
    "from pathlib import Path\n",
    "from collections import defaultdict # I like this collection. I am going to use this much more often.\n",
    "from Bio import SeqIO\n",
    "from Bio.Seq import Seq\n",
    "from Bio.Blast import NCBIXML\n",
    "from Bio.SeqRecord import SeqRecord\n",
    "from Bio.Blast.Applications import NcbiblastpCommandline"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "RUN_BLAST = True\n",
    "PARSE_BLAST = True"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "def parse_blast_results(blast_results : str, mode) -> dict:\n",
    "    file_size = os.path.getsize(blast_results)\n",
    "    if file_size == 0:\n",
    "        return None\n",
    "    results = {}\n",
    "    with open(blast_results, \"r\") as f:\n",
    "        blast_records = NCBIXML.parse(f)\n",
    "        for record in blast_records:\n",
    "            results[record.query] = []\n",
    "            for alignment in record.alignments:\n",
    "                #identity_and_coverage = calculate_percent_identity_and_coverage(alignment)\n",
    "                #results[record.query].append((alignment.hit_id, identity_and_coverage))\n",
    "                for hsp in alignment.hsps:\n",
    "                    results[record.query].append({'contig_id' : record.query,\n",
    "                                                  'contig_len' : record.query_length,\n",
    "                                                  'hit_id' : alignment.hit_id,\n",
    "                                                  'hit_title' : alignment.title,\n",
    "                                                  'hit_def' : alignment.hit_def,\n",
    "                                                  'hit_expect' : hsp.expect,\n",
    "                                                  'hit_score' : hsp.score,\n",
    "                                                  'hit_query_start' : hsp.query_start,\n",
    "                                                  'hit_query_end' : hsp.query_end,\n",
    "                                                  'hit_subject_start' : hsp.sbjct_start,\n",
    "                                                  'hit_subject_end' : hsp.sbjct_end,\n",
    "                                                  'hit_idents' : hsp.identities,\n",
    "                                                  'hit_pos' : hsp.positives,\n",
    "                                                  'hit_gaps' : hsp.gaps,\n",
    "                                                  'hit_frame' : hsp.frame,\n",
    "                                                  'hit_query' : hsp.query,\n",
    "                                                  'hit_subject' : hsp.sbjct,\n",
    "                                                  })\n",
    "    return results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "testing_set_file = 'dbs/Minimal-PF32-testor-set-March2021.faa'\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [],
   "source": [
    "# THIS NEEDS TO BE DONE BETTER, I DO NOT KNOW HOW TO DO THAT AT THIS POINT BUT I WILL FIGURE IT OUT.\n",
    "# MAYBE CALLING FROM A LIST OF PATHS OR SOMETHING?\n",
    "# TBD {{TODO}}\n",
    "\n",
    "#### Define Inputs and Paths ####\n",
    "# get CURRENT working directory\n",
    "init_cwd = os.getcwd()\n",
    "# Set working directory\n",
    "working_dir = '/home/mf019/borrelia_plasmid_classifier_v3'\n",
    "# path out the databases\n",
    "databases_dir = f'{working_dir}/dbs'\n",
    "whole_plasmid_db = f'{databases_dir}/plasmid_db_nf/plasmid_db'\n",
    "pf32_db = f'{databases_dir}/pf32_db/pfam32db'\n",
    "# lets specify where parsing tables are to be found!\n",
    "parsing_tables_dir = f'{working_dir}/parsing_tables'\n",
    "# set assemblies directory # DO I WANT TO ARGV THIS?\n",
    "assemblies_dir = f'{working_dir}/assemblies' #f'{working_dir}/assemblies'\n",
    "shortread_dir  = f'{assemblies_dir}/shortread'\n",
    "longread_dir = f'{assemblies_dir}/longread'\n",
    "shortread_contigs  = f'{shortread_dir}/contigs'\n",
    "longread_contigs = f'{longread_dir}/contigs'\n",
    "shortread_annotations  = f'{shortread_dir}/annotation'\n",
    "longread_annotations = f'{longread_dir}/annotation'\n",
    "output_dir = f'{working_dir}/output'\n",
    "blast_results_dir = f'{output_dir}/blast_results_v4'\n",
    "pf32_blast_results = f'{blast_results_dir}/pf32'\n",
    "plasmid_blast_results = f'{blast_results_dir}/whole_plasmid'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "analysis inputs\n",
      "CWD: /home/mf019/borrelia_plasmid_classifier_v3\n",
      "Working directory: /home/mf019/borrelia_plasmid_classifier_v3\n",
      "Databases directory: /home/mf019/borrelia_plasmid_classifier_v3/dbs\n",
      "Plasmid database: /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db_nf/plasmid_db\n",
      "PF32 database: /home/mf019/borrelia_plasmid_classifier_v3/dbs/pf32_db/pf32db\n",
      "Parsing tables directory: /home/mf019/borrelia_plasmid_classifier_v3/parsing_tables\n",
      "Assemblies directory: /home/mf019/borrelia_plasmid_classifier_v3/assemblies\n",
      "Shortread directory: /home/mf019/borrelia_plasmid_classifier_v3/assemblies/shortread\n",
      "Longread directory: /home/mf019/borrelia_plasmid_classifier_v3/assemblies/longread\n",
      "Shortread contigs directory: /home/mf019/borrelia_plasmid_classifier_v3/assemblies/shortread/contigs\n",
      "Shortread annotations directory: /home/mf019/borrelia_plasmid_classifier_v3/assemblies/shortread/annotation\n",
      "Longread contigs directory: /home/mf019/borrelia_plasmid_classifier_v3/assemblies/longread/contigs\n",
      "Longread annotations directory: /home/mf019/borrelia_plasmid_classifier_v3/assemblies/longread/annotation\n",
      "Output directory: /home/mf019/borrelia_plasmid_classifier_v3/output\n",
      "Whole contig blast results directory: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4\n",
      "PF32 Blast results: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32\n",
      "analysis begin\n"
     ]
    }
   ],
   "source": [
    "print('analysis inputs')\n",
    "print(f'CWD: {init_cwd}')\n",
    "print(f'Working directory: {working_dir}')\n",
    "print(f'Databases directory: {databases_dir}')\n",
    "print(f'Plasmid database: {whole_plasmid_db}')\n",
    "print(f'PF32 database: {pf32_db}')\n",
    "print(f'Parsing tables directory: {parsing_tables_dir}')\n",
    "print(f'Assemblies directory: {assemblies_dir}')\n",
    "print(f'Shortread directory: {shortread_dir}')\n",
    "print(f'Longread directory: {longread_dir}')\n",
    "print(f'Shortread contigs directory: {shortread_contigs}')\n",
    "print(f'Shortread annotations directory: {shortread_annotations}')\n",
    "print(f'Longread contigs directory: {longread_contigs}')\n",
    "print(f'Longread annotations directory: {longread_annotations}')\n",
    "print(f'Output directory: {output_dir}')\n",
    "print(f'Whole contig blast results directory: {blast_results_dir}')\n",
    "print(f'PF32 Blast results: {pf32_blast_results}')\n",
    "print('analysis begin')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['UCT30.fasta',\n",
       " 'URI41.fasta',\n",
       " 'URI44.fasta',\n",
       " 'UWI247P.fasta',\n",
       " 'URI111.fasta',\n",
       " '.ipynb_checkpoints',\n",
       " 'UWI247.fasta',\n",
       " 'URI91.fasta',\n",
       " 'URI48H.fasta',\n",
       " 'UWI248P.fasta',\n",
       " 'UCT29H.fasta',\n",
       " 'UCT92.fasta',\n",
       " 'UCT50H.fasta',\n",
       " 'UNY149P.fasta',\n",
       " 'URI47H.fasta',\n",
       " 'URI40.fasta',\n",
       " 'URI107H.fasta',\n",
       " 'UNY193P.fasta',\n",
       " 'UNY149.fasta',\n",
       " 'UCT109.fasta',\n",
       " 'UCT29.fasta',\n",
       " 'URI120.fasta',\n",
       " 'URI33H.fasta',\n",
       " 'UNY203P.fasta',\n",
       " 'URI91H.fasta',\n",
       " 'URI88H.fasta',\n",
       " 'URI44H.fasta',\n",
       " 'URI89.fasta',\n",
       " 'UCT109H.fasta',\n",
       " 'UNY172P.fasta',\n",
       " 'URI101.fasta',\n",
       " 'URI33.fasta',\n",
       " 'UCT110H.fasta',\n",
       " 'UCT31.fasta',\n",
       " 'UWI248.fasta',\n",
       " 'URI101H.fasta',\n",
       " 'URI86H.fasta',\n",
       " 'URI42.fasta',\n",
       " 'URI118H.fasta',\n",
       " 'ESI26H.fasta',\n",
       " 'URI34.fasta',\n",
       " 'URI47.fasta',\n",
       " 'UNY203.fasta',\n",
       " 'UCT35H.fasta',\n",
       " 'URI103H.fasta',\n",
       " 'URI103.fasta',\n",
       " 'UCT30H.fasta',\n",
       " 'UNY208.fasta',\n",
       " 'UCT113H.fasta',\n",
       " 'UCT96.fasta',\n",
       " 'UNY172.fasta',\n",
       " 'ESI26.fasta',\n",
       " 'UCT31H.fasta',\n",
       " 'URI88.fasta',\n",
       " 'UNY208P.fasta',\n",
       " 'URI87H.fasta',\n",
       " 'URI87.fasta',\n",
       " 'URI42H.fasta',\n",
       " 'URI112.fasta',\n",
       " 'UCT113.fasta',\n",
       " 'UWI263P.fasta',\n",
       " 'URI39H.fasta',\n",
       " 'URI112H.fasta',\n",
       " 'UCT32H.fasta',\n",
       " 'URI56H.fasta',\n",
       " 'URI117.fasta',\n",
       " 'URI89H.fasta',\n",
       " 'URI34H.fasta',\n",
       " 'URI93H.fasta',\n",
       " 'longread',\n",
       " 'UCT32.fasta',\n",
       " 'UCT92H.fasta',\n",
       " 'URI111H.fasta',\n",
       " 'URI56.fasta',\n",
       " 'UNY193.fasta',\n",
       " 'URI36.fasta',\n",
       " 'UWI283P.fasta',\n",
       " 'URI102.fasta',\n",
       " 'UWI283.fasta',\n",
       " 'shortread',\n",
       " 'URI102H.fasta',\n",
       " 'UCT96H.fasta',\n",
       " 'URI36H.fasta',\n",
       " 'URI86.fasta',\n",
       " 'URI48.fasta',\n",
       " 'URI118.fasta',\n",
       " 'URI93.fasta',\n",
       " 'UNY169.fasta',\n",
       " 'URI120H.fasta',\n",
       " 'URI40H.fasta',\n",
       " 'URI107.fasta',\n",
       " 'UNY169P.fasta',\n",
       " 'URI39.fasta',\n",
       " 'UWI263.fasta',\n",
       " 'UCT50.fasta',\n",
       " 'URI46.fasta',\n",
       " 'UCT35.fasta',\n",
       " 'URI117H.fasta',\n",
       " 'URI41H.fasta',\n",
       " 'URI46H.fasta',\n",
       " 'UCT110.fasta',\n",
       " '.DS_Store']"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "os.listdir(assemblies_dir)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Found 49 shortread gbff files\n",
      "Found 49 longread gbff files\n"
     ]
    }
   ],
   "source": [
    "# get annotations for each method\n",
    "sr_gbs = glob.glob(f'{shortread_annotations}/*/*.gbff')\n",
    "print(f'Found {len(sr_gbs)} shortread gbff files')\n",
    "lr_gbs = glob.glob(f'{longread_annotations}/*/*.gbff')\n",
    "print(f'Found {len(lr_gbs)} longread gbff files')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "URI47\n",
      "URI103\n",
      "UCT31\n",
      "UCT92\n",
      "URI46\n",
      "UWI247\n",
      "UCT30\n",
      "URI44\n",
      "URI87\n",
      "UCT50\n",
      "UWI283\n",
      "URI107\n",
      "UCT110\n",
      "URI48\n",
      "UNY203\n",
      "UWI263\n",
      "URI117\n",
      "UCT35\n",
      "URI111\n",
      "URI33\n",
      "URI86\n",
      "UNY149\n",
      "UCT96\n",
      "URI56\n",
      "URI118\n",
      "UNY172\n",
      "URI112\n",
      "UNY208\n",
      "URI89\n",
      "URI102\n",
      "URI36\n",
      "UWI248\n",
      "ESI26\n",
      "URI40\n",
      "URI39\n",
      "UNY169\n",
      "UCT113\n",
      "URI101\n",
      "URI42\n",
      "UCT109\n",
      "UCT32\n",
      "URI41\n",
      "URI88\n",
      "URI91\n",
      "UCT29\n",
      "URI34\n",
      "URI120\n",
      "UNY193\n",
      "URI93\n",
      "URI87H\n",
      "URI34H\n",
      "URI88H\n",
      "URI33H\n",
      "UCT110H\n",
      "URI39H\n",
      "URI91H\n",
      "UCT35H\n",
      "UWI247P\n",
      "URI120H\n",
      "URI107H\n",
      "UWI263P\n",
      "URI89H\n",
      "URI42H\n",
      "URI44H\n",
      "UCT109H\n",
      "URI40H\n",
      "URI117H\n",
      "URI47H\n",
      "URI86H\n",
      "URI36H\n",
      "UNY208P\n",
      "ESI26H\n",
      "UCT31H\n",
      "URI56H\n",
      "UCT30H\n",
      "URI103H\n",
      "UCT29H\n",
      "URI112H\n",
      "UWI248P\n",
      "UNY203P\n",
      "UCT96H\n",
      "UCT32H\n",
      "UNY193P\n",
      "UCT113H\n",
      "URI93H\n",
      "UNY169P\n",
      "UWI283P\n",
      "URI102H\n",
      "URI41H\n",
      "UNY172P\n",
      "UNY149P\n",
      "UCT92H\n",
      "URI118H\n",
      "UCT50H\n",
      "URI101H\n",
      "URI46H\n",
      "URI48H\n",
      "URI111H\n",
      "dictionary construction complete! pickling!\n",
      "dictionary pickled to: /home/mf019/borrelia_plasmid_classifier_v3/output/assembly_dict_v4.pkl\n"
     ]
    }
   ],
   "source": [
    "assembly_dict = {}\n",
    "for assembly in sr_gbs:\n",
    "    isolate_name = str(assembly.split('/')[-1].strip('.gbff'))\n",
    "    print(isolate_name)\n",
    "    assembly_dict[isolate_name] = {'shortread' : defaultdict(dict), 'longread' : defaultdict(dict), 'longread_name' : None}\n",
    "    for contig in SeqIO.parse(assembly, 'genbank'):\n",
    "        #print(contig.name)\n",
    "        assembly_dict[isolate_name]['shortread'][contig.name] = defaultdict(dict)\n",
    "        assembly_dict[isolate_name]['shortread'][contig.name]['seqrecord'] = contig\n",
    "        assembly_dict[isolate_name]['shortread'][contig.name]['wp_hits'] = []\n",
    "        #print(f'added keys for wphits to {isolate_name} {contig.name}')\n",
    "        assembly_dict[isolate_name]['shortread'][contig.name]['pf32_hits'] = []\n",
    "        #print(f'added keys for pf32hits to {isolate_name} {contig.name}')\n",
    "for assembly in lr_gbs:\n",
    "    isolate_name = assembly.split('/')[-1].strip('.gbff')\n",
    "    print(isolate_name)\n",
    "    sr_name = isolate_name[:-1:]\n",
    "    assembly_dict[sr_name]['longread_name'] = isolate_name\n",
    "    for contig in SeqIO.parse(assembly, 'genbank'):\n",
    "        #print(contig.name)\n",
    "        assembly_dict[sr_name]['longread'][contig.name] = defaultdict(dict)\n",
    "        assembly_dict[sr_name]['longread'][contig.name]['seqrecord'] = contig\n",
    "        assembly_dict[sr_name]['longread'][contig.name]['wp_hits'] = []\n",
    "        #print(f'added keys for wphits to {isolate_name} {contig.name}')\n",
    "        assembly_dict[sr_name]['longread'][contig.name]['pf32_hits'] = []\n",
    "        #print(f'added keys for pf32hits to {isolate_name} {contig.name}')\n",
    "print('dictionary construction complete! pickling!')\n",
    "pickle.dump(assembly_dict, open(f'{output_dir}/assembly_dict_v4.pkl', 'wb'))\n",
    "print(f'dictionary pickled to: {output_dir}/assembly_dict_v4.pkl')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Start Blastin\n",
      "setting up blast commands for URI47\n",
      "Running blast command:\n",
      "blastn -query /home/mf019/borrelia_plasmid_classifier_v3/assemblies/longread/annotation/URI47H/URI47H.fna -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db_nf/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI47H_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 10\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI47H_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI47H_pf32.xml\n",
      "Running blast command:\n",
      "blastn -query /home/mf019/borrelia_plasmid_classifier_v3/assemblies/shortread/annotation/URI47/URI47.fna -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db_nf/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI47_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 10\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI47_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI47_pf32.xml\n",
      "setting up blast commands for URI103\n",
      "Running blast command:\n",
      "blastn -query /home/mf019/borrelia_plasmid_classifier_v3/assemblies/longread/annotation/URI103H/URI103H.fna -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db_nf/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI103H_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 10\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI103H_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI103H_pf32.xml\n",
      "Running blast command:\n",
      "blastn -query /home/mf019/borrelia_plasmid_classifier_v3/assemblies/shortread/annotation/URI103/URI103.fna -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db_nf/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI103_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 10\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI103_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI103_pf32.xml\n",
      "setting up blast commands for UCT31\n",
      "Running blast command:\n",
      "blastn -query /home/mf019/borrelia_plasmid_classifier_v3/assemblies/longread/annotation/UCT31H/UCT31H.fna -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db_nf/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UCT31H_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 10\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UCT31H_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UCT31H_pf32.xml\n",
      "Running blast command:\n",
      "blastn -query /home/mf019/borrelia_plasmid_classifier_v3/assemblies/shortread/annotation/UCT31/UCT31.fna -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db_nf/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UCT31_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 10\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UCT31_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UCT31_pf32.xml\n",
      "setting up blast commands for UCT92\n",
      "Running blast command:\n",
      "blastn -query /home/mf019/borrelia_plasmid_classifier_v3/assemblies/longread/annotation/UCT92H/UCT92H.fna -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db_nf/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UCT92H_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 10\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UCT92H_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UCT92H_pf32.xml\n",
      "Running blast command:\n",
      "blastn -query /home/mf019/borrelia_plasmid_classifier_v3/assemblies/shortread/annotation/UCT92/UCT92.fna -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db_nf/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UCT92_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 10\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UCT92_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UCT92_pf32.xml\n",
      "setting up blast commands for URI46\n",
      "Running blast command:\n",
      "blastn -query /home/mf019/borrelia_plasmid_classifier_v3/assemblies/longread/annotation/URI46H/URI46H.fna -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db_nf/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI46H_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 10\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI46H_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI46H_pf32.xml\n",
      "Running blast command:\n",
      "blastn -query /home/mf019/borrelia_plasmid_classifier_v3/assemblies/shortread/annotation/URI46/URI46.fna -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db_nf/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI46_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 10\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI46_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI46_pf32.xml\n",
      "setting up blast commands for UWI247\n",
      "Running blast command:\n",
      "blastn -query /home/mf019/borrelia_plasmid_classifier_v3/assemblies/longread/annotation/UWI247P/UWI247P.fna -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db_nf/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UWI247P_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 10\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UWI247P_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UWI247P_pf32.xml\n",
      "Running blast command:\n",
      "blastn -query /home/mf019/borrelia_plasmid_classifier_v3/assemblies/shortread/annotation/UWI247/UWI247.fna -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db_nf/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UWI247_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 10\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UWI247_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UWI247_pf32.xml\n",
      "setting up blast commands for UCT30\n",
      "Running blast command:\n",
      "blastn -query /home/mf019/borrelia_plasmid_classifier_v3/assemblies/longread/annotation/UCT30H/UCT30H.fna -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db_nf/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UCT30H_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 10\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UCT30H_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UCT30H_pf32.xml\n",
      "Running blast command:\n",
      "blastn -query /home/mf019/borrelia_plasmid_classifier_v3/assemblies/shortread/annotation/UCT30/UCT30.fna -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db_nf/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UCT30_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 10\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UCT30_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UCT30_pf32.xml\n",
      "setting up blast commands for URI44\n",
      "Running blast command:\n",
      "blastn -query /home/mf019/borrelia_plasmid_classifier_v3/assemblies/longread/annotation/URI44H/URI44H.fna -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db_nf/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI44H_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 10\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI44H_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI44H_pf32.xml\n",
      "Running blast command:\n",
      "blastn -query /home/mf019/borrelia_plasmid_classifier_v3/assemblies/shortread/annotation/URI44/URI44.fna -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db_nf/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI44_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 10\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI44_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI44_pf32.xml\n",
      "setting up blast commands for URI87\n",
      "Running blast command:\n",
      "blastn -query /home/mf019/borrelia_plasmid_classifier_v3/assemblies/longread/annotation/URI87H/URI87H.fna -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db_nf/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI87H_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 10\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI87H_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI87H_pf32.xml\n",
      "Running blast command:\n",
      "blastn -query /home/mf019/borrelia_plasmid_classifier_v3/assemblies/shortread/annotation/URI87/URI87.fna -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db_nf/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI87_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 10\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI87_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI87_pf32.xml\n",
      "setting up blast commands for UCT50\n",
      "Running blast command:\n",
      "blastn -query /home/mf019/borrelia_plasmid_classifier_v3/assemblies/longread/annotation/UCT50H/UCT50H.fna -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db_nf/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UCT50H_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 10\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UCT50H_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UCT50H_pf32.xml\n",
      "Running blast command:\n",
      "blastn -query /home/mf019/borrelia_plasmid_classifier_v3/assemblies/shortread/annotation/UCT50/UCT50.fna -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db_nf/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UCT50_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 10\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UCT50_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UCT50_pf32.xml\n",
      "setting up blast commands for UWI283\n",
      "Running blast command:\n",
      "blastn -query /home/mf019/borrelia_plasmid_classifier_v3/assemblies/longread/annotation/UWI283P/UWI283P.fna -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db_nf/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UWI283P_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 10\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UWI283P_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UWI283P_pf32.xml\n",
      "Running blast command:\n",
      "blastn -query /home/mf019/borrelia_plasmid_classifier_v3/assemblies/shortread/annotation/UWI283/UWI283.fna -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db_nf/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UWI283_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 10\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UWI283_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UWI283_pf32.xml\n",
      "setting up blast commands for URI107\n",
      "Running blast command:\n",
      "blastn -query /home/mf019/borrelia_plasmid_classifier_v3/assemblies/longread/annotation/URI107H/URI107H.fna -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db_nf/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI107H_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 10\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI107H_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI107H_pf32.xml\n",
      "Running blast command:\n",
      "blastn -query /home/mf019/borrelia_plasmid_classifier_v3/assemblies/shortread/annotation/URI107/URI107.fna -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db_nf/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI107_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 10\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI107_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI107_pf32.xml\n",
      "setting up blast commands for UCT110\n",
      "Running blast command:\n",
      "blastn -query /home/mf019/borrelia_plasmid_classifier_v3/assemblies/longread/annotation/UCT110H/UCT110H.fna -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db_nf/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UCT110H_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 10\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UCT110H_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UCT110H_pf32.xml\n",
      "Running blast command:\n",
      "blastn -query /home/mf019/borrelia_plasmid_classifier_v3/assemblies/shortread/annotation/UCT110/UCT110.fna -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db_nf/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UCT110_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 10\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UCT110_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UCT110_pf32.xml\n",
      "setting up blast commands for URI48\n",
      "Running blast command:\n",
      "blastn -query /home/mf019/borrelia_plasmid_classifier_v3/assemblies/longread/annotation/URI48H/URI48H.fna -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db_nf/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI48H_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 10\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI48H_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI48H_pf32.xml\n",
      "Running blast command:\n",
      "blastn -query /home/mf019/borrelia_plasmid_classifier_v3/assemblies/shortread/annotation/URI48/URI48.fna -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db_nf/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI48_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 10\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI48_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI48_pf32.xml\n",
      "setting up blast commands for UNY203\n",
      "Running blast command:\n",
      "blastn -query /home/mf019/borrelia_plasmid_classifier_v3/assemblies/longread/annotation/UNY203P/UNY203P.fna -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db_nf/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UNY203P_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 10\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UNY203P_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UNY203P_pf32.xml\n",
      "Running blast command:\n",
      "blastn -query /home/mf019/borrelia_plasmid_classifier_v3/assemblies/shortread/annotation/UNY203/UNY203.fna -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db_nf/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UNY203_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 10\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UNY203_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UNY203_pf32.xml\n",
      "setting up blast commands for UWI263\n",
      "Running blast command:\n",
      "blastn -query /home/mf019/borrelia_plasmid_classifier_v3/assemblies/longread/annotation/UWI263P/UWI263P.fna -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db_nf/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UWI263P_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 10\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UWI263P_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UWI263P_pf32.xml\n",
      "Running blast command:\n",
      "blastn -query /home/mf019/borrelia_plasmid_classifier_v3/assemblies/shortread/annotation/UWI263/UWI263.fna -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db_nf/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UWI263_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 10\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UWI263_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UWI263_pf32.xml\n",
      "setting up blast commands for URI117\n",
      "Running blast command:\n",
      "blastn -query /home/mf019/borrelia_plasmid_classifier_v3/assemblies/longread/annotation/URI117H/URI117H.fna -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db_nf/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI117H_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 10\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI117H_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI117H_pf32.xml\n",
      "Running blast command:\n",
      "blastn -query /home/mf019/borrelia_plasmid_classifier_v3/assemblies/shortread/annotation/URI117/URI117.fna -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db_nf/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI117_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 10\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI117_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI117_pf32.xml\n",
      "setting up blast commands for UCT35\n",
      "Running blast command:\n",
      "blastn -query /home/mf019/borrelia_plasmid_classifier_v3/assemblies/longread/annotation/UCT35H/UCT35H.fna -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db_nf/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UCT35H_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 10\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UCT35H_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UCT35H_pf32.xml\n",
      "Running blast command:\n",
      "blastn -query /home/mf019/borrelia_plasmid_classifier_v3/assemblies/shortread/annotation/UCT35/UCT35.fna -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db_nf/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UCT35_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 10\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UCT35_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UCT35_pf32.xml\n",
      "setting up blast commands for URI111\n",
      "Running blast command:\n",
      "blastn -query /home/mf019/borrelia_plasmid_classifier_v3/assemblies/longread/annotation/URI111H/URI111H.fna -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db_nf/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI111H_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 10\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI111H_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI111H_pf32.xml\n",
      "Running blast command:\n",
      "blastn -query /home/mf019/borrelia_plasmid_classifier_v3/assemblies/shortread/annotation/URI111/URI111.fna -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db_nf/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI111_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 10\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI111_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI111_pf32.xml\n",
      "setting up blast commands for URI33\n",
      "Running blast command:\n",
      "blastn -query /home/mf019/borrelia_plasmid_classifier_v3/assemblies/longread/annotation/URI33H/URI33H.fna -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db_nf/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI33H_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 10\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI33H_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI33H_pf32.xml\n",
      "Running blast command:\n",
      "blastn -query /home/mf019/borrelia_plasmid_classifier_v3/assemblies/shortread/annotation/URI33/URI33.fna -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db_nf/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI33_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 10\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI33_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI33_pf32.xml\n",
      "setting up blast commands for URI86\n",
      "Running blast command:\n",
      "blastn -query /home/mf019/borrelia_plasmid_classifier_v3/assemblies/longread/annotation/URI86H/URI86H.fna -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db_nf/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI86H_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 10\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI86H_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI86H_pf32.xml\n",
      "Running blast command:\n",
      "blastn -query /home/mf019/borrelia_plasmid_classifier_v3/assemblies/shortread/annotation/URI86/URI86.fna -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db_nf/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI86_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 10\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI86_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI86_pf32.xml\n",
      "setting up blast commands for UNY149\n",
      "Running blast command:\n",
      "blastn -query /home/mf019/borrelia_plasmid_classifier_v3/assemblies/longread/annotation/UNY149P/UNY149P.fna -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db_nf/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UNY149P_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 10\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UNY149P_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UNY149P_pf32.xml\n",
      "Running blast command:\n",
      "blastn -query /home/mf019/borrelia_plasmid_classifier_v3/assemblies/shortread/annotation/UNY149/UNY149.fna -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db_nf/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UNY149_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 10\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UNY149_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UNY149_pf32.xml\n",
      "setting up blast commands for UCT96\n",
      "Running blast command:\n",
      "blastn -query /home/mf019/borrelia_plasmid_classifier_v3/assemblies/longread/annotation/UCT96H/UCT96H.fna -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db_nf/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UCT96H_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 10\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UCT96H_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UCT96H_pf32.xml\n",
      "Running blast command:\n",
      "blastn -query /home/mf019/borrelia_plasmid_classifier_v3/assemblies/shortread/annotation/UCT96/UCT96.fna -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db_nf/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UCT96_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 10\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UCT96_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UCT96_pf32.xml\n",
      "setting up blast commands for URI56\n",
      "Running blast command:\n",
      "blastn -query /home/mf019/borrelia_plasmid_classifier_v3/assemblies/longread/annotation/URI56H/URI56H.fna -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db_nf/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI56H_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 10\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI56H_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI56H_pf32.xml\n",
      "Running blast command:\n",
      "blastn -query /home/mf019/borrelia_plasmid_classifier_v3/assemblies/shortread/annotation/URI56/URI56.fna -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db_nf/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI56_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 10\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI56_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI56_pf32.xml\n",
      "setting up blast commands for URI118\n",
      "Running blast command:\n",
      "blastn -query /home/mf019/borrelia_plasmid_classifier_v3/assemblies/longread/annotation/URI118H/URI118H.fna -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db_nf/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI118H_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 10\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI118H_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI118H_pf32.xml\n",
      "Running blast command:\n",
      "blastn -query /home/mf019/borrelia_plasmid_classifier_v3/assemblies/shortread/annotation/URI118/URI118.fna -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db_nf/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI118_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 10\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI118_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI118_pf32.xml\n",
      "setting up blast commands for UNY172\n",
      "Running blast command:\n",
      "blastn -query /home/mf019/borrelia_plasmid_classifier_v3/assemblies/longread/annotation/UNY172P/UNY172P.fna -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db_nf/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UNY172P_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 10\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UNY172P_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UNY172P_pf32.xml\n",
      "Running blast command:\n",
      "blastn -query /home/mf019/borrelia_plasmid_classifier_v3/assemblies/shortread/annotation/UNY172/UNY172.fna -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db_nf/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UNY172_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 10\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UNY172_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UNY172_pf32.xml\n",
      "setting up blast commands for URI112\n",
      "Running blast command:\n",
      "blastn -query /home/mf019/borrelia_plasmid_classifier_v3/assemblies/longread/annotation/URI112H/URI112H.fna -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db_nf/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI112H_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 10\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI112H_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI112H_pf32.xml\n",
      "Running blast command:\n",
      "blastn -query /home/mf019/borrelia_plasmid_classifier_v3/assemblies/shortread/annotation/URI112/URI112.fna -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db_nf/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI112_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 10\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI112_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI112_pf32.xml\n",
      "setting up blast commands for UNY208\n",
      "Running blast command:\n",
      "blastn -query /home/mf019/borrelia_plasmid_classifier_v3/assemblies/longread/annotation/UNY208P/UNY208P.fna -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db_nf/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UNY208P_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 10\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UNY208P_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UNY208P_pf32.xml\n",
      "Running blast command:\n",
      "blastn -query /home/mf019/borrelia_plasmid_classifier_v3/assemblies/shortread/annotation/UNY208/UNY208.fna -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db_nf/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UNY208_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 10\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UNY208_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UNY208_pf32.xml\n",
      "setting up blast commands for URI89\n",
      "Running blast command:\n",
      "blastn -query /home/mf019/borrelia_plasmid_classifier_v3/assemblies/longread/annotation/URI89H/URI89H.fna -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db_nf/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI89H_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 10\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI89H_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI89H_pf32.xml\n",
      "Running blast command:\n",
      "blastn -query /home/mf019/borrelia_plasmid_classifier_v3/assemblies/shortread/annotation/URI89/URI89.fna -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db_nf/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI89_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 10\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI89_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI89_pf32.xml\n",
      "setting up blast commands for URI102\n",
      "Running blast command:\n",
      "blastn -query /home/mf019/borrelia_plasmid_classifier_v3/assemblies/longread/annotation/URI102H/URI102H.fna -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db_nf/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI102H_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 10\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI102H_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI102H_pf32.xml\n",
      "Running blast command:\n",
      "blastn -query /home/mf019/borrelia_plasmid_classifier_v3/assemblies/shortread/annotation/URI102/URI102.fna -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db_nf/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI102_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 10\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI102_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI102_pf32.xml\n",
      "setting up blast commands for URI36\n",
      "Running blast command:\n",
      "blastn -query /home/mf019/borrelia_plasmid_classifier_v3/assemblies/longread/annotation/URI36H/URI36H.fna -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db_nf/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI36H_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 10\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI36H_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI36H_pf32.xml\n",
      "Running blast command:\n",
      "blastn -query /home/mf019/borrelia_plasmid_classifier_v3/assemblies/shortread/annotation/URI36/URI36.fna -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db_nf/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI36_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 10\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI36_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI36_pf32.xml\n",
      "setting up blast commands for UWI248\n",
      "Running blast command:\n",
      "blastn -query /home/mf019/borrelia_plasmid_classifier_v3/assemblies/longread/annotation/UWI248P/UWI248P.fna -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db_nf/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UWI248P_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 10\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UWI248P_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UWI248P_pf32.xml\n",
      "Running blast command:\n",
      "blastn -query /home/mf019/borrelia_plasmid_classifier_v3/assemblies/shortread/annotation/UWI248/UWI248.fna -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db_nf/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UWI248_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 10\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UWI248_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UWI248_pf32.xml\n",
      "setting up blast commands for ESI26\n",
      "Running blast command:\n",
      "blastn -query /home/mf019/borrelia_plasmid_classifier_v3/assemblies/longread/annotation/ESI26H/ESI26H.fna -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db_nf/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/ESI26H_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 10\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/ESI26H_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/ESI26H_pf32.xml\n",
      "Running blast command:\n",
      "blastn -query /home/mf019/borrelia_plasmid_classifier_v3/assemblies/shortread/annotation/ESI26/ESI26.fna -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db_nf/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/ESI26_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 10\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/ESI26_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/ESI26_pf32.xml\n",
      "setting up blast commands for URI40\n",
      "Running blast command:\n",
      "blastn -query /home/mf019/borrelia_plasmid_classifier_v3/assemblies/longread/annotation/URI40H/URI40H.fna -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db_nf/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI40H_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 10\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI40H_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI40H_pf32.xml\n",
      "Running blast command:\n",
      "blastn -query /home/mf019/borrelia_plasmid_classifier_v3/assemblies/shortread/annotation/URI40/URI40.fna -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db_nf/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI40_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 10\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI40_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI40_pf32.xml\n",
      "setting up blast commands for URI39\n",
      "Running blast command:\n",
      "blastn -query /home/mf019/borrelia_plasmid_classifier_v3/assemblies/longread/annotation/URI39H/URI39H.fna -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db_nf/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI39H_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 10\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI39H_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI39H_pf32.xml\n",
      "Running blast command:\n",
      "blastn -query /home/mf019/borrelia_plasmid_classifier_v3/assemblies/shortread/annotation/URI39/URI39.fna -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db_nf/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI39_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 10\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI39_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI39_pf32.xml\n",
      "setting up blast commands for UNY169\n",
      "Running blast command:\n",
      "blastn -query /home/mf019/borrelia_plasmid_classifier_v3/assemblies/longread/annotation/UNY169P/UNY169P.fna -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db_nf/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UNY169P_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 10\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UNY169P_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UNY169P_pf32.xml\n",
      "Running blast command:\n",
      "blastn -query /home/mf019/borrelia_plasmid_classifier_v3/assemblies/shortread/annotation/UNY169/UNY169.fna -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db_nf/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UNY169_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 10\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UNY169_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UNY169_pf32.xml\n",
      "setting up blast commands for UCT113\n",
      "Running blast command:\n",
      "blastn -query /home/mf019/borrelia_plasmid_classifier_v3/assemblies/longread/annotation/UCT113H/UCT113H.fna -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db_nf/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UCT113H_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 10\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UCT113H_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UCT113H_pf32.xml\n",
      "Running blast command:\n",
      "blastn -query /home/mf019/borrelia_plasmid_classifier_v3/assemblies/shortread/annotation/UCT113/UCT113.fna -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db_nf/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UCT113_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 10\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UCT113_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UCT113_pf32.xml\n",
      "setting up blast commands for URI101\n",
      "Running blast command:\n",
      "blastn -query /home/mf019/borrelia_plasmid_classifier_v3/assemblies/longread/annotation/URI101H/URI101H.fna -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db_nf/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI101H_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 10\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI101H_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI101H_pf32.xml\n",
      "Running blast command:\n",
      "blastn -query /home/mf019/borrelia_plasmid_classifier_v3/assemblies/shortread/annotation/URI101/URI101.fna -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db_nf/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI101_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 10\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI101_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI101_pf32.xml\n",
      "setting up blast commands for URI42\n",
      "Running blast command:\n",
      "blastn -query /home/mf019/borrelia_plasmid_classifier_v3/assemblies/longread/annotation/URI42H/URI42H.fna -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db_nf/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI42H_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 10\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI42H_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI42H_pf32.xml\n",
      "Running blast command:\n",
      "blastn -query /home/mf019/borrelia_plasmid_classifier_v3/assemblies/shortread/annotation/URI42/URI42.fna -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db_nf/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI42_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 10\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI42_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI42_pf32.xml\n",
      "setting up blast commands for UCT109\n",
      "Running blast command:\n",
      "blastn -query /home/mf019/borrelia_plasmid_classifier_v3/assemblies/longread/annotation/UCT109H/UCT109H.fna -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db_nf/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UCT109H_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 10\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UCT109H_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UCT109H_pf32.xml\n",
      "Running blast command:\n",
      "blastn -query /home/mf019/borrelia_plasmid_classifier_v3/assemblies/shortread/annotation/UCT109/UCT109.fna -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db_nf/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UCT109_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 10\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UCT109_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UCT109_pf32.xml\n",
      "setting up blast commands for UCT32\n",
      "Running blast command:\n",
      "blastn -query /home/mf019/borrelia_plasmid_classifier_v3/assemblies/longread/annotation/UCT32H/UCT32H.fna -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db_nf/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UCT32H_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 10\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UCT32H_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UCT32H_pf32.xml\n",
      "Running blast command:\n",
      "blastn -query /home/mf019/borrelia_plasmid_classifier_v3/assemblies/shortread/annotation/UCT32/UCT32.fna -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db_nf/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UCT32_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 10\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UCT32_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UCT32_pf32.xml\n",
      "setting up blast commands for URI41\n",
      "Running blast command:\n",
      "blastn -query /home/mf019/borrelia_plasmid_classifier_v3/assemblies/longread/annotation/URI41H/URI41H.fna -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db_nf/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI41H_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 10\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI41H_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI41H_pf32.xml\n",
      "Running blast command:\n",
      "blastn -query /home/mf019/borrelia_plasmid_classifier_v3/assemblies/shortread/annotation/URI41/URI41.fna -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db_nf/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI41_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 10\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI41_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI41_pf32.xml\n",
      "setting up blast commands for URI88\n",
      "Running blast command:\n",
      "blastn -query /home/mf019/borrelia_plasmid_classifier_v3/assemblies/longread/annotation/URI88H/URI88H.fna -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db_nf/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI88H_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 10\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI88H_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI88H_pf32.xml\n",
      "Running blast command:\n",
      "blastn -query /home/mf019/borrelia_plasmid_classifier_v3/assemblies/shortread/annotation/URI88/URI88.fna -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db_nf/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI88_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 10\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI88_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI88_pf32.xml\n",
      "setting up blast commands for URI91\n",
      "Running blast command:\n",
      "blastn -query /home/mf019/borrelia_plasmid_classifier_v3/assemblies/longread/annotation/URI91H/URI91H.fna -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db_nf/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI91H_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 10\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI91H_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI91H_pf32.xml\n",
      "Running blast command:\n",
      "blastn -query /home/mf019/borrelia_plasmid_classifier_v3/assemblies/shortread/annotation/URI91/URI91.fna -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db_nf/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI91_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 10\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI91_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI91_pf32.xml\n",
      "setting up blast commands for UCT29\n",
      "Running blast command:\n",
      "blastn -query /home/mf019/borrelia_plasmid_classifier_v3/assemblies/longread/annotation/UCT29H/UCT29H.fna -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db_nf/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UCT29H_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 10\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UCT29H_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UCT29H_pf32.xml\n",
      "Running blast command:\n",
      "blastn -query /home/mf019/borrelia_plasmid_classifier_v3/assemblies/shortread/annotation/UCT29/UCT29.fna -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db_nf/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UCT29_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 10\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UCT29_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UCT29_pf32.xml\n",
      "setting up blast commands for URI34\n",
      "Running blast command:\n",
      "blastn -query /home/mf019/borrelia_plasmid_classifier_v3/assemblies/longread/annotation/URI34H/URI34H.fna -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db_nf/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI34H_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 10\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI34H_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI34H_pf32.xml\n",
      "Running blast command:\n",
      "blastn -query /home/mf019/borrelia_plasmid_classifier_v3/assemblies/shortread/annotation/URI34/URI34.fna -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db_nf/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI34_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 10\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI34_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI34_pf32.xml\n",
      "setting up blast commands for URI120\n",
      "Running blast command:\n",
      "blastn -query /home/mf019/borrelia_plasmid_classifier_v3/assemblies/longread/annotation/URI120H/URI120H.fna -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db_nf/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI120H_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 10\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI120H_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI120H_pf32.xml\n",
      "Running blast command:\n",
      "blastn -query /home/mf019/borrelia_plasmid_classifier_v3/assemblies/shortread/annotation/URI120/URI120.fna -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db_nf/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI120_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 10\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI120_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI120_pf32.xml\n",
      "setting up blast commands for UNY193\n",
      "Running blast command:\n",
      "blastn -query /home/mf019/borrelia_plasmid_classifier_v3/assemblies/longread/annotation/UNY193P/UNY193P.fna -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db_nf/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UNY193P_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 10\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UNY193P_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UNY193P_pf32.xml\n",
      "Running blast command:\n",
      "blastn -query /home/mf019/borrelia_plasmid_classifier_v3/assemblies/shortread/annotation/UNY193/UNY193.fna -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db_nf/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UNY193_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 10\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UNY193_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UNY193_pf32.xml\n",
      "setting up blast commands for URI93\n",
      "Running blast command:\n",
      "blastn -query /home/mf019/borrelia_plasmid_classifier_v3/assemblies/longread/annotation/URI93H/URI93H.fna -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db_nf/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI93H_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 10\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI93H_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI93H_pf32.xml\n",
      "Running blast command:\n",
      "blastn -query /home/mf019/borrelia_plasmid_classifier_v3/assemblies/shortread/annotation/URI93/URI93.fna -task \"blastn\" -db /home/mf019/borrelia_plasmid_classifier_v3/dbs/plasmid_db_nf/plasmid_db  -out /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI93_whole_plasmid.xml -evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 10\n",
      "whole plasmid blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI93_whole_plasmid.xml\n",
      "setting up blast command for pf32 blast\n",
      "pf32 blast results written to: /home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI93_pf32.xml\n",
      "saddle up, we are done here!\n"
     ]
    }
   ],
   "source": [
    "if RUN_BLAST is True:\n",
    "    print(str('Start Blastin'))\n",
    "    for isolate in assembly_dict:\n",
    "        input_files = [] # empty list to store input files\n",
    "        lr_data = assembly_dict[isolate]['longread']\n",
    "        sr_data = assembly_dict[isolate]['shortread']\n",
    "        lr_id = assembly_dict[isolate]['longread_name']\n",
    "        #lr_id = isolate # since we're operating on lr first\n",
    "        input_files.append(f'{longread_annotations}/{lr_id}/{lr_id}.fna') # add lr contigs to list\n",
    "        if sr_data == 'NOT AVAILABLE':\n",
    "            print(f'No shortread annotations found for {isolate}')\n",
    "            sr_id = None\n",
    "        else:\n",
    "            sr_id = isolate#[:-1:]\n",
    "            input_files.append(f'{shortread_annotations}/{sr_id}/{sr_id}.fna')\n",
    "        print(f'setting up blast commands for {isolate}')\n",
    "        for file in input_files:\n",
    "            name = Path(file).stem\n",
    "            wp_output_file = f'{plasmid_blast_results}/{name}_whole_plasmid.xml'\n",
    "            wp_blast_cmd = f'blastn -query {file} -task \"blastn\" '\n",
    "            wp_blast_cmd += f'-db {whole_plasmid_db} '\n",
    "            wp_blast_cmd += f' -out {wp_output_file} '\n",
    "            wp_blast_cmd += \"-evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 10\"\n",
    "            print(f'Running blast command:\\n{wp_blast_cmd}')\n",
    "            subprocess.run(wp_blast_cmd, shell=True) # run it\n",
    "            print(f'whole plasmid blast results written to: {wp_output_file}')\n",
    "            print(f'setting up blast command for pf32 blast')\n",
    "            pf_output_file = f'{pf32_blast_results}/{name}_pf32.xml'\n",
    "            pf_blast_cmd = f'blastx -query {file} -task \"blastx\" '\n",
    "            pf_blast_cmd += f'-db {pf32_db} '\n",
    "            pf_blast_cmd += f' -out {pf_output_file} '\n",
    "            pf_blast_cmd += \"-evalue 1e-100 -num_threads 60 -outfmt 5 -max_target_seqs 5 -max_hsps 10\"\n",
    "            subprocess.run(pf_blast_cmd, shell=True, check=False) # run it, but don't check for errors to avoid stopping the script\n",
    "            print(f'pf32 blast results written to: {pf_output_file}')\n",
    "    print('saddle up, we are done here!')\n",
    "else:\n",
    "    print('Blast already run, skipping blast')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Let's parse our blast hits now!"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Found 98 pf32 blast results\n",
      "Found 98 whole plasmid blast results\n"
     ]
    }
   ],
   "source": [
    "pf32_results = glob.glob(f'{pf32_blast_results}/*.xml')\n",
    "wp_results = glob.glob(f'{plasmid_blast_results}/*.xml')\n",
    "print(f'Found {len(pf32_results)} pf32 blast results')\n",
    "print(f'Found {len(wp_results)} whole plasmid blast results')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UCT35H_pf32.xml\n",
      "input is longread: UCT35H\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/ESI26_pf32.xml\n",
      "input is shortread: ESI26\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/ESI26H_pf32.xml\n",
      "input is longread: ESI26H\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UCT32H_pf32.xml\n",
      "input is longread: UCT32H\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UNY169P_pf32.xml\n",
      "input is longread: UNY169P\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UWI248P_pf32.xml\n",
      "input is longread: UWI248P\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI34H_pf32.xml\n",
      "input is longread: URI34H\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI111_pf32.xml\n",
      "input is shortread: URI111\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UCT29H_pf32.xml\n",
      "input is longread: UCT29H\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI33_pf32.xml\n",
      "input is shortread: URI33\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UWI247_pf32.xml\n",
      "input is shortread: UWI247\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UCT110_pf32.xml\n",
      "input is shortread: UCT110\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UCT31_pf32.xml\n",
      "input is shortread: UCT31\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI39_pf32.xml\n",
      "input is shortread: URI39\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UWI283_pf32.xml\n",
      "input is shortread: UWI283\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI120_pf32.xml\n",
      "input is shortread: URI120\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UCT35_pf32.xml\n",
      "input is shortread: UCT35\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI41_pf32.xml\n",
      "input is shortread: URI41\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UNY193P_pf32.xml\n",
      "input is longread: UNY193P\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UNY208P_pf32.xml\n",
      "input is longread: UNY208P\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI107_pf32.xml\n",
      "input is shortread: URI107\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI103_pf32.xml\n",
      "input is shortread: URI103\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UCT29_pf32.xml\n",
      "input is shortread: UCT29\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI89_pf32.xml\n",
      "input is shortread: URI89\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI39H_pf32.xml\n",
      "input is longread: URI39H\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI42H_pf32.xml\n",
      "input is longread: URI42H\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UNY172P_pf32.xml\n",
      "input is longread: UNY172P\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI41H_pf32.xml\n",
      "input is longread: URI41H\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UCT109H_pf32.xml\n",
      "input is longread: UCT109H\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI33H_pf32.xml\n",
      "input is longread: URI33H\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI40H_pf32.xml\n",
      "input is longread: URI40H\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI47_pf32.xml\n",
      "input is shortread: URI47\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI44H_pf32.xml\n",
      "input is longread: URI44H\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI46_pf32.xml\n",
      "input is shortread: URI46\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI86_pf32.xml\n",
      "input is shortread: URI86\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI102H_pf32.xml\n",
      "input is longread: URI102H\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UNY208_pf32.xml\n",
      "input is shortread: UNY208\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI88H_pf32.xml\n",
      "input is longread: URI88H\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UCT92_pf32.xml\n",
      "input is shortread: UCT92\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UCT50H_pf32.xml\n",
      "input is longread: UCT50H\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UCT92H_pf32.xml\n",
      "input is longread: UCT92H\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI48H_pf32.xml\n",
      "input is longread: URI48H\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI111H_pf32.xml\n",
      "input is longread: URI111H\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI44_pf32.xml\n",
      "input is shortread: URI44\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UCT109_pf32.xml\n",
      "input is shortread: UCT109\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI89H_pf32.xml\n",
      "input is longread: URI89H\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI46H_pf32.xml\n",
      "input is longread: URI46H\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UNY169_pf32.xml\n",
      "input is shortread: UNY169\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI88_pf32.xml\n",
      "input is shortread: URI88\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI102_pf32.xml\n",
      "input is shortread: URI102\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UCT113H_pf32.xml\n",
      "input is longread: UCT113H\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI40_pf32.xml\n",
      "input is shortread: URI40\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UWI263_pf32.xml\n",
      "input is shortread: UWI263\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI56_pf32.xml\n",
      "input is shortread: URI56\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI93H_pf32.xml\n",
      "input is longread: URI93H\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI86H_pf32.xml\n",
      "input is longread: URI86H\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI56H_pf32.xml\n",
      "input is longread: URI56H\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI48_pf32.xml\n",
      "input is shortread: URI48\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UNY149P_pf32.xml\n",
      "input is longread: UNY149P\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UWI283P_pf32.xml\n",
      "input is longread: UWI283P\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI47H_pf32.xml\n",
      "input is longread: URI47H\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UCT31H_pf32.xml\n",
      "input is longread: UCT31H\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI91_pf32.xml\n",
      "input is shortread: URI91\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI107H_pf32.xml\n",
      "input is longread: URI107H\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UNY172_pf32.xml\n",
      "input is shortread: UNY172\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UNY203_pf32.xml\n",
      "input is shortread: UNY203\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UCT32_pf32.xml\n",
      "input is shortread: UCT32\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI112H_pf32.xml\n",
      "input is longread: URI112H\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UWI247P_pf32.xml\n",
      "input is longread: UWI247P\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI103H_pf32.xml\n",
      "input is longread: URI103H\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UCT113_pf32.xml\n",
      "input is shortread: UCT113\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI118_pf32.xml\n",
      "input is shortread: URI118\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI87H_pf32.xml\n",
      "input is longread: URI87H\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UCT50_pf32.xml\n",
      "input is shortread: UCT50\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UCT96H_pf32.xml\n",
      "input is longread: UCT96H\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UNY203P_pf32.xml\n",
      "input is longread: UNY203P\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI118H_pf32.xml\n",
      "input is longread: URI118H\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UNY193_pf32.xml\n",
      "input is shortread: UNY193\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UNY149_pf32.xml\n",
      "input is shortread: UNY149\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UCT30H_pf32.xml\n",
      "input is longread: UCT30H\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI36H_pf32.xml\n",
      "input is longread: URI36H\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI36_pf32.xml\n",
      "input is shortread: URI36\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI42_pf32.xml\n",
      "input is shortread: URI42\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI34_pf32.xml\n",
      "input is shortread: URI34\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI101_pf32.xml\n",
      "input is shortread: URI101\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UCT110H_pf32.xml\n",
      "input is longread: UCT110H\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UCT30_pf32.xml\n",
      "input is shortread: UCT30\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UWI248_pf32.xml\n",
      "input is shortread: UWI248\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI120H_pf32.xml\n",
      "input is longread: URI120H\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI87_pf32.xml\n",
      "input is shortread: URI87\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI101H_pf32.xml\n",
      "input is longread: URI101H\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UCT96_pf32.xml\n",
      "input is shortread: UCT96\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI112_pf32.xml\n",
      "input is shortread: URI112\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI117H_pf32.xml\n",
      "input is longread: URI117H\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI117_pf32.xml\n",
      "input is shortread: URI117\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI91H_pf32.xml\n",
      "input is longread: URI91H\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI93_pf32.xml\n",
      "input is shortread: URI93\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UWI263P_pf32.xml\n",
      "input is longread: UWI263P\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI36_whole_plasmid.xml\n",
      "input is shortread: URI36\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI91H_whole_plasmid.xml\n",
      "input is longread: URI91H\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI88H_whole_plasmid.xml\n",
      "input is longread: URI88H\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI46_whole_plasmid.xml\n",
      "input is shortread: URI46\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UCT31H_whole_plasmid.xml\n",
      "input is longread: UCT31H\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UCT32_whole_plasmid.xml\n",
      "input is shortread: UCT32\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI89_whole_plasmid.xml\n",
      "input is shortread: URI89\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UCT109H_whole_plasmid.xml\n",
      "input is longread: UCT109H\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UCT109_whole_plasmid.xml\n",
      "input is shortread: UCT109\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UNY193_whole_plasmid.xml\n",
      "input is shortread: UNY193\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI44_whole_plasmid.xml\n",
      "input is shortread: URI44\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UWI283_whole_plasmid.xml\n",
      "input is shortread: UWI283\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UWI247_whole_plasmid.xml\n",
      "input is shortread: UWI247\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UNY203P_whole_plasmid.xml\n",
      "input is longread: UNY203P\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UNY193P_whole_plasmid.xml\n",
      "input is longread: UNY193P\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI36H_whole_plasmid.xml\n",
      "input is longread: URI36H\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UCT110_whole_plasmid.xml\n",
      "input is shortread: UCT110\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI42H_whole_plasmid.xml\n",
      "input is longread: URI42H\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UCT110H_whole_plasmid.xml\n",
      "input is longread: UCT110H\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI87H_whole_plasmid.xml\n",
      "input is longread: URI87H\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI39_whole_plasmid.xml\n",
      "input is shortread: URI39\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI117H_whole_plasmid.xml\n",
      "input is longread: URI117H\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UCT92H_whole_plasmid.xml\n",
      "input is longread: UCT92H\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI102_whole_plasmid.xml\n",
      "input is shortread: URI102\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UWI263_whole_plasmid.xml\n",
      "input is shortread: UWI263\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UNY203_whole_plasmid.xml\n",
      "input is shortread: UNY203\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UNY208P_whole_plasmid.xml\n",
      "input is longread: UNY208P\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UCT30H_whole_plasmid.xml\n",
      "input is longread: UCT30H\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UCT31_whole_plasmid.xml\n",
      "input is shortread: UCT31\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/ESI26H_whole_plasmid.xml\n",
      "input is longread: ESI26H\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI88_whole_plasmid.xml\n",
      "input is shortread: URI88\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UNY172P_whole_plasmid.xml\n",
      "input is longread: UNY172P\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI91_whole_plasmid.xml\n",
      "input is shortread: URI91\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UWI283P_whole_plasmid.xml\n",
      "input is longread: UWI283P\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI120H_whole_plasmid.xml\n",
      "input is longread: URI120H\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/ESI26_whole_plasmid.xml\n",
      "input is shortread: ESI26\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UCT113_whole_plasmid.xml\n",
      "input is shortread: UCT113\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UCT96_whole_plasmid.xml\n",
      "input is shortread: UCT96\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UWI263P_whole_plasmid.xml\n",
      "input is longread: UWI263P\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI107_whole_plasmid.xml\n",
      "input is shortread: URI107\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UNY149_whole_plasmid.xml\n",
      "input is shortread: UNY149\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI107H_whole_plasmid.xml\n",
      "input is longread: URI107H\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI42_whole_plasmid.xml\n",
      "input is shortread: URI42\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI101H_whole_plasmid.xml\n",
      "input is longread: URI101H\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI47H_whole_plasmid.xml\n",
      "input is longread: URI47H\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UCT29H_whole_plasmid.xml\n",
      "input is longread: UCT29H\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI86H_whole_plasmid.xml\n",
      "input is longread: URI86H\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UNY149P_whole_plasmid.xml\n",
      "input is longread: UNY149P\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI34H_whole_plasmid.xml\n",
      "input is longread: URI34H\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UCT32H_whole_plasmid.xml\n",
      "input is longread: UCT32H\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI117_whole_plasmid.xml\n",
      "input is shortread: URI117\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI89H_whole_plasmid.xml\n",
      "input is longread: URI89H\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UCT30_whole_plasmid.xml\n",
      "input is shortread: UCT30\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI103H_whole_plasmid.xml\n",
      "input is longread: URI103H\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UCT35H_whole_plasmid.xml\n",
      "input is longread: UCT35H\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI111_whole_plasmid.xml\n",
      "input is shortread: URI111\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UWI248_whole_plasmid.xml\n",
      "input is shortread: UWI248\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI46H_whole_plasmid.xml\n",
      "input is longread: URI46H\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UWI248P_whole_plasmid.xml\n",
      "input is longread: UWI248P\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI120_whole_plasmid.xml\n",
      "input is shortread: URI120\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI48H_whole_plasmid.xml\n",
      "input is longread: URI48H\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI112_whole_plasmid.xml\n",
      "input is shortread: URI112\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI93H_whole_plasmid.xml\n",
      "input is longread: URI93H\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI34_whole_plasmid.xml\n",
      "input is shortread: URI34\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI101_whole_plasmid.xml\n",
      "input is shortread: URI101\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI86_whole_plasmid.xml\n",
      "input is shortread: URI86\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UNY169_whole_plasmid.xml\n",
      "input is shortread: UNY169\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI40H_whole_plasmid.xml\n",
      "input is longread: URI40H\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI56H_whole_plasmid.xml\n",
      "input is longread: URI56H\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UNY208_whole_plasmid.xml\n",
      "input is shortread: UNY208\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UCT50H_whole_plasmid.xml\n",
      "input is longread: UCT50H\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI118_whole_plasmid.xml\n",
      "input is shortread: URI118\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI39H_whole_plasmid.xml\n",
      "input is longread: URI39H\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI56_whole_plasmid.xml\n",
      "input is shortread: URI56\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UCT92_whole_plasmid.xml\n",
      "input is shortread: UCT92\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI111H_whole_plasmid.xml\n",
      "input is longread: URI111H\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UNY172_whole_plasmid.xml\n",
      "input is shortread: UNY172\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UWI247P_whole_plasmid.xml\n",
      "input is longread: UWI247P\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI112H_whole_plasmid.xml\n",
      "input is longread: URI112H\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI87_whole_plasmid.xml\n",
      "input is shortread: URI87\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI40_whole_plasmid.xml\n",
      "input is shortread: URI40\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI102H_whole_plasmid.xml\n",
      "input is longread: URI102H\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UCT29_whole_plasmid.xml\n",
      "input is shortread: UCT29\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI41_whole_plasmid.xml\n",
      "input is shortread: URI41\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UCT96H_whole_plasmid.xml\n",
      "input is longread: UCT96H\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UCT113H_whole_plasmid.xml\n",
      "input is longread: UCT113H\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI103_whole_plasmid.xml\n",
      "input is shortread: URI103\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI33H_whole_plasmid.xml\n",
      "input is longread: URI33H\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI118H_whole_plasmid.xml\n",
      "input is longread: URI118H\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UNY169P_whole_plasmid.xml\n",
      "input is longread: UNY169P\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UCT35_whole_plasmid.xml\n",
      "input is shortread: UCT35\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI33_whole_plasmid.xml\n",
      "input is shortread: URI33\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI48_whole_plasmid.xml\n",
      "input is shortread: URI48\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI93_whole_plasmid.xml\n",
      "input is shortread: URI93\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UCT50_whole_plasmid.xml\n",
      "input is shortread: UCT50\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI44H_whole_plasmid.xml\n",
      "input is longread: URI44H\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI41H_whole_plasmid.xml\n",
      "input is longread: URI41H\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI47_whole_plasmid.xml\n",
      "input is shortread: URI47\n"
     ]
    }
   ],
   "source": [
    "if PARSE_BLAST is True:\n",
    "    for result in pf32_results:\n",
    "        print(result)\n",
    "        name = Path(result).stem.split(\"_\")[0]\n",
    "        if name.endswith(('H','P')):\n",
    "            print(f'input is longread: {name}')\n",
    "            name = name[:-1:] # chop the char for dict lookup.\n",
    "            current_asm = assembly_dict[name]['longread']\n",
    "        else:\n",
    "            print(f'input is shortread: {name}')\n",
    "            current_asm = assembly_dict[name]['shortread']\n",
    "        #print(current_asm.keys())\n",
    "        blasthits = parse_blast_results(result, \"pf\")\n",
    "        for hit in blasthits:\n",
    "            if blasthits[hit] is None:\n",
    "                print(f'No hits for {hit}')\n",
    "            else:\n",
    "                for h in blasthits[hit]:\n",
    "                    #print(h)\n",
    "                    #print(h['contig_id'])\n",
    "                    #print(current_asm[h['contig_id']])\n",
    "                    current_asm[h['contig_id']]['pf32_hits'].append(h)\n",
    "\n",
    "    for result in wp_results:\n",
    "        print(result)\n",
    "        name = Path(result).stem.split(\"_\")[0]\n",
    "        if name.endswith(('H','P')):\n",
    "            print(f'input is longread: {name}')\n",
    "            name = name[:-1:] # chop the char for dict lookup.\n",
    "            current_asm = assembly_dict[name]['longread']\n",
    "        else:\n",
    "            print(f'input is shortread: {name}')\n",
    "            current_asm = assembly_dict[name]['shortread']\n",
    "        #print(current_asm.keys())\n",
    "        blasthits = parse_blast_results(result, \"wp\")\n",
    "        for hit in blasthits:\n",
    "            if blasthits[hit] is None:\n",
    "                print(f'No hits for {hit}')\n",
    "            else:\n",
    "                for h in blasthits[hit]:\n",
    "                    #print(h)\n",
    "                    #print(h['contig_id'])\n",
    "                    #print(current_asm[h['contig_id']])\n",
    "                    current_asm[h['contig_id']]['wp_hits'].append(h)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [],
   "source": [
    "#pickle.dump(assembly_dict, open(f'{output_dir}/assembly_dict_v4_parsed.pkl', 'wb'))\n",
    "assembly_dict = pickle.load(open(f'{output_dir}/assembly_dict_v4_parsed.pkl', 'rb'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['chromosome', 'cp26', 'cp32-1', 'cp32-1+5', 'cp32-10', 'cp32-11', 'cp32-12', 'cp32-2', 'cp32-3', 'cp32-3+10', 'cp32-4', 'cp32-5', 'cp32-5+1', 'cp32-5-1', 'cp32-6', 'cp32-7', 'cp32-8', 'cp32-9', 'cp32-9-4', 'cp9', 'cp9-3', 'lp17', 'lp21', 'lp25', 'lp28-1', 'lp28-11', 'lp28-2', 'lp28-3', 'lp28-4', 'lp28-5', 'lp28-6', 'lp28-7', 'lp28-8', 'lp28-9', 'lp36', 'lp38', 'lp5', 'lp54', 'lp56']\n"
     ]
    }
   ],
   "source": [
    "# Ok let's crack open this jar of pickled plasmids.\n",
    "parsing_table = pickle.load(open(f'{parsing_tables_dir}/blast_parsing_dict.pkl', 'rb'))\n",
    "# ok now let's set up the columns for our matrix and drop the synthetic vector I added a while back...\n",
    "plasmids_in_db = {k : v for k, v in parsing_table.items()}\n",
    "matrix_cols = []\n",
    "for k in plasmids_in_db:\n",
    "    matrix_cols.append(plasmids_in_db[k]['name'])\n",
    "matrix_cols = list(set(matrix_cols))\n",
    "matrix_cols.sort()\n",
    "matrix_cols.remove('pBSV2')\n",
    "matrix_cols.remove('lp21-cp9')\n",
    "print(matrix_cols)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [],
   "source": [
    "for key in parsing_table:\n",
    "    if 'cp32' in parsing_table[key]['name']:\n",
    "        #print(parsing_table[key])\n",
    "        pass"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 112,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_hit_id(hit_title):\n",
    "    \"\"\"Extract hit id from hit title\"\"\"\n",
    "    #print(hit_title)\n",
    "    hit_id = hit_title.split(' ')[0].split(\"_\")[-1]\n",
    "    #print(f'{hit_title} => {hit_id}')\n",
    "    if \"|\" in hit_id:\n",
    "        if 'pdb' in hit_id: # I do not know why this singular db entry is different. thank you very much blast :)\n",
    "            hit_id = hit_id.split(\"|\")[-1]\n",
    "        else:\n",
    "            hit_id = hit_id.split(\"|\")[1]\n",
    "    return hit_id\n",
    "\n",
    "# This needs defaultdict() from collections.\n",
    "# I literally already have this above....... Why did I rewrite this? # nvm i remember. it needed it, dropped the redundancy above.\n",
    "def parse_hit(hit, **kwargs):\n",
    "    \"\"\"output the relevant information from a hit\n",
    "    ['hit_id', 'contig_id', 'contig_len', 'hit_title', 'hit_def', 'bitscore', 'e_score',\n",
    "    'query_start', 'query_end', 'ref_start', 'ref_end',\n",
    "    'hit_pos', 'idents', 'gaps', 'frame', 'query_seq']\n",
    "    enter a list of keys to return in the return_list kwarg to return only those keys\n",
    "    ex:\n",
    "    parse_hit(hit, return_list=['hit_id', 'contig_id', 'contig_len', 'hit_title'])\n",
    "    \"\"\"\n",
    "    output_list = []\n",
    "    if 'return_list' in kwargs.keys():\n",
    "        return_list = kwargs['return_list']\n",
    "    #print(\"inside parse_hit\")\n",
    "    #print(f\"Hit ID: {hit['hit_id']}\")\n",
    "    hit_id = get_hit_id(hit['hit_id'])\n",
    "    # I feel like I could make a simple class for this so that I could access the values in dot notation. i.e. hit.hit_id, hit.contig_id, etc.\n",
    "    hit_dict = {\n",
    "        'hit_id'  : hit_id,\n",
    "        'contig_id' : hit['contig_id'],\n",
    "        'contig_len' : hit['contig_len'],\n",
    "        'hit_title' : hit['hit_title'],\n",
    "        'hit_def' : hit['hit_def'],\n",
    "        'bitscore' : hit['hit_score'],\n",
    "        'e_score' : hit['hit_expect'],\n",
    "        'query_start' : hit['hit_query_start'],\n",
    "        'query_end' : hit['hit_query_end'],\n",
    "        'ref_start' : hit['hit_subject_start'],\n",
    "        'ref_end' : hit['hit_subject_end'],\n",
    "        'hit_pos': hit['hit_pos'],\n",
    "        'idents' : hit['hit_idents'],\n",
    "        'gaps' : hit['hit_gaps'],\n",
    "        'frame' : hit['hit_frame'],\n",
    "        'query_seq' : hit['hit_query']}\n",
    "    for key in return_list:\n",
    "        output_list.append(hit_dict[key])\n",
    "    return output_list\n",
    "\n",
    "def get_matching_feature(contig_start, contig_end, features):\n",
    "    \"\"\"if it exists, find the annotation feature present at the pf32 hit coordinates\"\"\"\n",
    "    for feature in features:\n",
    "        if feature.type == 'CDS' or feature.type == 'gene':\n",
    "            if feature.location.start - 10 <= contig_start <= feature.location.end + 10 and feature.location.start - 10 <= contig_end <= feature.location.end + 10:\n",
    "                return feature\n",
    "    return None\n",
    "\n",
    "def process_contig(contig_obj, contig_id, method, isolate, lr_name, parsing_dict, assembly_dict,cutoff):\n",
    "    \"\"\"Process a single contig\"\"\"\n",
    "    # {{TODO}}\n",
    "    # The logic is okay-ish but needs MAJOR revision and optimization for readability.\n",
    "    # I should honestly strip this down to the most atomic form to work on ONE SINGULAR INPUT FILE. THAT IS MY ISSUE. NOT THE NUMBER OF FILES BUT THE FACT THAT MY SOLUTIONS AND TOOLS\n",
    "    # ARE BUILT TO HANDLE MULTIPLE FILES AT ONCE AND THESE FILES ARE RARELY CONSISTENT. I can just work on one file at a time and then iterate over them in a loop.\n",
    "    # This is so obvious but I don't know why I haven't realized that yet.\n",
    "    # This is also basic C.S. and idk how I missed this. I'm writing complex, fragile, rigid code when I could be writing simple, robust, flexible code. This is all from the fact that\n",
    "    # I am trying to do too much at once. I need to slow down and take it one step at a time. I need to break it down into the smallest possible components and work on those and prioritize\n",
    "    # everything working together for a single input.\n",
    "    # THEN: I can include exceptions and catch faulty formatting.\n",
    "    # This should make things so much easier to manage and debug.\n",
    "    # Gosh, who even let me write code like this? I've been kneecapping myself at every step\n",
    "\n",
    "    sr_plasmids = []\n",
    "    lr_plasmids = []\n",
    "    pf_lines = ['']\n",
    "    wp_lines = ['']\n",
    "    use_wp = False\n",
    "    contig_len = len(assembly_dict[isolate][method][contig_id]['seqrecord'].seq)\n",
    "    if contig_len <= cutoff:\n",
    "        #print(f'{contig_id} is too short, len < 2500')\n",
    "        pf_lines.append(f'{contig_id}\\tlt_2500bp\\n')\n",
    "        wp_lines.append(f'{contig_id}\\tlt_2500bp\\n')\n",
    "        return sr_plasmids, lr_plasmids, pf_lines, wp_lines\n",
    "    else:\n",
    "        if contig_len > 100000:\n",
    "            #print(f'{contig_id} is prob chromosome or chromosome fragment')\n",
    "            use_wp = True\n",
    "        else:\n",
    "            if len(contig_obj.get(\"pf32_hits\")) > 0:\n",
    "                pf32_hits = contig_obj.get(\"pf32_hits\")\n",
    "                for pf32hit in pf32_hits:\n",
    "                    if method == 'shortread': # just tack on the first hit.\n",
    "                        #print(isolate)\n",
    "                        sr_plasmids.append(get_hit_id(pf32_hits[0]['hit_id']))\n",
    "                    else:\n",
    "                        #print(isolate)\n",
    "                        lr_plasmids.append(get_hit_id(pf32_hits[0]['hit_id']))\n",
    "                    # ok now let's get the stuff from our results\n",
    "                    parsed_pfhit_id, hit_contig_id, bitscore, escore,  contig_start, contig_end, ref_start, ref_end = parse_hit(pf32hit, return_list=['hit_id', 'contig_id', 'bitscore', 'e_score', 'query_start', 'query_end', 'ref_start', 'ref_end'])\n",
    "                    matching_feature = get_matching_feature(contig_start, contig_end, assembly_dict[isolate][method][contig_id]['seqrecord'].features)\n",
    "                    if matching_feature is not None:# I'm gonna have a stroke this literaelly donw't make sense. This also can probably either fit into the matching feature function\n",
    "                        if 'gene' in matching_feature.qualifiers: # or possibly be it's own. I'm leaning towards the former\n",
    "                            annotated_feat = matching_feature.qualifiers['gene']\n",
    "                        elif 'locus_tag' in matching_feature.qualifiers:\n",
    "                            annotated_feat = matching_feature.qualifiers['locus_tag']\n",
    "                        elif 'product' in matching_feature.qualifiers:\n",
    "                            annotated_feat = matching_feature.qualifiers['product']\n",
    "                        else:\n",
    "                            annotated_feat = 'not gene locus or cds'\n",
    "                        #print(f'{contig_id} {matching_feature}')\n",
    "                        annofeat_start = matching_feature.location.start\n",
    "                        annofeat_end = matching_feature.location.end\n",
    "                    else:\n",
    "                        annotated_feat = 'no annotated feature'\n",
    "                        annofeat_start = 'NA'\n",
    "                        annofeat_end = 'NA'\n",
    "                    pf_line = f'{contig_id}\\t{parsed_pfhit_id}\\t{contig_len}\\t{bitscore}\\t{escore}\\t{contig_start}\\t{contig_end}\\t{annotated_feat}\\t{annofeat_start}\\t{annofeat_end}\\n'\n",
    "                    pf_lines.append(pf_line)\n",
    "            else:\n",
    "                pf_lines.append(f'{contig_id}\\tNo pf32 hits!\\n')\n",
    "        if len(contig_obj.get(\"wp_hits\")) > 0:\n",
    "            wp_hits = contig_obj.get(\"wp_hits\")\n",
    "            for wp_hit in wp_hits:\n",
    "                #print(isolate)\n",
    "                #print(f'ID: {wp_hit[\"hit_id\"]}, TITLE: {wp_hit[\"hit_title\"]}, DEF: {wp_hit[\"hit_def\"]}')\n",
    "                wp_hit_id = get_hit_id(wp_hit['hit_id'])\n",
    "                wp_hit_name = parsing_dict[wp_hit_id]['name']\n",
    "                parsed_wphit_id, wphit_contig_id, bitscore, escore, query_start, query_end, ref_start, ref_end= parse_hit(wp_hit, return_list=['hit_id', 'contig_id', 'bitscore', 'e_score', 'query_start', 'query_end', 'ref_start', 'ref_end'])\n",
    "                expected_length = parsing_dict[wp_hit_id]['length']\n",
    "                wp_line = f'{contig_id}\\t{parsed_wphit_id}\\t{wp_hit_name}\\t{contig_len}\\t{expected_length}\\t{bitscore}\\t{escore}\\t{query_start}\\t{query_end}\\t{ref_start}\\t{ref_end}\\n'\n",
    "                wp_lines.append(wp_line)\n",
    "            if use_wp is True or len(contig_obj.get(\"pf32_hits\")) == 0:\n",
    "                if method == 'shortread':\n",
    "                    #print(isolate)\n",
    "                    sr_plasmids.append(parsing_dict[get_hit_id(wp_hits[0]['hit_id'])]['name'])\n",
    "                else:\n",
    "                    #print(isolate)\n",
    "                    lr_plasmids.append(parsing_dict[get_hit_id(wp_hits[0]['hit_id'])]['name'])\n",
    "        else:\n",
    "            wp_lines.append(f'{contig_id}\\tNo whole plasmid hits!\\n')\n",
    "    return sr_plasmids, lr_plasmids, pf_lines, wp_lines\n",
    "\n",
    "def process_isolate(isolate, assembly_dict, output_dir, method, parsing_dict,cutoff):\n",
    "    \"\"\"Process a single isolate\"\"\"\n",
    "    pf_header = 'contig_id\\tparsed_pfhit_id\\tcontig_len\\tbitscore\\tescore\\tquery_start\\tquery_end\\tannotated_gene\\tgene_start\\tgenet_end\\n'\n",
    "    wp_header = 'contig_id\\tparsed_wphit_id\\twp_hit_name\\tcontig_len\\texpected_length\\tbitscore\\tescore\\tquery_start\\tquery_end\\tref_start\\tref_end\\n'\n",
    "    lr_name = assembly_dict[isolate]['longread_name']\n",
    "    pfam_lines = []\n",
    "    wpl_lines = []\n",
    "    sr_hit_list = []\n",
    "    lr_hit_list = []\n",
    "    pfam_lines.insert(0, pf_header)\n",
    "    wpl_lines.insert(0, wp_header)\n",
    "    if method != 'longread_name':\n",
    "        if method == 'shortread':\n",
    "            iso_name = isolate\n",
    "        else:\n",
    "            iso_name = lr_name\n",
    "        if os.path.exists(f'{output_dir}/plasmid_hits/') is False:\n",
    "            os.mkdir(f'{output_dir}/plasmid_hits/')\n",
    "        with open(f'{output_dir}/plasmid_hits/{iso_name}_plasmid_hits.tsv', 'w') as outfile:\n",
    "            for contig in assembly_dict[isolate][method]:\n",
    "                if contig != 'seqrecord':\n",
    "                    contig_obj = assembly_dict[isolate][method][contig]\n",
    "                    #print(f'processing {contig}')\n",
    "                    sr_hits, lr_hits, pf_lines, wp_lines = process_contig(contig_obj, contig, method, isolate, lr_name, parsing_dict, assembly_dict,cutoff)\n",
    "                    pfam_lines.extend(pf_lines)\n",
    "                    wpl_lines.extend(wp_lines)\n",
    "                    sr_hit_list.extend(sr_hits)\n",
    "                    lr_hit_list.extend(lr_hits)\n",
    "            outfile.writelines(pfam_lines)\n",
    "            outfile.writelines(wpl_lines)\n",
    "    return sr_hit_list, lr_hit_list\n",
    "\n",
    "def process_assembly_dict(assembly_dict, output_dir, parsing_dict,cutoff):\n",
    "    \"\"\"Process the assembly dict\"\"\"\n",
    "    sr_plasmids = defaultdict(dict)\n",
    "    lr_plasmids = defaultdict(dict)\n",
    "    for isolate in assembly_dict:\n",
    "        sr_plasmids[isolate] = []\n",
    "        for method in assembly_dict[isolate]:\n",
    "            sr, lr = process_isolate(isolate, assembly_dict, output_dir, method, parsing_dict,cutoff)\n",
    "            if method == 'longread':\n",
    "                iso_name = assembly_dict[isolate]['longread_name']\n",
    "                lr_plasmids[iso_name] = []\n",
    "                lr_plasmids[iso_name].extend(lr)\n",
    "            else:\n",
    "                sr_plasmids[isolate].extend(sr)\n",
    "            #print(f'added lr plasmids for {isolate}')\n",
    "    return sr_plasmids, lr_plasmids\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 113,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "longread: URI47H : len:19 : ['cp32-12', 'cp32-4', 'lp28-6', 'cp9', 'cp26', 'cp32-7', 'cp32-6', 'cp32-3', 'lp54', 'lp28-1', 'cp32-11', 'lp28-3', 'lp17', 'lp28-4', 'cp32-9', 'cp32-5', 'chromosome', 'cp32-8', 'lp38']\n",
      "shortread: URI47 : len: 23['lp25', 'cp32-12', 'lp21', 'lp28-6', 'cp32-2', 'cp9', 'cp26', 'cp32-7', 'cp32-6', 'cp32-3', 'lp54', 'lp28-1', 'cp32-11', 'lp28-5', 'lp28-3', 'lp17', 'lp28-4', 'cp32-9', 'cp32-5', 'chromosome', 'lp36', 'cp32-8', 'lp38']\n",
      "longread: URI103H : len:18 : ['cp32-7', 'cp32-6', 'cp32-9', 'cp32-5', 'cp32-12', 'chromosome', 'lp25', 'lp28-1', 'cp32-3', 'cp32-4', 'lp38', 'lp28-6', 'cp32-11', 'lp17', 'lp28-3', 'lp28-4', 'lp54', 'cp26']\n",
      "shortread: URI103 : len: 21['lp25', 'cp32-12', 'lp21', 'lp28-6', 'cp32-2', 'cp26', 'cp32-7', 'cp32-6', 'cp32-3', 'lp54', 'lp28-1', 'cp32-11', 'lp28-5', 'lp28-3', 'lp17', 'lp28-4', 'cp32-9', 'cp32-5', 'chromosome', 'lp36', 'lp38']\n",
      "longread: UCT31H : len:21 : ['lp25', 'lp21', 'cp32-4', 'cp9', 'cp32-1', 'cp26', 'cp32-7', 'cp32-6', 'lp28-2', 'cp32-3', 'lp54', 'lp28-1', 'cp32-10', 'cp32-11', 'lp28-3', 'lp28-4', 'lp17', 'cp32-9', 'cp32-5', 'chromosome', 'lp36']\n",
      "shortread: UCT31 : len: 26['lp25', 'cp32-12', 'lp21', 'cp32-4', 'cp32-2', 'cp9', 'cp32-1', 'cp26', 'lp56', 'cp32-7', 'cp32-6', 'lp28-2', 'cp32-3', 'lp54', 'lp28-1', 'cp9-3', 'cp32-11', 'lp28-3', 'lp17', 'lp28-4', 'cp32-9-4', 'cp32-9', 'cp32-5', 'chromosome', 'lp36', 'cp32-8']\n",
      "longread: UCT92H : len:21 : ['cp32-12', 'cp32-4', 'lp28-6', 'cp32-1', 'cp26', 'cp32-7', 'cp32-6', 'cp32-3', 'lp54', 'lp28-1', 'cp32-11', 'lp28-5', 'lp28-3', 'lp28-4', 'lp17', 'cp32-9', 'cp32-5', 'chromosome', 'lp36', 'cp32-8', 'lp38']\n",
      "shortread: UCT92 : len: 23['lp25', 'cp32-12', 'lp21', 'lp28-6', 'cp32-2', 'cp26', 'cp32-6', 'cp32-3', 'lp54', 'lp28-1', 'cp9-3', 'cp32-11', 'lp28-5', 'lp28-11', 'lp28-3', 'lp17', 'lp28-4', 'cp32-9', 'cp32-5', 'chromosome', 'lp36', 'cp32-8', 'lp38']\n",
      "longread: URI46H : len:19 : ['lp25', 'cp32-12', 'lp21', 'cp32-4', 'lp28-6', 'cp26', 'cp32-7', 'cp32-6', 'cp32-3', 'lp54', 'lp28-1', 'cp32-11', 'lp28-3', 'lp17', 'lp28-4', 'cp32-9', 'cp32-5', 'chromosome', 'lp38']\n",
      "shortread: URI46 : len: 21['lp25', 'cp32-12', 'lp21', 'lp28-6', 'cp32-2', 'cp26', 'cp32-6', 'cp32-3', 'lp54', 'lp28-1', 'cp9-3', 'cp32-11', 'lp28-5', 'lp28-3', 'lp17', 'lp28-4', 'cp32-9', 'cp32-5', 'chromosome', 'lp36', 'lp38']\n",
      "longread: UWI247P : len:17 : ['cp32-7', 'cp32-6', 'lp28-1', 'lp25', 'chromosome', 'cp32-3', 'lp54', 'cp32-4', 'lp36', 'cp32-11', 'lp28-4', 'lp28-5', 'lp28-3', 'lp17', 'lp38', 'cp32-1', 'cp26']\n",
      "shortread: UWI247 : len: 19['lp25', 'cp32-12', 'cp32-4', 'cp32-2', 'cp32-1', 'cp26', 'cp32-7', 'cp32-6', 'cp32-3', 'lp54', 'lp28-1', 'cp32-11', 'lp28-5', 'lp28-3', 'lp28-4', 'lp17', 'chromosome', 'lp36', 'lp38']\n",
      "longread: UCT30H : len:19 : ['cp32-12', 'lp21', 'cp32-4', 'lp28-6', 'cp26', 'cp32-7', 'cp32-6', 'cp32-3', 'lp54', 'lp28-1', 'cp32-11', 'lp28-3', 'lp17', 'lp28-4', 'cp32-9', 'cp32-5', 'chromosome', 'lp36', 'lp38']\n",
      "shortread: UCT30 : len: 21['lp25', 'cp32-12', 'lp21', 'lp28-6', 'cp32-2', 'cp26', 'cp32-7', 'cp32-6', 'cp32-3', 'lp54', 'lp28-1', 'cp9-3', 'cp32-11', 'lp28-5', 'lp28-3', 'lp17', 'lp28-4', 'cp32-9', 'cp32-5', 'chromosome', 'lp38']\n",
      "longread: URI44H : len:20 : ['lp25', 'cp32-12', 'cp32-4', 'lp28-6', 'cp26', 'cp32-7', 'cp32-6', 'cp32-3', 'lp54', 'lp28-1', 'cp32-11', 'lp28-5', 'lp28-3', 'lp17', 'lp28-4', 'cp32-9', 'cp32-5', 'chromosome', 'lp36', 'lp38']\n",
      "shortread: URI44 : len: 19['lp25', 'lp21', 'lp28-6', 'cp32-2', 'cp26', 'cp32-6', 'cp32-3', 'lp54', 'lp28-1', 'cp32-11', 'lp28-5', 'lp28-3', 'lp17', 'lp28-4', 'cp32-9', 'cp32-5', 'chromosome', 'lp36', 'lp38']\n",
      "longread: URI87H : len:20 : ['lp25', 'cp32-4', 'cp26', 'lp56', 'cp32-7', 'cp32-6', 'lp28-2', 'cp32-3', 'lp54', 'lp28-1', 'cp32-10', 'cp32-11', 'lp28-3', 'lp17', 'lp28-4', 'cp32-9', 'cp32-5', 'chromosome', 'lp36', 'lp38']\n",
      "shortread: URI87 : len: 21['lp25', 'cp32-4', 'cp32-1', 'cp26', 'lp56', 'cp32-7', 'cp32-6', 'lp28-2', 'cp32-3', 'lp54', 'lp28-1', 'cp32-11', 'lp28-3', 'lp17', 'lp28-4', 'cp32-9-4', 'cp32-9', 'cp32-5', 'chromosome', 'lp36', 'lp38']\n",
      "longread: UCT50H : len:15 : ['cp32-7', 'cp32-9', 'cp32-5', 'cp32-12', 'chromosome', 'lp25', 'lp28-2', 'cp32-10', 'cp9', 'lp36', 'lp38', 'cp32-4', 'lp17', 'lp54', 'cp26']\n",
      "shortread: UCT50 : len: 18['cp32-7', 'cp32-9', 'lp25', 'cp32-12', 'chromosome', 'cp32-10', 'lp28-2', 'cp9', 'cp32-5', 'lp36', 'lp38', 'cp32-4', 'lp28-4', 'cp32-11', 'cp32-8', 'lp17', 'lp54', 'cp26']\n",
      "longread: UWI283P : len:16 : ['cp32-9', 'cp32-5', 'cp32-12', 'chromosome', 'lp25', 'lp28-2', 'cp32-3', 'cp32-4', 'lp36', 'lp28-4', 'lp28-5', 'lp28-3', 'lp17', 'lp54', 'cp32-1', 'cp26']\n",
      "shortread: UWI283 : len: 20['lp25', 'cp32-12', 'cp32-4', 'lp28-6', 'cp32-1', 'cp26', 'lp56', 'cp32-7', 'lp28-2', 'cp32-3', 'lp54', 'cp32-10', 'lp28-5', 'lp28-3', 'lp17', 'lp28-4', 'cp32-9', 'cp32-5', 'chromosome', 'lp36']\n",
      "longread: URI107H : len:19 : ['lp25', 'cp32-4', 'cp26', 'cp32-7', 'cp32-6', 'lp28-2', 'cp32-3', 'lp54', 'lp28-1', 'cp32-10', 'cp32-11', 'lp28-3', 'lp17', 'lp28-4', 'cp32-9', 'cp32-5', 'chromosome', 'lp36', 'lp38']\n",
      "shortread: URI107 : len: 21['lp25', 'cp32-4', 'cp32-2', 'cp26', 'lp56', 'cp32-7', 'cp32-6', 'lp28-2', 'cp32-3', 'lp54', 'lp28-1', 'cp32-11', 'lp28-3', 'lp17', 'lp28-4', 'cp32-9-4', 'cp32-9', 'cp32-5', 'chromosome', 'lp36', 'lp38']\n",
      "longread: UCT110H : len:21 : ['lp25', 'cp32-12', 'lp21', 'cp32-4', 'lp28-6', 'cp26', 'cp32-7', 'cp32-6', 'cp32-3', 'lp54', 'lp28-1', 'cp32-11', 'lp28-5', 'lp28-3', 'lp17', 'lp28-4', 'cp32-9', 'cp32-5', 'chromosome', 'cp32-8', 'lp38']\n",
      "shortread: UCT110 : len: 22['lp25', 'cp32-4', 'lp28-6', 'cp32-2', 'cp26', 'cp32-7', 'cp32-6', 'cp32-3', 'lp54', 'lp28-1', 'cp32-5+1', 'cp32-11', 'lp28-5', 'lp28-3', 'lp17', 'lp28-4', 'cp32-9', 'cp32-5', 'chromosome', 'lp36', 'cp32-8', 'lp38']\n",
      "longread: URI48H : len:16 : ['cp32-6', 'lp25', 'cp32-5', 'chromosome', 'cp32-10', 'cp9-3', 'lp28-7', 'lp36', 'lp38', 'cp32-11', 'lp17', 'cp32-8', 'lp28-3', 'lp28-4', 'lp54', 'cp26']\n",
      "shortread: URI48 : len: 19['lp25', 'lp28-7', 'cp32-4', 'cp26', 'cp32-6', 'cp32-3', 'lp28-8', 'lp54', 'cp9-3', 'cp32-11', 'lp28-3', 'lp17', 'lp28-4', 'cp32-9', 'cp32-5', 'chromosome', 'lp36', 'cp32-8', 'lp38']\n",
      "longread: UNY203P : len:17 : ['cp32-7', 'cp32-6', 'lp28-1', 'cp32-5', 'cp32-12', 'chromosome', 'cp32-10', 'cp9-3', 'lp54', 'cp32-4', 'lp28-6', 'lp36', 'lp28-3', 'lp17', 'lp38', 'cp26', 'lp5']\n",
      "shortread: UNY203 : len: 22['lp25', 'cp32-12', 'lp28-6', 'cp26', 'lp5', 'cp32-7', 'cp32-6', 'cp32-3', 'lp54', 'cp32-13', 'lp28-1', 'cp32-10', 'cp9-3', 'cp32-11', 'lp28-11', 'lp28-3', 'lp28-4', 'lp17', 'cp32-5', 'chromosome', 'lp36', 'lp38']\n",
      "longread: UWI263P : len:19 : ['lp25', 'cp9', 'cp26', 'cp32-6', 'cp32-3', 'lp54', 'lp28-1', 'cp32-10', 'cp32-11', 'lp28-5', 'lp28-3', 'lp17', 'lp28-4', 'cp32-9', 'cp32-5', 'chromosome', 'lp36', 'cp32-8', 'lp38']\n",
      "shortread: UWI263 : len: 20['lp25', 'cp32-12', 'cp9', 'cp32-1', 'cp26', 'cp32-7', 'cp32-6', 'cp32-1+5', 'lp54', 'lp28-1', 'cp32-11', 'lp28-5', 'lp28-3', 'lp17', 'lp28-4', 'cp32-9', 'cp32-5', 'chromosome', 'lp36', 'lp38']\n",
      "longread: URI117H : len:20 : ['lp25', 'cp32-12', 'lp21', 'cp32-4', 'lp28-6', 'cp26', 'cp32-7', 'cp32-6', 'cp32-3', 'lp54', 'lp28-1', 'cp32-11', 'lp28-5', 'lp28-3', 'lp17', 'lp28-4', 'cp32-9', 'chromosome', 'lp36', 'lp38']\n",
      "shortread: URI117 : len: 16['cp32-6', 'lp28-1', 'lp25', 'cp32-9', 'chromosome', 'cp32-5', 'lp21', 'lp36', 'lp38', 'lp28-6', 'lp28-4', 'lp28-5', 'lp28-3', 'lp17', 'lp54', 'cp26']\n",
      "longread: UCT35H : len:15 : ['cp32-6', 'cp32-13', 'lp28-1', 'cp32-5', 'cp32-12', 'chromosome', 'cp9-3', 'cp32-10', 'cp32-3', 'lp38', 'lp17', 'lp28-4', 'lp54', 'cp26', 'lp5']\n",
      "shortread: UCT35 : len: 21['lp25', 'cp32-12', 'cp32-4', 'cp26', 'lp5', 'cp32-7', 'cp32-6', 'cp32-3', 'lp54', 'lp28-1', 'cp32-10', 'cp9-3', 'cp32-11', 'lp28-11', 'lp28-3', 'lp17', 'lp28-4', 'cp32-5', 'chromosome', 'lp36', 'lp38']\n",
      "longread: URI111H : len:19 : ['lp25', 'lp21', 'cp32-4', 'cp26', 'cp32-7', 'cp32-6', 'lp28-2', 'lp54', 'lp28-1', 'cp32-10', 'cp32-11', 'lp28-3', 'lp17', 'lp28-4', 'cp32-9', 'cp32-5', 'chromosome', 'lp36', 'lp38']\n",
      "shortread: URI111 : len: 21['lp25', 'cp32-12', 'lp21', 'cp32-4', 'cp32-2', 'cp26', 'lp56', 'cp32-7', 'lp28-2', 'cp32-3', 'lp54', 'lp28-1', 'cp9-3', 'lp28-3', 'lp17', 'lp28-4', 'cp32-9', 'cp32-5', 'chromosome', 'lp36', 'lp38']\n",
      "longread: URI33H : len:17 : ['cp32-6', 'lp25', 'cp32-5', 'chromosome', 'cp32-10', 'lp28-7', 'lp21', 'lp36', 'lp38', 'lp17', 'lp28-8', 'cp32-11', 'cp32-8', 'lp28-3', 'lp28-4', 'lp54', 'cp26']\n",
      "shortread: URI33 : len: 23['lp25', 'cp32-12', 'lp28-7', 'lp21', 'cp32-4', 'cp32-2', 'cp26', 'cp32-7', 'cp32-6', 'cp32-1+5', 'lp28-8', 'lp54', 'cp9-3', 'cp32-11', 'lp28-5', 'lp28-3', 'lp17', 'lp28-4', 'cp32-5', 'chromosome', 'lp36', 'cp32-8', 'lp38']\n",
      "longread: URI86H : len:17 : ['cp32-7', 'cp32-6', 'cp32-9', 'lp25', 'lp28-1', 'chromosome', 'cp32-5', 'cp32-12', 'cp32-4', 'lp38', 'lp36', 'cp32-11', 'lp28-6', 'lp17', 'lp28-4', 'lp54', 'cp26']\n",
      "shortread: URI86 : len: 19['lp25', 'cp32-12', 'lp21', 'lp28-6', 'cp32-2', 'cp26', 'cp32-7', 'cp32-6', 'lp54', 'lp28-1', 'cp32-11', 'lp28-5', 'lp28-3', 'lp17', 'lp28-4', 'cp32-9', 'chromosome', 'lp36', 'lp38']\n",
      "longread: UNY149P : len:19 : ['lp25', 'cp32-4', 'lp28-6', 'cp26', 'cp32-7', 'cp32-6', 'cp32-3', 'lp54', 'lp28-1', 'cp32-11', 'lp28-5', 'lp28-3', 'lp28-4', 'lp17', 'cp32-9', 'cp32-5', 'chromosome', 'lp36', 'lp38']\n",
      "shortread: UNY149 : len: 22['lp25', 'cp32-12', 'lp21', 'cp32-4', 'lp28-6', 'cp32-2', 'cp26', 'cp32-7', 'cp32-6', 'cp32-3', 'lp54', 'lp28-1', 'cp32-11', 'lp28-5', 'lp28-3', 'lp28-4', 'cp32-9', 'cp32-5', 'chromosome', 'lp36', 'cp32-8', 'lp38']\n",
      "longread: UCT96H : len:20 : ['lp25', 'cp32-4', 'cp32-1', 'cp26', 'cp32-7', 'cp32-6', 'lp28-2', 'cp32-3', 'lp54', 'lp28-1', 'cp32-11', 'lp28-5', 'lp28-3', 'lp17', 'lp28-4', 'cp32-9', 'chromosome', 'lp36', 'cp32-8', 'lp38']\n",
      "shortread: UCT96 : len: 15['lp28-1', 'lp25', 'cp32-9', 'chromosome', 'lp28-2', 'lp36', 'lp38', 'lp28-4', 'lp28-5', 'cp32-8', 'lp28-3', 'lp17', 'lp54', 'cp32-1', 'cp26']\n",
      "longread: URI56H : len:17 : ['cp32-7', 'cp32-6', 'cp32-5', 'chromosome', 'cp9-3', 'cp32-3', 'lp21', 'lp36', 'lp38', 'lp28-6', 'lp17', 'cp32-8', 'lp28-3', 'lp28-4', 'lp54', 'cp26', 'lp56']\n",
      "shortread: URI56 : len: 19['lp25', 'lp21', 'lp28-6', 'cp26', 'lp56', 'cp32-7', 'cp32-6', 'cp32-3', 'lp54', 'lp28-1', 'cp9-3', 'lp28-3', 'lp17', 'lp28-4', 'cp32-5', 'chromosome', 'lp36', 'cp32-8', 'lp38']\n",
      "longread: URI118H : len:18 : ['cp32-7', 'cp32-6', 'cp32-9', 'cp32-5', 'lp28-1', 'chromosome', 'cp32-12', 'cp32-3', 'cp32-4', 'lp28-6', 'lp36', 'lp17', 'cp32-11', 'lp28-5', 'lp28-3', 'lp28-4', 'lp54', 'cp26']\n",
      "shortread: URI118 : len: 14['cp32-6', 'lp28-1', 'lp25', 'cp32-9', 'chromosome', 'cp32-5', 'lp36', 'lp28-6', 'lp28-4', 'lp28-5', 'lp28-3', 'lp17', 'lp54', 'cp26']\n",
      "longread: UNY172P : len:16 : ['cp32-7', 'cp32-9', 'cp32-5', 'cp32-12', 'chromosome', 'lp25', 'lp28-2', 'cp32-3', 'lp54', 'cp32-4', 'lp36', 'lp28-4', 'lp28-3', 'lp17', 'lp38', 'cp26']\n",
      "shortread: UNY172 : len: 17['cp32-7', 'cp32-9', 'lp25', 'cp32-12', 'cp32-10', 'cp32-5', 'lp28-2', 'chromosome', 'cp32-3', 'cp32-4', 'lp36', 'lp38', 'lp17', 'lp28-3', 'lp28-4', 'lp54', 'cp26']\n",
      "longread: URI112H : len:19 : ['lp25', 'cp32-12', 'lp21', 'cp32-4', 'lp28-6', 'cp26', 'cp32-7', 'cp32-6', 'cp32-3', 'lp54', 'lp28-1', 'cp32-11', 'lp17', 'lp28-4', 'cp32-9', 'cp32-5', 'chromosome', 'cp32-8', 'lp38']\n",
      "shortread: URI112 : len: 22['lp25', 'cp32-12', 'lp21', 'lp28-6', 'cp32-2', 'cp26', 'cp32-6', 'cp32-3', 'lp54', 'lp28-1', 'cp9-3', 'cp32-11', 'lp28-5', 'lp28-3', 'lp17', 'lp28-4', 'cp32-9', 'cp32-5', 'chromosome', 'lp36', 'cp32-8', 'lp38']\n",
      "longread: UNY208P : len:18 : ['cp32-7', 'lp28-1', 'cp32-5', 'cp32-12', 'chromosome', 'lp25', 'cp32-3', 'cp32-4', 'lp28-6', 'lp36', 'cp32-11', 'lp28-4', 'cp32-8', 'lp28-3', 'lp17', 'lp54', 'cp32-1', 'cp26']\n",
      "shortread: UNY208 : len: 18['cp32-7', 'lp28-1', 'lp25', 'cp32-12', 'chromosome', 'cp32-5', 'cp32-3', 'lp36', 'lp28-6', 'cp32-4', 'lp17', 'cp32-11', 'cp32-8', 'lp28-3', 'lp28-4', 'lp54', 'cp32-1', 'cp26']\n",
      "longread: URI89H : len:22 : ['lp25', 'lp21', 'cp32-4', 'cp32-1', 'cp26', 'lp56', 'cp32-7', 'cp32-6', 'lp28-2', 'cp32-3', 'lp54', 'lp28-1', 'cp32-11', 'lp28-3', 'lp28-4', 'lp17', 'cp32-9', 'cp32-5', 'chromosome', 'lp36', 'cp32-8', 'lp38']\n",
      "shortread: URI89 : len: 22['lp25', 'cp32-12', 'lp21', 'cp32-4', 'cp32-1', 'cp26', 'lp56', 'cp32-7', 'cp32-6', 'lp28-2', 'cp32-3', 'lp54', 'lp28-1', 'cp9-3', 'lp28-3', 'lp17', 'lp28-4', 'cp32-9', 'chromosome', 'lp36', 'cp32-8', 'lp38']\n",
      "longread: URI102H : len:17 : ['cp32-7', 'cp32-9', 'lp25', 'cp32-5', 'chromosome', 'cp32-10', 'lp28-2', 'lp28-1', 'cp32-3', 'lp36', 'lp38', 'cp32-4', 'lp28-3', 'lp17', 'lp54', 'cp26', 'lp56']\n",
      "shortread: URI102 : len: 20['lp25', 'cp32-4', 'cp32-2', 'cp26', 'lp56', 'cp32-7', 'lp28-2', 'cp32-3', 'lp54', 'lp28-1', 'cp32-11', 'lp28-3', 'lp17', 'lp28-4', 'cp32-9-4', 'cp32-9', 'cp32-5', 'chromosome', 'lp36', 'lp38']\n",
      "longread: URI36H : len:21 : ['lp25', 'lp21', 'cp32-4', 'lp28-6', 'cp26', 'lp56', 'cp32-7', 'cp32-6', 'cp32-3', 'lp54', 'lp28-1', 'cp9-3', 'cp32-11', 'lp28-3', 'lp28-4', 'lp17', 'cp32-5', 'chromosome', 'lp36', 'cp32-8', 'lp38']\n",
      "shortread: URI36 : len: 20['lp25', 'lp21', 'lp28-6', 'cp26', 'lp56', 'cp32-7', 'cp32-6', 'cp32-3', 'lp54', 'lp28-1', 'cp9-3', 'cp32-11', 'lp28-5', 'lp28-3', 'lp17', 'lp28-4', 'cp32-5', 'chromosome', 'lp36', 'lp38']\n",
      "longread: UWI248P : len:18 : ['cp32-7', 'cp32-6', 'lp28-1', 'lp25', 'cp32-12', 'chromosome', 'cp32-3', 'lp54', 'cp32-4', 'lp36', 'cp32-11', 'lp28-4', 'lp28-5', 'lp28-3', 'lp17', 'lp38', 'cp32-1', 'cp26']\n",
      "shortread: UWI248 : len: 19['lp25', 'cp32-12', 'cp32-4', 'cp32-2', 'cp32-1', 'cp26', 'cp32-7', 'cp32-6', 'cp32-3', 'lp54', 'lp28-1', 'cp32-11', 'lp28-5', 'lp28-3', 'lp17', 'lp28-4', 'chromosome', 'lp36', 'lp38']\n",
      "longread: ESI26H : len:13 : ['cp32-6', 'lp28-9', 'cp32-9', 'cp32-12', 'chromosome', 'cp32-10', 'lp36', 'lp17', 'lp28-3', 'lp28-4', 'lp54', 'cp32-1', 'cp26']\n",
      "shortread: ESI26 : len: 17['cp32-6', 'lp28-9', 'cp32-9', 'cp32-5', 'cp32-12', 'chromosome', 'cp32-10', 'lp28-2', 'lp28-7', 'lp36', 'lp17', 'cp32-8', 'lp28-3', 'lp28-4', 'lp54', 'cp32-1', 'cp26']\n",
      "longread: URI40H : len:21 : ['lp25', 'cp32-4', 'cp26', 'lp56', 'cp32-7', 'cp32-6', 'lp28-2', 'cp32-3', 'lp54', 'lp28-1', 'cp32-10', 'cp9-3', 'cp32-11', 'lp28-3', 'lp17', 'lp28-4', 'cp32-9', 'cp32-5', 'chromosome', 'lp36', 'lp38']\n",
      "shortread: URI40 : len: 24['lp25', 'cp32-12', 'lp21', 'cp32-4', 'cp32-2', 'cp9', 'cp26', 'lp56', 'cp32-7', 'cp32-6', 'lp28-2', 'cp32-3', 'lp54', 'lp28-1', 'cp9-3', 'cp32-11', 'lp28-3', 'lp17', 'lp28-4', 'cp32-9', 'cp32-5', 'chromosome', 'lp36', 'lp38']\n",
      "longread: URI39H : len:21 : ['lp25', 'lp21', 'cp32-4', 'cp26', 'cp32-7', 'cp32-6', 'lp28-2', 'cp32-3', 'lp54', 'lp28-1', 'cp32-10', 'cp32-11', 'lp28-3', 'lp28-4', 'lp17', 'cp32-9', 'cp32-5', 'chromosome', 'lp36', 'cp32-8', 'lp38']\n",
      "shortread: URI39 : len: 22['lp25', 'lp21', 'cp32-4', 'cp26', 'lp56', 'cp32-7', 'cp32-6', 'lp28-2', 'cp32-3', 'lp54', 'lp28-1', 'cp9-3', 'cp32-11', 'lp28-3', 'lp17', 'lp28-4', 'cp32-9', 'cp32-5', 'chromosome', 'lp36', 'cp32-8', 'lp38']\n",
      "longread: UNY169P : len:17 : ['cp32-7', 'cp32-9', 'cp32-5', 'cp32-12', 'chromosome', 'cp32-10', 'lp25', 'lp28-2', 'lp54', 'cp32-4', 'lp36', 'lp17', 'lp28-4', 'lp28-3', 'cp9', 'lp38', 'cp26']\n",
      "shortread: UNY169 : len: 17['cp32-7', 'cp32-9', 'lp25', 'cp32-12', 'chromosome', 'cp9', 'lp28-2', 'cp32-10', 'cp32-5', 'lp36', 'lp38', 'cp32-4', 'lp17', 'lp28-3', 'lp28-4', 'lp54', 'cp26']\n",
      "longread: UCT113H : len:16 : ['cp32-7', 'lp28-1', 'lp25', 'chromosome', 'lp21', 'cp32-4', 'lp28-6', 'lp38', 'lp36', 'lp28-4', 'cp32-11', 'lp28-5', 'lp28-3', 'lp17', 'lp54', 'cp26']\n",
      "shortread: UCT113 : len: 21['lp25', 'lp21', 'cp32-4', 'lp28-6', 'cp32-1', 'cp26', 'cp32-7', 'cp32-6', 'cp32-3', 'lp54', 'lp28-1', 'cp32-11', 'lp28-5', 'lp28-3', 'lp17', 'lp28-4', 'cp32-9', 'chromosome', 'lp36', 'cp32-8', 'lp38']\n",
      "longread: URI101H : len:21 : ['lp25', 'lp21', 'cp32-4', 'cp26', 'cp32-7', 'cp32-6', 'lp28-2', 'cp32-3', 'lp54', 'lp28-1', 'cp32-10', 'cp32-11', 'lp28-5', 'lp28-3', 'lp28-4', 'lp17', 'cp32-9', 'cp32-5', 'chromosome', 'lp36', 'lp38']\n",
      "shortread: URI101 : len: 23['lp25', 'lp21', 'cp32-4', 'cp32-2', 'cp26', 'lp56', 'cp32-7', 'cp32-6', 'lp28-2', 'cp32-3', 'lp54', 'lp28-1', 'cp9-3', 'cp32-11', 'lp28-5', 'lp28-3', 'lp28-4', 'lp17', 'cp32-9', 'cp32-5', 'chromosome', 'lp36', 'lp38']\n",
      "longread: URI42H : len:18 : ['cp32-7', 'cp32-6', 'lp28-1', 'lp25', 'cp32-9', 'chromosome', 'cp32-10', 'lp28-2', 'cp32-5', 'cp32-3', 'lp36', 'cp32-4', 'cp32-11', 'lp28-3', 'lp17', 'lp54', 'cp26', 'lp56']\n",
      "shortread: URI42 : len: 19['lp25', 'cp32-12', 'cp32-2', 'cp26', 'lp56', 'cp32-7', 'cp32-6', 'lp28-2', 'cp32-3', 'lp54', 'lp28-1', 'cp32-11', 'lp28-3', 'lp17', 'lp28-4', 'cp32-9', 'cp32-5', 'chromosome', 'lp36']\n",
      "longread: UCT109H : len:17 : ['cp32-7', 'cp32-6', 'lp28-1', 'lp25', 'cp32-5', 'chromosome', 'cp32-10', 'lp28-2', 'cp32-9', 'cp32-3', 'lp36', 'lp38', 'cp32-4', 'lp28-3', 'lp17', 'lp54', 'cp26']\n",
      "shortread: UCT109 : len: 21['lp25', 'cp32-4', 'cp32-2', 'cp26', 'lp56', 'cp32-7', 'cp32-6', 'lp28-2', 'cp32-3', 'lp54', 'lp28-1', 'cp32-11', 'lp28-3', 'lp17', 'lp28-4', 'cp32-9-4', 'cp32-9', 'cp32-5', 'chromosome', 'lp36', 'lp38']\n",
      "longread: UCT32H : len:20 : ['lp25', 'cp32-12', 'cp32-4', 'lp28-6', 'cp26', 'cp32-7', 'cp32-6', 'cp32-3', 'lp54', 'lp28-1', 'cp32-11', 'lp28-5', 'lp28-3', 'lp17', 'lp28-4', 'cp32-9', 'cp32-5', 'chromosome', 'cp32-8', 'lp38']\n",
      "shortread: UCT32 : len: 22['lp25', 'cp32-12', 'lp21', 'lp28-6', 'cp32-2', 'cp26', 'cp32-7', 'cp32-6', 'cp32-3', 'lp54', 'lp28-1', 'cp32-11', 'lp28-5', 'lp28-3', 'lp17', 'lp28-4', 'cp32-9', 'cp32-5', 'chromosome', 'lp36', 'cp32-8', 'lp38']\n",
      "longread: URI41H : len:17 : ['cp32-7', 'cp32-6', 'lp28-1', 'cp32-5', 'cp32-9', 'chromosome', 'cp32-10', 'lp28-2', 'lp25', 'lp21', 'lp36', 'cp32-4', 'lp28-3', 'lp28-4', 'lp54', 'cp26', 'lp56']\n",
      "shortread: URI41 : len: 21['lp25', 'cp32-12', 'lp21', 'cp32-4', 'cp32-2', 'cp26', 'lp56', 'cp32-7', 'lp28-2', 'cp32-3', 'lp54', 'lp28-1', 'cp32-10', 'cp9-3', 'cp32-11', 'lp28-3', 'lp17', 'lp28-4', 'cp32-9', 'chromosome', 'lp36']\n",
      "longread: URI88H : len:20 : ['cp32-12', 'lp21', 'cp32-4', 'lp28-6', 'cp32-1', 'cp26', 'cp32-7', 'cp32-6', 'cp32-3', 'lp54', 'lp28-1', 'cp32-11', 'lp28-11', 'lp28-3', 'lp17', 'lp28-4', 'cp32-5', 'chromosome', 'cp32-8', 'lp38']\n",
      "shortread: URI88 : len: 23['cp32-12', 'lp28-7', 'lp21', 'cp32-4', 'lp28-6', 'cp32-1', 'cp26', 'lp56', 'cp32-6', 'cp32-3', 'lp54', 'lp28-1', 'cp9-3', 'cp32-11', 'lp28-11', 'lp28-3', 'lp17', 'lp28-4', 'cp32-5', 'chromosome', 'lp36', 'cp32-8', 'lp38']\n",
      "longread: URI91H : len:22 : ['lp25', 'lp21', 'cp32-4', 'cp32-1', 'cp26', 'cp32-7', 'cp32-6', 'lp28-2', 'cp32-3', 'lp54', 'lp28-1', 'cp32-10', 'cp32-11', 'lp28-3', 'lp28-4', 'lp17', 'cp32-9', 'cp32-5', 'chromosome', 'lp36', 'cp32-8', 'lp38']\n",
      "shortread: URI91 : len: 23['lp25', 'cp32-12', 'lp21', 'cp32-4', 'cp32-1', 'cp26', 'lp56', 'cp32-7', 'cp32-6', 'lp28-2', 'cp32-3', 'lp54', 'lp28-1', 'cp32-11', 'lp28-3', 'lp17', 'lp28-4', 'cp32-9', 'cp32-5', 'chromosome', 'lp36', 'cp32-8', 'lp38']\n",
      "longread: UCT29H : len:21 : ['lp25', 'lp21', 'cp32-4', 'cp9', 'cp32-1', 'cp26', 'cp32-7', 'cp32-6', 'lp28-2', 'cp32-3', 'lp54', 'lp28-1', 'cp32-10', 'cp32-11', 'lp28-3', 'lp28-4', 'lp17', 'cp32-9', 'cp32-5', 'chromosome', 'lp38']\n",
      "shortread: UCT29 : len: 23['lp25', 'cp32-12', 'lp21', 'cp32-4', 'cp9', 'cp32-1', 'cp26', 'lp56', 'cp32-7', 'cp32-6', 'lp28-2', 'cp32-3', 'lp54', 'lp28-1', 'cp32-11', 'lp28-3', 'lp28-4', 'lp17', 'cp32-9', 'chromosome', 'lp36', 'lp38', 'cp32-5-1']\n",
      "longread: URI34H : len:19 : ['cp32-12', 'lp28-7', 'lp28-6', 'cp9', 'cp26', 'lp5', 'cp32-7', 'cp32-6', 'lp28-2', 'lp54', 'cp32-13', 'cp32-10', 'lp28-5', 'lp28-3', 'lp17', 'lp28-4', 'chromosome', 'lp36', 'lp38']\n",
      "shortread: URI34 : len: 22['cp32-12', 'lp28-7', 'lp28-6', 'cp9', 'cp26', 'lp5', 'cp32-7', 'cp32-6', 'lp28-2', 'lp54', 'cp32-13', 'cp32-10', 'cp32-5+1', 'lp28-5', 'lp28-3', 'lp17', 'lp28-4', 'cp32-9', 'chromosome', 'lp36', 'cp32-8', 'lp38']\n",
      "longread: URI120H : len:23 : ['lp25', 'cp32-12', 'lp21', 'cp32-4', 'cp32-1', 'cp26', 'cp32-7', 'cp32-6', 'lp28-2', 'cp32-3', 'lp54', 'lp28-1', 'cp32-10', 'cp32-11', 'lp28-3', 'lp28-4', 'lp17', 'cp32-9', 'cp32-5', 'chromosome', 'lp36', 'cp32-8', 'lp38']\n",
      "shortread: URI120 : len: 16['lp28-1', 'lp25', 'cp9', 'chromosome', 'cp9-3', 'lp28-2', 'lp21', 'lp36', 'lp38', 'cp32-4', 'lp28-4', 'lp28-3', 'lp17', 'lp54', 'cp26', 'lp56']\n",
      "longread: UNY193P : len:19 : ['lp25', 'cp32-12', 'lp28-7', 'lp28-6', 'cp9', 'cp26', 'lp5', 'cp32-7', 'cp32-6', 'lp28-2', 'lp54', 'cp32-13', 'cp32-10', 'lp28-3', 'lp17', 'lp28-4', 'chromosome', 'lp36', 'lp38']\n",
      "shortread: UNY193 : len: 27['lp25', 'cp32-12', 'lp28-7', 'lp21', 'cp32-4', 'lp28-6', 'cp9', 'cp26', 'lp56', 'cp32-7', 'cp32-6', 'lp28-2', 'cp32-3', 'lp28-8', 'lp54', 'cp32-13', 'lp28-1', 'cp9-3', 'lp28-3', 'lp17', 'lp28-4', 'cp32-9-4', 'cp32-9', 'cp32-5', 'chromosome', 'lp36', 'lp38']\n",
      "longread: URI93H : len:17 : ['cp32-7', 'lp28-1', 'lp25', 'cp32-5', 'chromosome', 'cp32-10', 'cp32-9', 'lp28-2', 'cp32-3', 'lp21', 'lp36', 'lp38', 'cp32-4', 'lp28-3', 'lp28-4', 'lp54', 'cp26']\n",
      "shortread: URI93 : len: 22['lp25', 'lp21', 'cp32-4', 'cp32-2', 'cp26', 'lp56', 'cp32-7', 'lp28-2', 'cp32-3', 'lp54', 'lp28-1', 'cp9-3', 'cp32-11', 'lp28-3', 'lp17', 'lp28-4', 'cp32-9-4', 'cp32-9', 'cp32-5', 'chromosome', 'lp36', 'lp38']\n",
      "val      chromosome  cp26  cp32-1  cp32-1+5  cp32-10  cp32-11  cp32-12  \\\n",
      "key                                                                      \n",
      "ESI26             1     1       1         0        1        0        1   \n",
      "ESI26H            1     1       1         0        1        0        1   \n",
      "UCT109            1     1       0         0        0        1        0   \n",
      "UCT109H           1     1       0         0        1        0        0   \n",
      "UCT110            1     1       0         0        0        1        0   \n",
      "...             ...   ...     ...       ...      ...      ...      ...   \n",
      "UWI248P           1     1       1         0        0        1        1   \n",
      "UWI263            1     1       1         1        0        1        1   \n",
      "UWI263P           1     1       0         0        1        1        0   \n",
      "UWI283            1     1       1         0        1        0        1   \n",
      "UWI283P           1     1       1         0        0        0        1   \n",
      "\n",
      "val      cp32-13  cp32-2  cp32-3  ...  lp28-6  lp28-7  lp28-8  lp28-9  lp36  \\\n",
      "key                               ...                                         \n",
      "ESI26          0       0       0  ...       0       1       0       1     1   \n",
      "ESI26H         0       0       0  ...       0       0       0       1     1   \n",
      "UCT109         0       1       1  ...       0       0       0       0     1   \n",
      "UCT109H        0       0       1  ...       0       0       0       0     1   \n",
      "UCT110         0       1       1  ...       1       0       0       0     1   \n",
      "...          ...     ...     ...  ...     ...     ...     ...     ...   ...   \n",
      "UWI248P        0       0       1  ...       0       0       0       0     1   \n",
      "UWI263         0       0       0  ...       0       0       0       0     1   \n",
      "UWI263P        0       0       1  ...       0       0       0       0     1   \n",
      "UWI283         0       0       1  ...       1       0       0       0     1   \n",
      "UWI283P        0       0       1  ...       0       0       0       0     1   \n",
      "\n",
      "val      lp38  lp5  lp54  lp56  Sum  \n",
      "key                                  \n",
      "ESI26       0    0     1     0   17  \n",
      "ESI26H      0    0     1     0   13  \n",
      "UCT109      1    0     1     1   21  \n",
      "UCT109H     1    0     1     0   17  \n",
      "UCT110      1    0     1     0   22  \n",
      "...       ...  ...   ...   ...  ...  \n",
      "UWI248P     1    0     1     0   18  \n",
      "UWI263      1    0     1     0   20  \n",
      "UWI263P     1    0     1     0   19  \n",
      "UWI283      0    0     1     1   20  \n",
      "UWI283P     0    0     1     0   16  \n",
      "\n",
      "[98 rows x 40 columns]\n"
     ]
    }
   ],
   "source": [
    "sr_plasmids, lr_plasmids = {}, {}\n",
    "sr_plasmids, lr_plasmids = process_assembly_dict(assembly_dict, f\"{output_dir}/blast_results_v4\", parsing_table, 2500)\n",
    "\n",
    "for k in lr_plasmids:\n",
    "    print(f'longread: {k} : len:{len(list(set(lr_plasmids[k])))} : {list(set(lr_plasmids[k]))}')\n",
    "    lr_plasmids[k] = list(set(lr_plasmids[k]))\n",
    "    print(f'shortread: {k[:-1]} : len: {len(list(set(sr_plasmids[k[:-1]])))}{list(set(sr_plasmids[k[:-1]]))}')\n",
    "    sr_plasmids[k[:-1]] = list(set(sr_plasmids[k[:-1]]))\n",
    "\n",
    "dicts_of_results = [sr_plasmids, lr_plasmids]\n",
    "plasmid_presence_matrix = []\n",
    "for dict1 in dicts_of_results:\n",
    "    for k in dict1:\n",
    "        #print(k)\n",
    "        for v in dict1[k]:\n",
    "            #print(v)\n",
    "            plasmid_presence_matrix.append((k,v))\n",
    "\n",
    "plasmid_df = pandas.DataFrame(plasmid_presence_matrix, columns=['key','val']).set_index(['key'])\n",
    "plasmid_count_matrix = plasmid_df.pivot_table(index='key', columns='val',aggfunc=len,fill_value=0,sort=True)\n",
    "plasmid_count_matrix['Sum'] = plasmid_count_matrix.sum(axis=1)\n",
    "print(plasmid_count_matrix)\n",
    "\n",
    "plasmid_count_matrix.to_csv(f'{output_dir}/plasmid_presence_matrix_v4_2500cutoff.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "longread: URI47H : len:19 : ['cp32-12', 'cp32-4', 'lp28-6', 'cp9', 'cp26', 'cp32-7', 'cp32-6', 'cp32-3', 'lp54', 'lp28-1', 'cp32-11', 'lp28-3', 'lp17', 'lp28-4', 'cp32-9', 'cp32-5', 'chromosome', 'cp32-8', 'lp38']\n",
      "shortread: URI47 : len: 23['lp25', 'cp32-12', 'lp21', 'lp28-6', 'cp32-2', 'cp9', 'cp26', 'cp32-7', 'cp32-6', 'cp32-3', 'lp54', 'lp28-1', 'cp32-11', 'lp28-5', 'lp28-3', 'lp17', 'lp28-4', 'cp32-9', 'cp32-5', 'chromosome', 'lp36', 'cp32-8', 'lp38']\n",
      "longread: URI103H : len:18 : ['cp32-7', 'cp32-6', 'cp32-9', 'cp32-5', 'cp32-12', 'chromosome', 'lp25', 'lp28-1', 'cp32-3', 'cp32-4', 'lp38', 'lp28-6', 'cp32-11', 'lp17', 'lp28-3', 'lp28-4', 'lp54', 'cp26']\n",
      "shortread: URI103 : len: 21['lp25', 'cp32-12', 'lp21', 'lp28-6', 'cp32-2', 'cp26', 'cp32-7', 'cp32-6', 'cp32-3', 'lp54', 'lp28-1', 'cp32-11', 'lp28-5', 'lp28-3', 'lp17', 'lp28-4', 'cp32-9', 'cp32-5', 'chromosome', 'lp36', 'lp38']\n",
      "longread: UCT31H : len:21 : ['lp25', 'lp21', 'cp32-4', 'cp9', 'cp32-1', 'cp26', 'cp32-7', 'cp32-6', 'lp28-2', 'cp32-3', 'lp54', 'lp28-1', 'cp32-10', 'cp32-11', 'lp28-3', 'lp28-4', 'lp17', 'cp32-9', 'cp32-5', 'chromosome', 'lp36']\n",
      "shortread: UCT31 : len: 26['lp25', 'cp32-12', 'lp21', 'cp32-4', 'cp32-2', 'cp9', 'cp32-1', 'cp26', 'lp56', 'cp32-7', 'cp32-6', 'lp28-2', 'cp32-3', 'lp54', 'lp28-1', 'cp9-3', 'cp32-11', 'lp28-3', 'lp17', 'lp28-4', 'cp32-9-4', 'cp32-9', 'cp32-5', 'chromosome', 'lp36', 'cp32-8']\n",
      "longread: UCT92H : len:21 : ['cp32-12', 'cp32-4', 'lp28-6', 'cp32-1', 'cp26', 'cp32-7', 'cp32-6', 'cp32-3', 'lp54', 'lp28-1', 'cp32-11', 'lp28-5', 'lp28-3', 'lp28-4', 'lp17', 'cp32-9', 'cp32-5', 'chromosome', 'lp36', 'cp32-8', 'lp38']\n",
      "shortread: UCT92 : len: 23['lp25', 'cp32-12', 'lp21', 'lp28-6', 'cp32-2', 'cp26', 'cp32-6', 'cp32-3', 'lp54', 'lp28-1', 'cp9-3', 'cp32-11', 'lp28-5', 'lp28-11', 'lp28-3', 'lp17', 'lp28-4', 'cp32-9', 'cp32-5', 'chromosome', 'lp36', 'cp32-8', 'lp38']\n",
      "longread: URI46H : len:19 : ['lp25', 'cp32-12', 'lp21', 'cp32-4', 'lp28-6', 'cp26', 'cp32-7', 'cp32-6', 'cp32-3', 'lp54', 'lp28-1', 'cp32-11', 'lp28-3', 'lp17', 'lp28-4', 'cp32-9', 'cp32-5', 'chromosome', 'lp38']\n",
      "shortread: URI46 : len: 21['lp25', 'cp32-12', 'lp21', 'lp28-6', 'cp32-2', 'cp26', 'cp32-6', 'cp32-3', 'lp54', 'lp28-1', 'cp9-3', 'cp32-11', 'lp28-5', 'lp28-3', 'lp17', 'lp28-4', 'cp32-9', 'cp32-5', 'chromosome', 'lp36', 'lp38']\n",
      "longread: UWI247P : len:17 : ['cp32-7', 'cp32-6', 'lp28-1', 'lp25', 'chromosome', 'cp32-3', 'lp54', 'cp32-4', 'lp36', 'cp32-11', 'lp28-4', 'lp28-5', 'lp28-3', 'lp17', 'lp38', 'cp32-1', 'cp26']\n",
      "shortread: UWI247 : len: 19['lp25', 'cp32-12', 'cp32-4', 'cp32-2', 'cp32-1', 'cp26', 'cp32-7', 'cp32-6', 'cp32-3', 'lp54', 'lp28-1', 'cp32-11', 'lp28-5', 'lp28-3', 'lp28-4', 'lp17', 'chromosome', 'lp36', 'lp38']\n",
      "longread: UCT30H : len:19 : ['cp32-12', 'lp21', 'cp32-4', 'lp28-6', 'cp26', 'cp32-7', 'cp32-6', 'cp32-3', 'lp54', 'lp28-1', 'cp32-11', 'lp28-3', 'lp17', 'lp28-4', 'cp32-9', 'cp32-5', 'chromosome', 'lp36', 'lp38']\n",
      "shortread: UCT30 : len: 21['lp25', 'cp32-12', 'lp21', 'lp28-6', 'cp32-2', 'cp26', 'cp32-7', 'cp32-6', 'cp32-3', 'lp54', 'lp28-1', 'cp9-3', 'cp32-11', 'lp28-5', 'lp28-3', 'lp17', 'lp28-4', 'cp32-9', 'cp32-5', 'chromosome', 'lp38']\n",
      "longread: URI44H : len:20 : ['lp25', 'cp32-12', 'cp32-4', 'lp28-6', 'cp26', 'cp32-7', 'cp32-6', 'cp32-3', 'lp54', 'lp28-1', 'cp32-11', 'lp28-5', 'lp28-3', 'lp17', 'lp28-4', 'cp32-9', 'cp32-5', 'chromosome', 'lp36', 'lp38']\n",
      "shortread: URI44 : len: 19['lp25', 'lp21', 'lp28-6', 'cp32-2', 'cp26', 'cp32-6', 'cp32-3', 'lp54', 'lp28-1', 'cp32-11', 'lp28-5', 'lp28-3', 'lp17', 'lp28-4', 'cp32-9', 'cp32-5', 'chromosome', 'lp36', 'lp38']\n",
      "longread: URI87H : len:20 : ['lp25', 'cp32-4', 'cp26', 'lp56', 'cp32-7', 'cp32-6', 'lp28-2', 'cp32-3', 'lp54', 'lp28-1', 'cp32-10', 'cp32-11', 'lp28-3', 'lp17', 'lp28-4', 'cp32-9', 'cp32-5', 'chromosome', 'lp36', 'lp38']\n",
      "shortread: URI87 : len: 21['lp25', 'cp32-4', 'cp32-1', 'cp26', 'lp56', 'cp32-7', 'cp32-6', 'lp28-2', 'cp32-3', 'lp54', 'lp28-1', 'cp32-11', 'lp28-3', 'lp17', 'lp28-4', 'cp32-9-4', 'cp32-9', 'cp32-5', 'chromosome', 'lp36', 'lp38']\n",
      "longread: UCT50H : len:15 : ['cp32-7', 'cp32-9', 'cp32-5', 'cp32-12', 'chromosome', 'lp25', 'lp28-2', 'cp32-10', 'cp9', 'lp36', 'lp38', 'cp32-4', 'lp17', 'lp54', 'cp26']\n",
      "shortread: UCT50 : len: 18['cp32-7', 'cp32-9', 'lp25', 'cp32-12', 'chromosome', 'cp32-10', 'lp28-2', 'cp9', 'cp32-5', 'lp36', 'lp38', 'cp32-4', 'lp28-4', 'cp32-11', 'cp32-8', 'lp17', 'lp54', 'cp26']\n",
      "longread: UWI283P : len:16 : ['cp32-9', 'cp32-5', 'cp32-12', 'chromosome', 'lp25', 'lp28-2', 'cp32-3', 'cp32-4', 'lp36', 'lp28-4', 'lp28-5', 'lp28-3', 'lp17', 'lp54', 'cp32-1', 'cp26']\n",
      "shortread: UWI283 : len: 20['lp25', 'cp32-12', 'cp32-4', 'lp28-6', 'cp32-1', 'cp26', 'lp56', 'cp32-7', 'lp28-2', 'cp32-3', 'lp54', 'cp32-10', 'lp28-5', 'lp28-3', 'lp17', 'lp28-4', 'cp32-9', 'cp32-5', 'chromosome', 'lp36']\n",
      "longread: URI107H : len:19 : ['lp25', 'cp32-4', 'cp26', 'cp32-7', 'cp32-6', 'lp28-2', 'cp32-3', 'lp54', 'lp28-1', 'cp32-10', 'cp32-11', 'lp28-3', 'lp17', 'lp28-4', 'cp32-9', 'cp32-5', 'chromosome', 'lp36', 'lp38']\n",
      "shortread: URI107 : len: 21['lp25', 'cp32-4', 'cp32-2', 'cp26', 'lp56', 'cp32-7', 'cp32-6', 'lp28-2', 'cp32-3', 'lp54', 'lp28-1', 'cp32-11', 'lp28-3', 'lp17', 'lp28-4', 'cp32-9-4', 'cp32-9', 'cp32-5', 'chromosome', 'lp36', 'lp38']\n",
      "longread: UCT110H : len:21 : ['lp25', 'cp32-12', 'lp21', 'cp32-4', 'lp28-6', 'cp26', 'cp32-7', 'cp32-6', 'cp32-3', 'lp54', 'lp28-1', 'cp32-11', 'lp28-5', 'lp28-3', 'lp17', 'lp28-4', 'cp32-9', 'cp32-5', 'chromosome', 'cp32-8', 'lp38']\n",
      "shortread: UCT110 : len: 22['lp25', 'cp32-4', 'lp28-6', 'cp32-2', 'cp26', 'cp32-7', 'cp32-6', 'cp32-3', 'lp54', 'lp28-1', 'cp32-5+1', 'cp32-11', 'lp28-5', 'lp28-3', 'lp17', 'lp28-4', 'cp32-9', 'cp32-5', 'chromosome', 'lp36', 'cp32-8', 'lp38']\n",
      "longread: URI48H : len:16 : ['cp32-6', 'lp25', 'cp32-5', 'chromosome', 'cp32-10', 'cp9-3', 'lp28-7', 'lp36', 'lp38', 'cp32-11', 'lp17', 'cp32-8', 'lp28-3', 'lp28-4', 'lp54', 'cp26']\n",
      "shortread: URI48 : len: 19['lp25', 'lp28-7', 'cp32-4', 'cp26', 'cp32-6', 'cp32-3', 'lp28-8', 'lp54', 'cp9-3', 'cp32-11', 'lp28-3', 'lp17', 'lp28-4', 'cp32-9', 'cp32-5', 'chromosome', 'lp36', 'cp32-8', 'lp38']\n",
      "longread: UNY203P : len:17 : ['cp32-7', 'cp32-6', 'lp28-1', 'cp32-5', 'cp32-12', 'chromosome', 'cp32-10', 'cp9-3', 'lp54', 'cp32-4', 'lp28-6', 'lp36', 'lp28-3', 'lp17', 'lp38', 'cp26', 'lp5']\n",
      "shortread: UNY203 : len: 22['lp25', 'cp32-12', 'lp28-6', 'cp26', 'lp5', 'cp32-7', 'cp32-6', 'cp32-3', 'lp54', 'cp32-13', 'lp28-1', 'cp32-10', 'cp9-3', 'cp32-11', 'lp28-11', 'lp28-3', 'lp28-4', 'lp17', 'cp32-5', 'chromosome', 'lp36', 'lp38']\n",
      "longread: UWI263P : len:19 : ['lp25', 'cp9', 'cp26', 'cp32-6', 'cp32-3', 'lp54', 'lp28-1', 'cp32-10', 'cp32-11', 'lp28-5', 'lp28-3', 'lp17', 'lp28-4', 'cp32-9', 'cp32-5', 'chromosome', 'lp36', 'cp32-8', 'lp38']\n",
      "shortread: UWI263 : len: 20['lp25', 'cp32-12', 'cp9', 'cp32-1', 'cp26', 'cp32-7', 'cp32-6', 'cp32-1+5', 'lp54', 'lp28-1', 'cp32-11', 'lp28-5', 'lp28-3', 'lp17', 'lp28-4', 'cp32-9', 'cp32-5', 'chromosome', 'lp36', 'lp38']\n",
      "longread: URI117H : len:20 : ['lp25', 'cp32-12', 'lp21', 'cp32-4', 'lp28-6', 'cp26', 'cp32-7', 'cp32-6', 'cp32-3', 'lp54', 'lp28-1', 'cp32-11', 'lp28-5', 'lp28-3', 'lp17', 'lp28-4', 'cp32-9', 'chromosome', 'lp36', 'lp38']\n",
      "shortread: URI117 : len: 16['cp32-6', 'lp28-1', 'lp25', 'cp32-9', 'chromosome', 'cp32-5', 'lp21', 'lp36', 'lp38', 'lp28-6', 'lp28-4', 'lp28-5', 'lp28-3', 'lp17', 'lp54', 'cp26']\n",
      "longread: UCT35H : len:15 : ['cp32-6', 'cp32-13', 'lp28-1', 'cp32-5', 'cp32-12', 'chromosome', 'cp9-3', 'cp32-10', 'cp32-3', 'lp38', 'lp17', 'lp28-4', 'lp54', 'cp26', 'lp5']\n",
      "shortread: UCT35 : len: 21['lp25', 'cp32-12', 'cp32-4', 'cp26', 'lp5', 'cp32-7', 'cp32-6', 'cp32-3', 'lp54', 'lp28-1', 'cp32-10', 'cp9-3', 'cp32-11', 'lp28-11', 'lp28-3', 'lp17', 'lp28-4', 'cp32-5', 'chromosome', 'lp36', 'lp38']\n",
      "longread: URI111H : len:19 : ['lp25', 'lp21', 'cp32-4', 'cp26', 'cp32-7', 'cp32-6', 'lp28-2', 'lp54', 'lp28-1', 'cp32-10', 'cp32-11', 'lp28-3', 'lp17', 'lp28-4', 'cp32-9', 'cp32-5', 'chromosome', 'lp36', 'lp38']\n",
      "shortread: URI111 : len: 21['lp25', 'cp32-12', 'lp21', 'cp32-4', 'cp32-2', 'cp26', 'lp56', 'cp32-7', 'lp28-2', 'cp32-3', 'lp54', 'lp28-1', 'cp9-3', 'lp28-3', 'lp17', 'lp28-4', 'cp32-9', 'cp32-5', 'chromosome', 'lp36', 'lp38']\n",
      "longread: URI33H : len:17 : ['cp32-6', 'lp25', 'cp32-5', 'chromosome', 'cp32-10', 'lp28-7', 'lp21', 'lp36', 'lp38', 'lp17', 'lp28-8', 'cp32-11', 'cp32-8', 'lp28-3', 'lp28-4', 'lp54', 'cp26']\n",
      "shortread: URI33 : len: 23['lp25', 'cp32-12', 'lp28-7', 'lp21', 'cp32-4', 'cp32-2', 'cp26', 'cp32-7', 'cp32-6', 'cp32-1+5', 'lp28-8', 'lp54', 'cp9-3', 'cp32-11', 'lp28-5', 'lp28-3', 'lp17', 'lp28-4', 'cp32-5', 'chromosome', 'lp36', 'cp32-8', 'lp38']\n",
      "longread: URI86H : len:17 : ['cp32-7', 'cp32-6', 'cp32-9', 'lp25', 'lp28-1', 'chromosome', 'cp32-5', 'cp32-12', 'cp32-4', 'lp38', 'lp36', 'cp32-11', 'lp28-6', 'lp17', 'lp28-4', 'lp54', 'cp26']\n",
      "shortread: URI86 : len: 19['lp25', 'cp32-12', 'lp21', 'lp28-6', 'cp32-2', 'cp26', 'cp32-7', 'cp32-6', 'lp54', 'lp28-1', 'cp32-11', 'lp28-5', 'lp28-3', 'lp17', 'lp28-4', 'cp32-9', 'chromosome', 'lp36', 'lp38']\n",
      "longread: UNY149P : len:19 : ['lp25', 'cp32-4', 'lp28-6', 'cp26', 'cp32-7', 'cp32-6', 'cp32-3', 'lp54', 'lp28-1', 'cp32-11', 'lp28-5', 'lp28-3', 'lp28-4', 'lp17', 'cp32-9', 'cp32-5', 'chromosome', 'lp36', 'lp38']\n",
      "shortread: UNY149 : len: 22['lp25', 'cp32-12', 'lp21', 'cp32-4', 'lp28-6', 'cp32-2', 'cp26', 'cp32-7', 'cp32-6', 'cp32-3', 'lp54', 'lp28-1', 'cp32-11', 'lp28-5', 'lp28-3', 'lp28-4', 'cp32-9', 'cp32-5', 'chromosome', 'lp36', 'cp32-8', 'lp38']\n",
      "longread: UCT96H : len:20 : ['lp25', 'cp32-4', 'cp32-1', 'cp26', 'cp32-7', 'cp32-6', 'lp28-2', 'cp32-3', 'lp54', 'lp28-1', 'cp32-11', 'lp28-5', 'lp28-3', 'lp17', 'lp28-4', 'cp32-9', 'chromosome', 'lp36', 'cp32-8', 'lp38']\n",
      "shortread: UCT96 : len: 15['lp28-1', 'lp25', 'cp32-9', 'chromosome', 'lp28-2', 'lp36', 'lp38', 'lp28-4', 'lp28-5', 'cp32-8', 'lp28-3', 'lp17', 'lp54', 'cp32-1', 'cp26']\n",
      "longread: URI56H : len:17 : ['cp32-7', 'cp32-6', 'cp32-5', 'chromosome', 'cp9-3', 'cp32-3', 'lp21', 'lp36', 'lp38', 'lp28-6', 'lp17', 'cp32-8', 'lp28-3', 'lp28-4', 'lp54', 'cp26', 'lp56']\n",
      "shortread: URI56 : len: 19['lp25', 'lp21', 'lp28-6', 'cp26', 'lp56', 'cp32-7', 'cp32-6', 'cp32-3', 'lp54', 'lp28-1', 'cp9-3', 'lp28-3', 'lp17', 'lp28-4', 'cp32-5', 'chromosome', 'lp36', 'cp32-8', 'lp38']\n",
      "longread: URI118H : len:18 : ['cp32-7', 'cp32-6', 'cp32-9', 'cp32-5', 'lp28-1', 'chromosome', 'cp32-12', 'cp32-3', 'cp32-4', 'lp28-6', 'lp36', 'lp17', 'cp32-11', 'lp28-5', 'lp28-3', 'lp28-4', 'lp54', 'cp26']\n",
      "shortread: URI118 : len: 14['cp32-6', 'lp28-1', 'lp25', 'cp32-9', 'chromosome', 'cp32-5', 'lp36', 'lp28-6', 'lp28-4', 'lp28-5', 'lp28-3', 'lp17', 'lp54', 'cp26']\n",
      "longread: UNY172P : len:16 : ['cp32-7', 'cp32-9', 'cp32-5', 'cp32-12', 'chromosome', 'lp25', 'lp28-2', 'cp32-3', 'lp54', 'cp32-4', 'lp36', 'lp28-4', 'lp28-3', 'lp17', 'lp38', 'cp26']\n",
      "shortread: UNY172 : len: 17['cp32-7', 'cp32-9', 'lp25', 'cp32-12', 'cp32-10', 'cp32-5', 'lp28-2', 'chromosome', 'cp32-3', 'cp32-4', 'lp36', 'lp38', 'lp17', 'lp28-3', 'lp28-4', 'lp54', 'cp26']\n",
      "longread: URI112H : len:19 : ['lp25', 'cp32-12', 'lp21', 'cp32-4', 'lp28-6', 'cp26', 'cp32-7', 'cp32-6', 'cp32-3', 'lp54', 'lp28-1', 'cp32-11', 'lp17', 'lp28-4', 'cp32-9', 'cp32-5', 'chromosome', 'cp32-8', 'lp38']\n",
      "shortread: URI112 : len: 22['lp25', 'cp32-12', 'lp21', 'lp28-6', 'cp32-2', 'cp26', 'cp32-6', 'cp32-3', 'lp54', 'lp28-1', 'cp9-3', 'cp32-11', 'lp28-5', 'lp28-3', 'lp17', 'lp28-4', 'cp32-9', 'cp32-5', 'chromosome', 'lp36', 'cp32-8', 'lp38']\n",
      "longread: UNY208P : len:18 : ['cp32-7', 'lp28-1', 'cp32-5', 'cp32-12', 'chromosome', 'lp25', 'cp32-3', 'cp32-4', 'lp28-6', 'lp36', 'cp32-11', 'lp28-4', 'cp32-8', 'lp28-3', 'lp17', 'lp54', 'cp32-1', 'cp26']\n",
      "shortread: UNY208 : len: 18['cp32-7', 'lp28-1', 'lp25', 'cp32-12', 'chromosome', 'cp32-5', 'cp32-3', 'lp36', 'lp28-6', 'cp32-4', 'lp17', 'cp32-11', 'cp32-8', 'lp28-3', 'lp28-4', 'lp54', 'cp32-1', 'cp26']\n",
      "longread: URI89H : len:22 : ['lp25', 'lp21', 'cp32-4', 'cp32-1', 'cp26', 'lp56', 'cp32-7', 'cp32-6', 'lp28-2', 'cp32-3', 'lp54', 'lp28-1', 'cp32-11', 'lp28-3', 'lp28-4', 'lp17', 'cp32-9', 'cp32-5', 'chromosome', 'lp36', 'cp32-8', 'lp38']\n",
      "shortread: URI89 : len: 22['lp25', 'cp32-12', 'lp21', 'cp32-4', 'cp32-1', 'cp26', 'lp56', 'cp32-7', 'cp32-6', 'lp28-2', 'cp32-3', 'lp54', 'lp28-1', 'cp9-3', 'lp28-3', 'lp17', 'lp28-4', 'cp32-9', 'chromosome', 'lp36', 'cp32-8', 'lp38']\n",
      "longread: URI102H : len:17 : ['cp32-7', 'cp32-9', 'lp25', 'cp32-5', 'chromosome', 'cp32-10', 'lp28-2', 'lp28-1', 'cp32-3', 'lp36', 'lp38', 'cp32-4', 'lp28-3', 'lp17', 'lp54', 'cp26', 'lp56']\n",
      "shortread: URI102 : len: 20['lp25', 'cp32-4', 'cp32-2', 'cp26', 'lp56', 'cp32-7', 'lp28-2', 'cp32-3', 'lp54', 'lp28-1', 'cp32-11', 'lp28-3', 'lp17', 'lp28-4', 'cp32-9-4', 'cp32-9', 'cp32-5', 'chromosome', 'lp36', 'lp38']\n",
      "longread: URI36H : len:21 : ['lp25', 'lp21', 'cp32-4', 'lp28-6', 'cp26', 'lp56', 'cp32-7', 'cp32-6', 'cp32-3', 'lp54', 'lp28-1', 'cp9-3', 'cp32-11', 'lp28-3', 'lp28-4', 'lp17', 'cp32-5', 'chromosome', 'lp36', 'cp32-8', 'lp38']\n",
      "shortread: URI36 : len: 20['lp25', 'lp21', 'lp28-6', 'cp26', 'lp56', 'cp32-7', 'cp32-6', 'cp32-3', 'lp54', 'lp28-1', 'cp9-3', 'cp32-11', 'lp28-5', 'lp28-3', 'lp17', 'lp28-4', 'cp32-5', 'chromosome', 'lp36', 'lp38']\n",
      "longread: UWI248P : len:18 : ['cp32-7', 'cp32-6', 'lp28-1', 'lp25', 'cp32-12', 'chromosome', 'cp32-3', 'lp54', 'cp32-4', 'lp36', 'cp32-11', 'lp28-4', 'lp28-5', 'lp28-3', 'lp17', 'lp38', 'cp32-1', 'cp26']\n",
      "shortread: UWI248 : len: 19['lp25', 'cp32-12', 'cp32-4', 'cp32-2', 'cp32-1', 'cp26', 'cp32-7', 'cp32-6', 'cp32-3', 'lp54', 'lp28-1', 'cp32-11', 'lp28-5', 'lp28-3', 'lp17', 'lp28-4', 'chromosome', 'lp36', 'lp38']\n",
      "longread: ESI26H : len:13 : ['cp32-6', 'lp28-9', 'cp32-9', 'cp32-12', 'chromosome', 'cp32-10', 'lp36', 'lp17', 'lp28-3', 'lp28-4', 'lp54', 'cp32-1', 'cp26']\n",
      "shortread: ESI26 : len: 17['cp32-6', 'lp28-9', 'cp32-9', 'cp32-5', 'cp32-12', 'chromosome', 'cp32-10', 'lp28-2', 'lp28-7', 'lp36', 'lp17', 'cp32-8', 'lp28-3', 'lp28-4', 'lp54', 'cp32-1', 'cp26']\n",
      "longread: URI40H : len:21 : ['lp25', 'cp32-4', 'cp26', 'lp56', 'cp32-7', 'cp32-6', 'lp28-2', 'cp32-3', 'lp54', 'lp28-1', 'cp32-10', 'cp9-3', 'cp32-11', 'lp28-3', 'lp17', 'lp28-4', 'cp32-9', 'cp32-5', 'chromosome', 'lp36', 'lp38']\n",
      "shortread: URI40 : len: 24['lp25', 'cp32-12', 'lp21', 'cp32-4', 'cp32-2', 'cp9', 'cp26', 'lp56', 'cp32-7', 'cp32-6', 'lp28-2', 'cp32-3', 'lp54', 'lp28-1', 'cp9-3', 'cp32-11', 'lp28-3', 'lp17', 'lp28-4', 'cp32-9', 'cp32-5', 'chromosome', 'lp36', 'lp38']\n",
      "longread: URI39H : len:21 : ['lp25', 'lp21', 'cp32-4', 'cp26', 'cp32-7', 'cp32-6', 'lp28-2', 'cp32-3', 'lp54', 'lp28-1', 'cp32-10', 'cp32-11', 'lp28-3', 'lp28-4', 'lp17', 'cp32-9', 'cp32-5', 'chromosome', 'lp36', 'cp32-8', 'lp38']\n",
      "shortread: URI39 : len: 22['lp25', 'lp21', 'cp32-4', 'cp26', 'lp56', 'cp32-7', 'cp32-6', 'lp28-2', 'cp32-3', 'lp54', 'lp28-1', 'cp9-3', 'cp32-11', 'lp28-3', 'lp17', 'lp28-4', 'cp32-9', 'cp32-5', 'chromosome', 'lp36', 'cp32-8', 'lp38']\n",
      "longread: UNY169P : len:17 : ['cp32-7', 'cp32-9', 'cp32-5', 'cp32-12', 'chromosome', 'cp32-10', 'lp25', 'lp28-2', 'lp54', 'cp32-4', 'lp36', 'lp17', 'lp28-4', 'lp28-3', 'cp9', 'lp38', 'cp26']\n",
      "shortread: UNY169 : len: 17['cp32-7', 'cp32-9', 'lp25', 'cp32-12', 'chromosome', 'cp9', 'lp28-2', 'cp32-10', 'cp32-5', 'lp36', 'lp38', 'cp32-4', 'lp17', 'lp28-3', 'lp28-4', 'lp54', 'cp26']\n",
      "longread: UCT113H : len:16 : ['cp32-7', 'lp28-1', 'lp25', 'chromosome', 'lp21', 'cp32-4', 'lp28-6', 'lp38', 'lp36', 'lp28-4', 'cp32-11', 'lp28-5', 'lp28-3', 'lp17', 'lp54', 'cp26']\n",
      "shortread: UCT113 : len: 21['lp25', 'lp21', 'cp32-4', 'lp28-6', 'cp32-1', 'cp26', 'cp32-7', 'cp32-6', 'cp32-3', 'lp54', 'lp28-1', 'cp32-11', 'lp28-5', 'lp28-3', 'lp17', 'lp28-4', 'cp32-9', 'chromosome', 'lp36', 'cp32-8', 'lp38']\n",
      "longread: URI101H : len:21 : ['lp25', 'lp21', 'cp32-4', 'cp26', 'cp32-7', 'cp32-6', 'lp28-2', 'cp32-3', 'lp54', 'lp28-1', 'cp32-10', 'cp32-11', 'lp28-5', 'lp28-3', 'lp28-4', 'lp17', 'cp32-9', 'cp32-5', 'chromosome', 'lp36', 'lp38']\n",
      "shortread: URI101 : len: 23['lp25', 'lp21', 'cp32-4', 'cp32-2', 'cp26', 'lp56', 'cp32-7', 'cp32-6', 'lp28-2', 'cp32-3', 'lp54', 'lp28-1', 'cp9-3', 'cp32-11', 'lp28-5', 'lp28-3', 'lp28-4', 'lp17', 'cp32-9', 'cp32-5', 'chromosome', 'lp36', 'lp38']\n",
      "longread: URI42H : len:18 : ['cp32-7', 'cp32-6', 'lp28-1', 'lp25', 'cp32-9', 'chromosome', 'cp32-10', 'lp28-2', 'cp32-5', 'cp32-3', 'lp36', 'cp32-4', 'cp32-11', 'lp28-3', 'lp17', 'lp54', 'cp26', 'lp56']\n",
      "shortread: URI42 : len: 19['lp25', 'cp32-12', 'cp32-2', 'cp26', 'lp56', 'cp32-7', 'cp32-6', 'lp28-2', 'cp32-3', 'lp54', 'lp28-1', 'cp32-11', 'lp28-3', 'lp17', 'lp28-4', 'cp32-9', 'cp32-5', 'chromosome', 'lp36']\n",
      "longread: UCT109H : len:17 : ['cp32-7', 'cp32-6', 'lp28-1', 'lp25', 'cp32-5', 'chromosome', 'cp32-10', 'lp28-2', 'cp32-9', 'cp32-3', 'lp36', 'lp38', 'cp32-4', 'lp28-3', 'lp17', 'lp54', 'cp26']\n",
      "shortread: UCT109 : len: 21['lp25', 'cp32-4', 'cp32-2', 'cp26', 'lp56', 'cp32-7', 'cp32-6', 'lp28-2', 'cp32-3', 'lp54', 'lp28-1', 'cp32-11', 'lp28-3', 'lp17', 'lp28-4', 'cp32-9-4', 'cp32-9', 'cp32-5', 'chromosome', 'lp36', 'lp38']\n",
      "longread: UCT32H : len:20 : ['lp25', 'cp32-12', 'cp32-4', 'lp28-6', 'cp26', 'cp32-7', 'cp32-6', 'cp32-3', 'lp54', 'lp28-1', 'cp32-11', 'lp28-5', 'lp28-3', 'lp17', 'lp28-4', 'cp32-9', 'cp32-5', 'chromosome', 'cp32-8', 'lp38']\n",
      "shortread: UCT32 : len: 22['lp25', 'cp32-12', 'lp21', 'lp28-6', 'cp32-2', 'cp26', 'cp32-7', 'cp32-6', 'cp32-3', 'lp54', 'lp28-1', 'cp32-11', 'lp28-5', 'lp28-3', 'lp17', 'lp28-4', 'cp32-9', 'cp32-5', 'chromosome', 'lp36', 'cp32-8', 'lp38']\n",
      "longread: URI41H : len:17 : ['cp32-7', 'cp32-6', 'lp28-1', 'cp32-5', 'cp32-9', 'chromosome', 'cp32-10', 'lp28-2', 'lp25', 'lp21', 'lp36', 'cp32-4', 'lp28-3', 'lp28-4', 'lp54', 'cp26', 'lp56']\n",
      "shortread: URI41 : len: 21['lp25', 'cp32-12', 'lp21', 'cp32-4', 'cp32-2', 'cp26', 'lp56', 'cp32-7', 'lp28-2', 'cp32-3', 'lp54', 'lp28-1', 'cp32-10', 'cp9-3', 'cp32-11', 'lp28-3', 'lp17', 'lp28-4', 'cp32-9', 'chromosome', 'lp36']\n",
      "longread: URI88H : len:20 : ['cp32-12', 'lp21', 'cp32-4', 'lp28-6', 'cp32-1', 'cp26', 'cp32-7', 'cp32-6', 'cp32-3', 'lp54', 'lp28-1', 'cp32-11', 'lp28-11', 'lp28-3', 'lp17', 'lp28-4', 'cp32-5', 'chromosome', 'cp32-8', 'lp38']\n",
      "shortread: URI88 : len: 23['cp32-12', 'lp28-7', 'lp21', 'cp32-4', 'lp28-6', 'cp32-1', 'cp26', 'lp56', 'cp32-6', 'cp32-3', 'lp54', 'lp28-1', 'cp9-3', 'cp32-11', 'lp28-11', 'lp28-3', 'lp17', 'lp28-4', 'cp32-5', 'chromosome', 'lp36', 'cp32-8', 'lp38']\n",
      "longread: URI91H : len:22 : ['lp25', 'lp21', 'cp32-4', 'cp32-1', 'cp26', 'cp32-7', 'cp32-6', 'lp28-2', 'cp32-3', 'lp54', 'lp28-1', 'cp32-10', 'cp32-11', 'lp28-3', 'lp28-4', 'lp17', 'cp32-9', 'cp32-5', 'chromosome', 'lp36', 'cp32-8', 'lp38']\n",
      "shortread: URI91 : len: 23['lp25', 'cp32-12', 'lp21', 'cp32-4', 'cp32-1', 'cp26', 'lp56', 'cp32-7', 'cp32-6', 'lp28-2', 'cp32-3', 'lp54', 'lp28-1', 'cp32-11', 'lp28-3', 'lp17', 'lp28-4', 'cp32-9', 'cp32-5', 'chromosome', 'lp36', 'cp32-8', 'lp38']\n",
      "longread: UCT29H : len:21 : ['lp25', 'lp21', 'cp32-4', 'cp9', 'cp32-1', 'cp26', 'cp32-7', 'cp32-6', 'lp28-2', 'cp32-3', 'lp54', 'lp28-1', 'cp32-10', 'cp32-11', 'lp28-3', 'lp28-4', 'lp17', 'cp32-9', 'cp32-5', 'chromosome', 'lp38']\n",
      "shortread: UCT29 : len: 23['lp25', 'cp32-12', 'lp21', 'cp32-4', 'cp9', 'cp32-1', 'cp26', 'lp56', 'cp32-7', 'cp32-6', 'lp28-2', 'cp32-3', 'lp54', 'lp28-1', 'cp32-11', 'lp28-3', 'lp28-4', 'lp17', 'cp32-9', 'chromosome', 'lp36', 'lp38', 'cp32-5-1']\n",
      "longread: URI34H : len:19 : ['cp32-12', 'lp28-7', 'lp28-6', 'cp9', 'cp26', 'lp5', 'cp32-7', 'cp32-6', 'lp28-2', 'lp54', 'cp32-13', 'cp32-10', 'lp28-5', 'lp28-3', 'lp17', 'lp28-4', 'chromosome', 'lp36', 'lp38']\n",
      "shortread: URI34 : len: 22['cp32-12', 'lp28-7', 'lp28-6', 'cp9', 'cp26', 'lp5', 'cp32-7', 'cp32-6', 'lp28-2', 'lp54', 'cp32-13', 'cp32-10', 'cp32-5+1', 'lp28-5', 'lp28-3', 'lp17', 'lp28-4', 'cp32-9', 'chromosome', 'lp36', 'cp32-8', 'lp38']\n",
      "longread: URI120H : len:23 : ['lp25', 'cp32-12', 'lp21', 'cp32-4', 'cp32-1', 'cp26', 'cp32-7', 'cp32-6', 'lp28-2', 'cp32-3', 'lp54', 'lp28-1', 'cp32-10', 'cp32-11', 'lp28-3', 'lp28-4', 'lp17', 'cp32-9', 'cp32-5', 'chromosome', 'lp36', 'cp32-8', 'lp38']\n",
      "shortread: URI120 : len: 16['lp28-1', 'lp25', 'cp9', 'chromosome', 'cp9-3', 'lp28-2', 'lp21', 'lp36', 'lp38', 'cp32-4', 'lp28-4', 'lp28-3', 'lp17', 'lp54', 'cp26', 'lp56']\n",
      "longread: UNY193P : len:19 : ['lp25', 'cp32-12', 'lp28-7', 'lp28-6', 'cp9', 'cp26', 'lp5', 'cp32-7', 'cp32-6', 'lp28-2', 'lp54', 'cp32-13', 'cp32-10', 'lp28-3', 'lp17', 'lp28-4', 'chromosome', 'lp36', 'lp38']\n",
      "shortread: UNY193 : len: 27['lp25', 'cp32-12', 'lp28-7', 'lp21', 'cp32-4', 'lp28-6', 'cp9', 'cp26', 'lp56', 'cp32-7', 'cp32-6', 'lp28-2', 'cp32-3', 'lp28-8', 'lp54', 'cp32-13', 'lp28-1', 'cp9-3', 'lp28-3', 'lp17', 'lp28-4', 'cp32-9-4', 'cp32-9', 'cp32-5', 'chromosome', 'lp36', 'lp38']\n",
      "longread: URI93H : len:17 : ['cp32-7', 'lp28-1', 'lp25', 'cp32-5', 'chromosome', 'cp32-10', 'cp32-9', 'lp28-2', 'cp32-3', 'lp21', 'lp36', 'lp38', 'cp32-4', 'lp28-3', 'lp28-4', 'lp54', 'cp26']\n",
      "shortread: URI93 : len: 22['lp25', 'lp21', 'cp32-4', 'cp32-2', 'cp26', 'lp56', 'cp32-7', 'lp28-2', 'cp32-3', 'lp54', 'lp28-1', 'cp9-3', 'cp32-11', 'lp28-3', 'lp17', 'lp28-4', 'cp32-9-4', 'cp32-9', 'cp32-5', 'chromosome', 'lp36', 'lp38']\n",
      "val      chromosome  cp26  cp32-1  cp32-1+5  cp32-10  cp32-11  cp32-12  \\\n",
      "key                                                                      \n",
      "ESI26             1     1       1         0        1        0        1   \n",
      "ESI26H            1     1       1         0        1        0        1   \n",
      "UCT109            1     1       0         0        0        1        0   \n",
      "UCT109H           1     1       0         0        1        0        0   \n",
      "UCT110            1     1       0         0        0        1        0   \n",
      "...             ...   ...     ...       ...      ...      ...      ...   \n",
      "UWI248P           1     1       1         0        0        1        1   \n",
      "UWI263            1     1       1         1        0        1        1   \n",
      "UWI263P           1     1       0         0        1        1        0   \n",
      "UWI283            1     1       1         0        1        0        1   \n",
      "UWI283P           1     1       1         0        0        0        1   \n",
      "\n",
      "val      cp32-13  cp32-2  cp32-3  ...  lp28-6  lp28-7  lp28-8  lp28-9  lp36  \\\n",
      "key                               ...                                         \n",
      "ESI26          0       0       0  ...       0       1       0       1     1   \n",
      "ESI26H         0       0       0  ...       0       0       0       1     1   \n",
      "UCT109         0       1       1  ...       0       0       0       0     1   \n",
      "UCT109H        0       0       1  ...       0       0       0       0     1   \n",
      "UCT110         0       1       1  ...       1       0       0       0     1   \n",
      "...          ...     ...     ...  ...     ...     ...     ...     ...   ...   \n",
      "UWI248P        0       0       1  ...       0       0       0       0     1   \n",
      "UWI263         0       0       0  ...       0       0       0       0     1   \n",
      "UWI263P        0       0       1  ...       0       0       0       0     1   \n",
      "UWI283         0       0       1  ...       1       0       0       0     1   \n",
      "UWI283P        0       0       1  ...       0       0       0       0     1   \n",
      "\n",
      "val      lp38  lp5  lp54  lp56  Sum  \n",
      "key                                  \n",
      "ESI26       0    0     1     0   17  \n",
      "ESI26H      0    0     1     0   13  \n",
      "UCT109      1    0     1     1   21  \n",
      "UCT109H     1    0     1     0   17  \n",
      "UCT110      1    0     1     0   22  \n",
      "...       ...  ...   ...   ...  ...  \n",
      "UWI248P     1    0     1     0   18  \n",
      "UWI263      1    0     1     0   20  \n",
      "UWI263P     1    0     1     0   19  \n",
      "UWI283      0    0     1     1   20  \n",
      "UWI283P     0    0     1     0   16  \n",
      "\n",
      "[98 rows x 40 columns]\n"
     ]
    }
   ],
   "source": [
    "sr_plasmids, lr_plasmids = {}, {}\n",
    "sr_plasmids, lr_plasmids = process_assembly_dict(assembly_dict, output_dir, parsing_table, 2500)\n",
    "\n",
    "for k in lr_plasmids:\n",
    "    print(f'longread: {k} : len:{len(list(set(lr_plasmids[k])))} : {list(set(lr_plasmids[k]))}')\n",
    "    lr_plasmids[k] = list(set(lr_plasmids[k]))\n",
    "    print(f'shortread: {k[:-1]} : len: {len(list(set(sr_plasmids[k[:-1]])))}{list(set(sr_plasmids[k[:-1]]))}')\n",
    "    sr_plasmids[k[:-1]] = list(set(sr_plasmids[k[:-1]]))\n",
    "\n",
    "dicts_of_results = [sr_plasmids, lr_plasmids]\n",
    "plasmid_presence_matrix = []\n",
    "for dict1 in dicts_of_results:\n",
    "    for k in dict1:\n",
    "        #print(k)\n",
    "        for v in dict1[k]:\n",
    "            #print(v)\n",
    "            plasmid_presence_matrix.append((k,v))\n",
    "\n",
    "plasmid_df = pandas.DataFrame(plasmid_presence_matrix, columns=['key','val']).set_index(['key'])\n",
    "plasmid_count_matrix = plasmid_df.pivot_table(index='key', columns='val',aggfunc=len,fill_value=0,sort=True)\n",
    "plasmid_count_matrix['Sum'] = plasmid_count_matrix.sum(axis=1)\n",
    "print(plasmid_count_matrix)\n",
    "#plasmid_count_matrix.to_csv(f'{output_dir}/plasmid_presence_matrix_v3_2000cutoff.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>val</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>cp26</th>\n",
       "      <th>cp32-1</th>\n",
       "      <th>cp32-1+5</th>\n",
       "      <th>cp32-10</th>\n",
       "      <th>cp32-11</th>\n",
       "      <th>cp32-12</th>\n",
       "      <th>cp32-13</th>\n",
       "      <th>cp32-2</th>\n",
       "      <th>cp32-3</th>\n",
       "      <th>...</th>\n",
       "      <th>lp28-6</th>\n",
       "      <th>lp28-7</th>\n",
       "      <th>lp28-8</th>\n",
       "      <th>lp28-9</th>\n",
       "      <th>lp36</th>\n",
       "      <th>lp38</th>\n",
       "      <th>lp5</th>\n",
       "      <th>lp54</th>\n",
       "      <th>lp56</th>\n",
       "      <th>Sum</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>key</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>UCT110</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>URI34</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows  40 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "val     chromosome  cp26  cp32-1  cp32-1+5  cp32-10  cp32-11  cp32-12  \\\n",
       "key                                                                     \n",
       "UCT110           1     1       0         0        0        1        0   \n",
       "URI34            1     1       0         0        1        0        1   \n",
       "\n",
       "val     cp32-13  cp32-2  cp32-3  ...  lp28-6  lp28-7  lp28-8  lp28-9  lp36  \\\n",
       "key                              ...                                         \n",
       "UCT110        0       1       1  ...       1       0       0       0     1   \n",
       "URI34         1       0       0  ...       1       1       0       0     1   \n",
       "\n",
       "val     lp38  lp5  lp54  lp56  Sum  \n",
       "key                                 \n",
       "UCT110     1    0     1     0   22  \n",
       "URI34      1    1     1     0   22  \n",
       "\n",
       "[2 rows x 40 columns]"
      ]
     },
     "execution_count": 63,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "plasmid_count_matrix[plasmid_count_matrix.get('cp32-5+1') ==1]\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "def throwexcept():\n",
    "    raise Exception(\"this is an exception, stop hjere please\")\n",
    "throwexcept()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### With all due disrespect\n",
    "what in the name of all things good and holy was I thinking when I wrote the below code????"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "URI103--NODE_17_length_7333_cov_478.155733__FEATURE'S DIFFERENT!: dict_keys(['mol_type'])\n",
      "URI46--contig000014__FEATURE'S DIFFERENT!: dict_keys(['mol_type'])\n",
      "UCT30--contig000016__FEATURE'S DIFFERENT!: dict_keys(['mol_type'])\n",
      "UCT110--contig000010__FEATURE'S DIFFERENT!: dict_keys(['mol_type'])\n",
      "UCT110--contig000016__FEATURE'S DIFFERENT!: dict_keys(['mol_type'])\n",
      "UNY203--NODE_60_length_6810_cov_5.51115__FEATURE'S DIFFERENT!: dict_keys(['mol_type'])\n",
      "URI117--contig000019__FEATURE'S DIFFERENT!: dict_keys(['mol_type'])\n",
      "URI117--contig000022__FEATURE'S DIFFERENT!: dict_keys(['mol_type'])\n",
      "UCT35--contig000015__FEATURE'S DIFFERENT!: dict_keys(['mol_type'])\n",
      "UCT35--contig000015__FEATURE'S DIFFERENT!: dict_keys(['mol_type'])\n",
      "UCT35--contig000015__FEATURE'S DIFFERENT!: dict_keys(['mol_type'])\n",
      "URI33--contig000016__FEATURE'S DIFFERENT!: dict_keys(['mol_type'])\n",
      "URI86--contig000013__FEATURE'S DIFFERENT!: dict_keys(['mol_type'])\n",
      "URI118--contig000014__FEATURE'S DIFFERENT!: dict_keys(['mol_type'])\n",
      "UNY172--B356_cp32-5+1__FEATURE'S DIFFERENT!: dict_keys(['mol_type'])\n",
      "URI112--contig000014__FEATURE'S DIFFERENT!: dict_keys(['mol_type'])\n",
      "UCT113--contig000015__FEATURE'S DIFFERENT!: dict_keys(['mol_type'])\n",
      "UCT32--contig000019__FEATURE'S DIFFERENT!: dict_keys(['mol_type'])\n",
      "URI41--contig000014__FEATURE'S DIFFERENT!: dict_keys(['mol_type'])\n",
      "URI120--contig000024__FEATURE'S DIFFERENT!: dict_keys(['mol_type'])\n",
      "URI120--contig000024__FEATURE'S DIFFERENT!: dict_keys(['mol_type'])\n",
      "URI120--contig000024__FEATURE'S DIFFERENT!: dict_keys(['mol_type'])\n",
      "URI120--contig000024__FEATURE'S DIFFERENT!: dict_keys(['mol_type'])\n"
     ]
    }
   ],
   "source": [
    "#sr_plasmids = {}\n",
    "#lr_plasmids = {}\n",
    "\n",
    "sr_plasmids = defaultdict(dict)\n",
    "lr_plasmids = defaultdict(dict)\n",
    "\n",
    "for isolate in assembly_dict:\n",
    "    lriso_name = assembly_dict[isolate]['longread_name']\n",
    "    sr_plasmids[isolate] = []\n",
    "    lr_plasmids[lriso_name] = []\n",
    "    for method in assembly_dict[isolate]:\n",
    "        if method == 'longread_name':\n",
    "            break\n",
    "        else:\n",
    "            with open(f'{output_dir}/blast_results_v4/{isolate}_{method}_plasmid_hits_pf32.tsv', 'w') as f:\n",
    "                with open(f'{output_dir}/blast_results_v4/{isolate}_{method}_plasmid_hits_wp.tsv', 'w') as g:\n",
    "                    if method == 'shortread':\n",
    "                        f.write(f'Assembly ID: {isolate}\\n')\n",
    "                        g.write(f'Assembly ID: {isolate}\\n')\n",
    "                        f.write('PF32 Hits:\\n contig_id\\thit_id\\tcontig len\\tbitscore\\te_score\\tcontig_start\\tcontig_end\\tgene_at_coords\\tgene_start\\tgene_end\\n')\n",
    "                        g.write('Whole Plasmid Hits:\\ncontig_id\\thit_name\\thit_acc_id\\tcontig_len\\texpected_len\\tbitscore\\te_score\\tref_start\\tref_end\\tcontig_start\\tcontig_end\\n')\n",
    "                    else:\n",
    "                        f.write(f'Assembly ID: {lriso_name}\\n')\n",
    "                        g.write(f'Assembly ID: {lriso_name}\\n')\n",
    "                        f.write('PF32 Hits:\\n contig_id\\thit_id\\tcontig len\\tbitscore\\te_score\\tcontig_start\\tcontig_end\\tgene_at_coords\\tgene_start\\tgene_end\\n')\n",
    "                        g.write('Whole Plasmid Hits:\\ncontig_id\\thit_name\\thit_acc_id\\tcontig_len\\texpected_len\\tbitscore\\te_score\\tref_start\\tref_end\\tcontig_start\\tcontig_end\\n')\n",
    "                    for contig in assembly_dict[isolate][method]:\n",
    "                        contig_len = len(assembly_dict[isolate][method][contig]['seqrecord'].seq)\n",
    "                        if contig_len > 900000:\n",
    "                                #print('prob chromosome')\n",
    "                                f.write(f'{contig}\\tchromosome\\t{contig_len}\\t\"det_by_len\"\\t\"det_by_len\"\\t\"det_by_len\"\\t\"det_by_len\"\\t\"det_by_len\"\\t\"det_by_len\"\\t\"det_by_len\"\\n')\n",
    "                        else:\n",
    "                            #print(f'\\ncurrent: {isolate}-{method}-{contig}')\n",
    "                            if len(assembly_dict[isolate][method][contig]['seqrecord'].seq) < 2500:\n",
    "                                #print('too short!')\n",
    "                                f.write(f'{contig}\\tcontig_len < 2500bp\\n')\n",
    "                            else:\n",
    "                                if len(assembly_dict[isolate][method][contig].get(\"pf32_hits\")) > 0:\n",
    "                                    pfhit_title = assembly_dict[isolate][method][contig][\"pf32_hits\"]\n",
    "                                    for pfhit in pfhit_title:\n",
    "                                        #dict_keys(['contig_id', 'contig_len', 'hit_id', 'hit_title', 'hit_def', 'hit_expect', 'hit_score', 'hit_query_start', 'hit_query_end', 'hit_subject_start', 'hit_subject_end', 'hit_idents', 'hit_pos', 'hit_gaps', 'hit_frame', 'hit_query', 'hit_subject'])\n",
    "                                        hit_id = pfhit[\"hit_title\"].split(' ')[0].split(\"_\")[-1]\n",
    "                                        if \"|\" in hit_id:\n",
    "                                            hit_id = hit_id.split(\"|\")[-1]\n",
    "                                        score = pfhit[\"hit_score\"]\n",
    "                                        contig_len = pfhit[\"contig_len\"]\n",
    "                                        contig_start = pfhit['hit_query_start']\n",
    "                                        contig_end = pfhit['hit_query_end']\n",
    "                                        e_score = hit['hit_expect']\n",
    "                                        # Iterate over the features and find the one that matches the coordinates, this needs rewriting!!!!!\n",
    "                                        matching_feature = None\n",
    "                                        for feature in assembly_dict[isolate][method][contig]['seqrecord'].features:\n",
    "                                            #print(feature)\n",
    "                                            if feature.location.start - 10 <= contig_start <= feature.location.end + 10 and feature.location.start - 10 <= contig_end <= feature.location.end + 10:\n",
    "                                                # Match found\n",
    "                                                matching_feature = feature\n",
    "                                                #print(matching_feature)\n",
    "                                        # Print the annotation of the matching feature\n",
    "                                        if matching_feature: # this should go within the above block.\n",
    "                                            #['db_xref', 'product', 'locus_tag', 'protein_id', 'translation', 'codon_start', 'transl_table', 'inference', 'gene']\n",
    "                                            if 'locus_tag' in matching_feature.qualifiers:\n",
    "                                                feat_id = matching_feature.qualifiers['locus_tag'][0]\n",
    "                                                product = matching_feature.qualifiers['product'][0]\n",
    "                                                gene_start = matching_feature.location.start\n",
    "                                                gene_end = matching_feature.location.end\n",
    "                                                if 'gene' in matching_feature.qualifiers:\n",
    "                                                    feat_id = matching_feature.qualifiers['gene'][0]\n",
    "                                                    #print(f\"found match => {gene_name}\")\n",
    "                                                else:\n",
    "                                                    feat_id = product\n",
    "                                                    #print(f\"{isolate}{method} found match => {product}\")\n",
    "                                            else:\n",
    "                                                print(f\"{isolate}--{contig}__FEATURE'S DIFFERENT!: {matching_feature.qualifiers.keys()}\") # this is a bit of a hack to see what's in the feature\n",
    "                                                feat_id = 'NaN'\n",
    "                                                gene_start = 'NaN'\n",
    "                                                gene_end = 'NaN'\n",
    "                                        else:\n",
    "                                            pass\n",
    "                                            #print('No feature found at the specified coordinates.')\n",
    "                                    if \"|\" in pfhit_title[0]['hit_title'].split(' ')[0].split(\"_\")[-1]:\n",
    "                                        pf32_hit_id = pfhit_title[0]['hit_title'].split(' ')[0].split(\"_\")[-1].split(\"|\")[-1]\n",
    "                                    else:\n",
    "                                        pf32_hit_id = pfhit_title[0]['hit_title'].split(' ')[0].split(\"_\")[-1]\n",
    "                                    f.write(f'{contig}\\t{pf32_hit_id}\\t{contig_len}\\t{score}\\t{e_score}\\t{contig_start}\\t{contig_end}\\t{feat_id}\\t{gene_start}\\t{gene_end}\\n')\n",
    "                                    if method == 'shortread':\n",
    "                                        pftuple = (isolate, contig, \"pf32\", pf32_hit_id, contig_len,0)\n",
    "                                        sr_plasmids[isolate].append(pftuple)\n",
    "                                    else:\n",
    "                                        pftuple = (lriso_name, contig, \"pf32\", pf32_hit_id, contig_len,0)\n",
    "                                        lr_plasmids[lriso_name].append(pftuple)\n",
    "                                else:\n",
    "                                    f.write(f'{contig}\\tNo PF32 hits\\n')\n",
    "                        if len(assembly_dict[isolate][method][contig].get(\"wp_hits\")) > 0:\n",
    "                            hit_title = assembly_dict[isolate][method][contig].get(\"wp_hits\")\n",
    "                            for hit in hit_title:\n",
    "                                #dict_keys(['contig_id', 'contig_len', 'hit_id', 'hit_title', 'hit_def', 'hit_expect', 'hit_score', 'hit_query_start', 'hit_query_end', 'hit_subject_start', 'hit_subject_end', 'hit_idents', 'hit_pos', 'hit_gaps', 'hit_frame', 'hit_query', 'hit_subject'])\n",
    "                                hit_id = hit[\"hit_title\"].split('|')[1]\n",
    "                                wp_hit = parsing_table[hit_id]['name']\n",
    "                                score = hit[\"hit_score\"]\n",
    "                                contig_len = hit[\"contig_len\"]\n",
    "                                replicon_len = parsing_table[hit_id]['length']\n",
    "                                ref_start = hit['hit_subject_start']\n",
    "                                ref_end = hit['hit_subject_end']\n",
    "                                e_score = hit['hit_expect']\n",
    "                                contig_start = hit['hit_query_start']\n",
    "                                contig_end = hit['hit_query_end']\n",
    "                                #if len(assembly_dict[isolate][method][contig].get(\"pf32_hits\")) == 0 or contig_len > 100000:\n",
    "                                if len(assembly_dict[isolate][method][contig]['seqrecord'].seq) < 2500:\n",
    "                                    g.write(f'{contig}\\t contig_len < 2500bp\\n')\n",
    "                                else:           #contig_id\\thit_name\\thit_acc_id\\tcontig_len\\texpected_len\\tbitscore\\te_score\\tref_start\\tref_end\\tcontig_start\\tcontig_end\\n\n",
    "                                    g.write(f'{contig}\\t{wp_hit}\\t{hit_id}\\t{contig_len}\\t{replicon_len}\\t{score}\\t{e_score}\\t{ref_start}\\t{ref_end}\\t{contig_start}\\t{contig_end}\\n')\n",
    "                                    if method == 'shortread':\n",
    "                                        name_tuple = (isolate, contig, \"wp\", parsing_table[hit_title[0]['hit_title'].split('|')[1]]['name'], contig_len, replicon_len)\n",
    "                                        #print(name_tuple)\n",
    "                                        sr_plasmids[isolate].append(name_tuple)\n",
    "                                    else:\n",
    "                                        name_tuple = (lriso_name, contig, \"wp\", parsing_table[hit_title[0]['hit_title'].split('|')[1]]['name'], contig_len, replicon_len)\n",
    "                                        #print(name_tuple)\n",
    "                                        lr_plasmids[lriso_name].append(name_tuple)\n",
    "                        else:\n",
    "                            pass\n",
    "                            #print(f'{contig}\\tNo whole plasmid hits')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 148,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Function to calculate overall percent identity and coverage using HSPs\n",
    "def calculate_percent_identity_and_coverage(alignment):\n",
    "    total_identities = 0\n",
    "    total_alignment_length = 0\n",
    "    covered_positions = set()\n",
    "\n",
    "    for hsp in alignment.hsps:\n",
    "        total_identities += hsp.identities\n",
    "        total_alignment_length += hsp.align_length\n",
    "\n",
    "        # Track covered positions in the reference\n",
    "        covered_positions.update(range(hsp.sbjct_start, hsp.sbjct_end))\n",
    "\n",
    "    # Calculate overall percent identity\n",
    "    overall_percent_identity = (total_identities / total_alignment_length) * 100 if total_alignment_length != 0 else 0\n",
    "\n",
    "    # Calculate coverage percentage\n",
    "    reference_length = alignment.length\n",
    "    coverage_percentage = (len(covered_positions) / reference_length) * 100 if reference_length != 0 else 0\n",
    "\n",
    "    return overall_percent_identity, coverage_percentage, covered_positions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 204,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Function to calculate overall percent identity and coverage using HSPs\n",
    "def calculate_percent_identity_and_coverage(alignment):\n",
    "    total_identities = 0\n",
    "    total_alignment_length = 0\n",
    "    covered_intervals = []\n",
    "\n",
    "    for hsp in alignment.hsps:\n",
    "        total_identities += hsp.identities\n",
    "        total_alignment_length += hsp.align_length\n",
    "\n",
    "        # Add HSP interval to the covered intervals list, make sure the coordinates are correct!\n",
    "        start = min(hsp.sbjct_start, hsp.sbjct_end)\n",
    "        end = max(hsp.sbjct_start, hsp.sbjct_end)\n",
    "\n",
    "        # Add HSP interval to the covered intervals list\n",
    "        covered_intervals.append((start, end))\n",
    "\n",
    "    # Merge overlapping intervals to calculate the covered length\n",
    "    merged_intervals = merge_intervals(covered_intervals)\n",
    "    covered_length = sum(end - start for start, end in merged_intervals)\n",
    "\n",
    "    # Calculate overall percent identity\n",
    "    overall_percent_identity = (total_identities / total_alignment_length) * 100 if total_alignment_length != 0 else 0\n",
    "\n",
    "    # Calculate coverage percentage\n",
    "    reference_length = alignment.length\n",
    "    coverage_percentage = (covered_length / reference_length) * 100 if reference_length != 0 else 0\n",
    "\n",
    "    return overall_percent_identity, coverage_percentage, covered_length\n",
    "\n",
    "def merge_intervals(intervals):\n",
    "    # Sort intervals by start position\n",
    "    sorted_intervals = sorted(intervals, key=lambda x: x[0])\n",
    "    merged_intervals = []\n",
    "\n",
    "    for interval in sorted_intervals:\n",
    "        if not merged_intervals or merged_intervals[-1][1] < interval[0]:\n",
    "            merged_intervals.append(interval)\n",
    "        else:\n",
    "            merged_intervals[-1] = (merged_intervals[-1][0], max(merged_intervals[-1][1], interval[1]))\n",
    "\n",
    "    return merged_intervals\n",
    "\n",
    "def add_to_nested_dict(nested_dict, outer_key, inner_key, value):\n",
    "    if outer_key not in nested_dict:\n",
    "        nested_dict[outer_key] = {}  # Initialize a dictionary for this outer key\n",
    "    if inner_key not in nested_dict[outer_key]:\n",
    "        nested_dict[outer_key][inner_key] = value  # instantiate inner_key = value\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 224,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI36_whole_plasmid.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI91H_whole_plasmid.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI88H_whole_plasmid.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI46_whole_plasmid.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UCT31H_whole_plasmid.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UCT32_whole_plasmid.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI89_whole_plasmid.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UCT109H_whole_plasmid.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UCT109_whole_plasmid.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UNY193_whole_plasmid.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI44_whole_plasmid.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UWI283_whole_plasmid.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UWI247_whole_plasmid.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UNY203P_whole_plasmid.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UNY193P_whole_plasmid.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI36H_whole_plasmid.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UCT110_whole_plasmid.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI42H_whole_plasmid.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UCT110H_whole_plasmid.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI87H_whole_plasmid.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI39_whole_plasmid.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI117H_whole_plasmid.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UCT92H_whole_plasmid.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI102_whole_plasmid.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UWI263_whole_plasmid.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UNY203_whole_plasmid.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UNY208P_whole_plasmid.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UCT30H_whole_plasmid.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UCT31_whole_plasmid.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/ESI26H_whole_plasmid.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI88_whole_plasmid.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UNY172P_whole_plasmid.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI91_whole_plasmid.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UWI283P_whole_plasmid.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI120H_whole_plasmid.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/ESI26_whole_plasmid.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UCT113_whole_plasmid.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UCT96_whole_plasmid.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UWI263P_whole_plasmid.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI107_whole_plasmid.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UNY149_whole_plasmid.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI107H_whole_plasmid.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI42_whole_plasmid.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI101H_whole_plasmid.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI47H_whole_plasmid.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UCT29H_whole_plasmid.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI86H_whole_plasmid.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UNY149P_whole_plasmid.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI34H_whole_plasmid.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UCT32H_whole_plasmid.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI117_whole_plasmid.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI89H_whole_plasmid.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UCT30_whole_plasmid.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI103H_whole_plasmid.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UCT35H_whole_plasmid.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI111_whole_plasmid.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UWI248_whole_plasmid.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI46H_whole_plasmid.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UWI248P_whole_plasmid.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI120_whole_plasmid.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI48H_whole_plasmid.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI112_whole_plasmid.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI93H_whole_plasmid.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI34_whole_plasmid.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI101_whole_plasmid.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI86_whole_plasmid.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UNY169_whole_plasmid.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI40H_whole_plasmid.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI56H_whole_plasmid.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UNY208_whole_plasmid.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UCT50H_whole_plasmid.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI118_whole_plasmid.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI39H_whole_plasmid.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI56_whole_plasmid.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UCT92_whole_plasmid.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI111H_whole_plasmid.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UNY172_whole_plasmid.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UWI247P_whole_plasmid.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI112H_whole_plasmid.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI87_whole_plasmid.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI40_whole_plasmid.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI102H_whole_plasmid.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UCT29_whole_plasmid.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI41_whole_plasmid.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UCT96H_whole_plasmid.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UCT113H_whole_plasmid.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI103_whole_plasmid.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI33H_whole_plasmid.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI118H_whole_plasmid.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UNY169P_whole_plasmid.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UCT35_whole_plasmid.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI33_whole_plasmid.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI48_whole_plasmid.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI93_whole_plasmid.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/UCT50_whole_plasmid.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI44H_whole_plasmid.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI41H_whole_plasmid.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/whole_plasmid/URI47_whole_plasmid.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UCT35H_pf32.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/ESI26_pf32.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/ESI26H_pf32.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UCT32H_pf32.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UNY169P_pf32.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UWI248P_pf32.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI34H_pf32.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI111_pf32.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UCT29H_pf32.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI33_pf32.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UWI247_pf32.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UCT110_pf32.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UCT31_pf32.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI39_pf32.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UWI283_pf32.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI120_pf32.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UCT35_pf32.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI41_pf32.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UNY193P_pf32.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UNY208P_pf32.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI107_pf32.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI103_pf32.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UCT29_pf32.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI89_pf32.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI39H_pf32.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI42H_pf32.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UNY172P_pf32.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI41H_pf32.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UCT109H_pf32.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI33H_pf32.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI40H_pf32.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI47_pf32.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI44H_pf32.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI46_pf32.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI86_pf32.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI102H_pf32.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UNY208_pf32.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI88H_pf32.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UCT92_pf32.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UCT50H_pf32.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UCT92H_pf32.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI48H_pf32.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI111H_pf32.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI44_pf32.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UCT109_pf32.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI89H_pf32.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI46H_pf32.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UNY169_pf32.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI88_pf32.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI102_pf32.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UCT113H_pf32.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI40_pf32.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UWI263_pf32.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI56_pf32.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI93H_pf32.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI86H_pf32.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI56H_pf32.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI48_pf32.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UNY149P_pf32.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UWI283P_pf32.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI47H_pf32.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UCT31H_pf32.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI91_pf32.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI107H_pf32.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UNY172_pf32.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UNY203_pf32.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UCT32_pf32.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI112H_pf32.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UWI247P_pf32.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI103H_pf32.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UCT113_pf32.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI118_pf32.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI87H_pf32.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UCT50_pf32.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UCT96H_pf32.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UNY203P_pf32.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI118H_pf32.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UNY193_pf32.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UNY149_pf32.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UCT30H_pf32.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI36H_pf32.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI36_pf32.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI42_pf32.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI34_pf32.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI101_pf32.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UCT110H_pf32.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UCT30_pf32.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UWI248_pf32.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI120H_pf32.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI87_pf32.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI101H_pf32.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UCT96_pf32.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI112_pf32.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI117H_pf32.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI117_pf32.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI91H_pf32.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/URI93_pf32.xml\n",
      "/home/mf019/borrelia_plasmid_classifier_v3/output/blast_results_v4/pf32/UWI263P_pf32.xml\n"
     ]
    }
   ],
   "source": [
    "# dir(blast_record)\n",
    "#['__class__', '__delattr__', '__dict__', '__dir__', '__doc__', '__eq__', '__format__', '__ge__', '__getattribute__', '__getstate__', '__gt__', '__hash__', '__init__', '__init_subclass__', \n",
    "#'__le__', '__lt__', '__module__', '__ne__', '__new__', '__reduce__', '__reduce_ex__', '__repr__', '__setattr__', '__sizeof__', '__str__', '__subclasshook__', '__weakref__', 'alignments', \n",
    "#'application', 'blast_cutoff', 'database', 'database_length', 'database_letters', 'database_name', 'database_sequences', 'date', 'descriptions', 'dropoff_1st_pass', 'effective_database_length', \n",
    "#'effective_hsp_length', 'effective_query_length', 'effective_search_space', 'effective_search_space_used', 'expect', 'filter', 'frameshift', 'gap_penalties', 'gap_trigger', 'gap_x_dropoff', 'gap_x_dropoff_final',\n",
    "#'gapped', 'hsps_gapped', 'hsps_no_gap', 'hsps_prelim_gapped', 'hsps_prelim_gapped_attemped', 'ka_params', 'ka_params_gap', 'matrix', 'multiple_alignment', 'num_good_extends', 'num_hits', 'num_letters_in_database', \n",
    "#'num_seqs_better_e', 'num_sequences', 'num_sequences_in_database', 'posted_date', 'query', 'query_id', 'query_length', 'query_letters', 'reference', 'sc_match', 'sc_mismatch', 'threshold', 'version', 'window_size']\n",
    "####\n",
    "# dir(alignment)\n",
    "#['__class__', '__delattr__', '__dict__', '__dir__', '__doc__', '__eq__', '__format__', '__ge__', '__getattribute__', '__getstate__', '__gt__', '__hash__', '__init__', '__init_subclass__', '__le__', '__lt__', \n",
    "#'__module__', '__ne__', '__new__', '__reduce__', '__reduce_ex__', '__repr__', '__setattr__', '__sizeof__', '__str__', '__subclasshook__', '__weakref__', 'accession', 'hit_def', 'hit_id', 'hsps', 'length', 'title']\n",
    "test_dict = defaultdict()\n",
    "for result in wp_results:\n",
    "    print(result)\n",
    "    name = Path(result).stem.split(\"_\")[0]\n",
    "    method = 'longread' if name.endswith(('H','P')) else 'shortread'\n",
    "    name = name[:-1] if name.endswith(('H','P')) else name\n",
    "        \n",
    "    add_to_nested_dict(test_dict, name, method, defaultdict(dict))\n",
    "    \n",
    "    with open(result, \"r\") as blast_file:\n",
    "       # Parse BLAST XML output\n",
    "        blast_records = NCBIXML.parse(blast_file)\n",
    "        for blast_record in blast_records:\n",
    "            contig = blast_record.query\n",
    "            query_length = blast_record.query_length\n",
    "            test_dict[name][method][contig] = {'wp' : [], 'pf32' : []}\n",
    "            if query_length >= 2500:\n",
    "                for alignment in blast_record.alignments:\n",
    "                    alignment_id = alignment.hit_id.split('|')[1]\n",
    "                    ref_length = alignment.length\n",
    "                    plasmid_name = parsing_table.get(alignment_id, {}).get('name', 'uh-oh!')\n",
    "                    overall_percent_identity, coverage_percentage, covered_positions = calculate_percent_identity_and_coverage(alignment)\n",
    "                    test_dict[name][method][contig]['wp'].append({\n",
    "                        \"alignment_id\": alignment_id,\n",
    "                        \"plasmid_name\": plasmid_name,\n",
    "                        \"query_length\": query_length,\n",
    "                        \"ref_length\": ref_length,\n",
    "                        \"overall_percent_identity\": overall_percent_identity,\n",
    "                        \"coverage_percentage\": coverage_percentage,\n",
    "                        'covered_positions': covered_positions\n",
    "                    })\n",
    "\n",
    "for result in pf32_results:\n",
    "    print(result)\n",
    "    name = Path(result).stem.split(\"_\")[0]\n",
    "    method = 'longread' if name.endswith(('H','P')) else 'shortread'\n",
    "    name = name[:-1] if name.endswith(('H','P')) else name\n",
    "    with open(result, \"r\") as blast_file:\n",
    "       # Parse BLAST XML output\n",
    "        blast_records = NCBIXML.parse(blast_file)\n",
    "        for blast_record in blast_records:\n",
    "            contig = blast_record.query\n",
    "            query_length = blast_record.query_length\n",
    "            if query_length >= 2500:\n",
    "                for alignment in blast_record.alignments:\n",
    "                    alignment_id = alignment.hit_id\n",
    "                    plasmid_name = alignment.hit_id.split('_')[-1]\n",
    "                    ref_length = alignment.length\n",
    "                    overall_percent_identity, coverage_percentage, covered_positions = calculate_percent_identity_and_coverage(alignment)\n",
    "                    test_dict[name][method][contig]['pf32'].append({\n",
    "                        \"alignment_id\": alignment_id,\n",
    "                        \"plasmid_name\": plasmid_name,\n",
    "                        \"query_length\": query_length,\n",
    "                        \"ref_length\": ref_length,\n",
    "                        \"overall_percent_identity\": overall_percent_identity,\n",
    "                        \"coverage_percentage\": coverage_percentage,\n",
    "                        'covered_positions': covered_positions\n",
    "                    })"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 236,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "import csv\n",
    "# Function to determine the best match\n",
    "def get_best_match(matches, key):\n",
    "    best_match = None\n",
    "    best_score = -1\n",
    "    for match in matches:\n",
    "        if match[key] > best_score:\n",
    "            best_score = match[key]\n",
    "            best_match = match\n",
    "    return best_match\n",
    "\n",
    "with open('best_matches.tsv', 'w', newline='') as tsvfile:\n",
    "    fieldnames = [\n",
    "        'name', 'method', 'contig', \n",
    "        'wp_alignment_id', 'pf_alignment_id',\n",
    "        'wp_query_length', 'ref_length',\n",
    "        'wp_percent_identity', 'wp_coverage_percentage', \n",
    "        'wp_plasmid_name', 'pf_plasmid_name',\n",
    "        'best_hit', 'completeness'\n",
    "    ]\n",
    "    writer = csv.DictWriter(tsvfile, fieldnames=fieldnames, delimiter='\\t')\n",
    "    writer.writeheader()\n",
    "\n",
    "    for name, methods in test_dict.items():\n",
    "        for method, contigs in methods.items():\n",
    "            for contig, homologies in contigs.items():\n",
    "                wp_best_identity = get_best_match(homologies['wp'], 'overall_percent_identity')\n",
    "                wp_best_coverage = get_best_match(homologies['wp'], 'coverage_percentage')\n",
    "\n",
    "                pf32_best_identity = None\n",
    "                if homologies['pf32'] and (not homologies['wp'] or homologies['wp'][0]['query_length'] <= 100000):\n",
    "                    pf32_best_identity = get_best_match(homologies['pf32'], 'overall_percent_identity')\n",
    "                \n",
    "                best_hit = None\n",
    "                wp_best = None\n",
    "                pf_best = None\n",
    "                wp_alignment_id = ''\n",
    "                pf_alignment_id = ''\n",
    "                wp_query_length = ''\n",
    "                ref_length = ''\n",
    "                wp_percent_identity = ''\n",
    "                wp_coverage_percentage = ''\n",
    "                wp_plasmid_name = ''\n",
    "                pf_plasmid_name = ''\n",
    "\n",
    "                if wp_best_identity and wp_best_coverage and wp_best_coverage['coverage_percentage'] > wp_best_identity['coverage_percentage']:\n",
    "                    wp_best = wp_best_coverage\n",
    "                else:\n",
    "                    wp_best = wp_best_identity\n",
    "\n",
    "                if wp_best:\n",
    "                    wp_alignment_id = wp_best['alignment_id']\n",
    "                    wp_query_length = wp_best['query_length']\n",
    "                    ref_length = wp_best['ref_length']\n",
    "                    wp_percent_identity = wp_best['overall_percent_identity']\n",
    "                    wp_coverage_percentage = wp_best['coverage_percentage']\n",
    "                    wp_plasmid_name = wp_best['plasmid_name']\n",
    "\n",
    "                if pf32_best_identity:\n",
    "                    pf_alignment_id = pf32_best_identity['alignment_id']\n",
    "                    pf_plasmid_name = pf32_best_identity['plasmid_name']\n",
    "\n",
    "                if wp_best and pf32_best_identity:\n",
    "                    if pf32_best_identity['overall_percent_identity'] > wp_best['overall_percent_identity']:\n",
    "                        best_hit = 'pf32'\n",
    "                    else:\n",
    "                        best_hit = 'wp'\n",
    "                else:\n",
    "                    best_hit = 'wp' if wp_best else 'pf32'\n",
    "\n",
    "                completeness = 'presumed' if wp_best and wp_best['coverage_percentage'] >= 97 else 'incomplete'\n",
    "\n",
    "                writer.writerow({\n",
    "                    'name': name,\n",
    "                    'method': method,\n",
    "                    'contig': contig,\n",
    "                    'wp_alignment_id': wp_alignment_id,\n",
    "                    'pf_alignment_id': pf_alignment_id,\n",
    "                    'wp_query_length': wp_query_length,\n",
    "                    'ref_length': ref_length,\n",
    "                    'wp_percent_identity': wp_percent_identity,\n",
    "                    'wp_coverage_percentage': wp_coverage_percentage,\n",
    "                    'wp_plasmid_name': wp_plasmid_name,\n",
    "                    'pf_plasmid_name': pf_plasmid_name,\n",
    "                    'best_hit': best_hit,\n",
    "                    'completeness': completeness\n",
    "                })"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 237,
   "metadata": {},
   "outputs": [],
   "source": [
    "#### END OF NEW CODE!! MOVING TO NEW NOTEBOOK!!! ####"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "sr_contig_plasmid_df = pandas.DataFrame(columns=['isolate', 'contig','wp_hit', 'pf32_hit'])\n",
    "for key in sr_plasmids:\n",
    "    for item in set(sr_plasmids[key]):\n",
    "        if item[2] == 'wp':\n",
    "            new_row = {'isolate': item[0], 'contig': item[1], 'wp_hit': item[3], 'contig_len' : item[4]}#, 'replicon_len' : item[5]}\n",
    "        else:\n",
    "            new_row = {'isolate': item[0], 'contig': item[1], 'pf32_hit': item[3], 'contig_len' : item[4]}\n",
    "        new_row = pandas.DataFrame(new_row, index=[0])\n",
    "        sr_contig_plasmid_df = pandas.concat([sr_contig_plasmid_df, new_row], ignore_index=True)\n",
    "sr_df = sr_contig_plasmid_df.groupby(['isolate', 'contig'], as_index = False).agg({\n",
    "    'wp_hit': lambda x: [item for item in x if str(item) != 'nan'][0],\n",
    "    'pf32_hit': lambda x: \"None\" if len(x) == 1 else str([item for item in x if str(item) != 'nan']),\n",
    "    })\n",
    "sr_df.sort_index(inplace=True)\n",
    "sr_df['pf32_hit'] = sr_df['pf32_hit'].str.replace('[', '').str.replace(']', '').str.replace(\"'\", '').replace(\"nan\", 'NaN')\n",
    "sr_df.to_csv(f'{output_dir}/blast_results_v4/shortread_plasmid_hits_scuffed.tsv', sep='\\t', index=False)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "sr_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#sr_contig_plasmid_df.groupby(['isolate','contig'], as_index=False).aggregate({\n",
    "#    'wp_hit': lambda x: [val for val in x if str(val) != 'nan'],\n",
    "#    'pf32_hit': lambda x: [val for val in x if str(val) != 'nan']})\n",
    "#sr_contig_plasmid_df\n",
    "##sr_contig_plasmid_df.to_csv(f'{output_dir}/contig_plasmid_hits.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "lr_contig_plasmid_df = pandas.DataFrame(columns=['isolate', 'contig', 'wp_hit', 'pf32_hit'])\n",
    "for key in lr_plasmids:\n",
    "    for item in set(lr_plasmids[key]):\n",
    "        if item[2] == 'wp':\n",
    "            new_row = {'isolate': item[0], 'contig': item[1], 'wp_hit': item[3]}\n",
    "        else:\n",
    "            new_row = {'isolate': item[0], 'contig': item[1], 'pf32_hit': item[3]}\n",
    "        new_row = pandas.DataFrame(new_row, index=[0])\n",
    "        lr_contig_plasmid_df = pandas.concat([lr_contig_plasmid_df, new_row], ignore_index=True)\n",
    "lr_df = lr_contig_plasmid_df.groupby(['isolate', 'contig'], as_index = False).agg({\n",
    "    'wp_hit': lambda x: [item for item in x if str(item) != 'nan'][0],\n",
    "    'pf32_hit': lambda x: \"None\" if len(x) == 1 else str([item for item in x if str(item) != 'nan'])})\n",
    "lr_df.sort_index(inplace=True)\n",
    "lr_df['pf32_hit'] = lr_df['pf32_hit'].str.replace('[', '').str.replace(']', '').str.replace(\"'\", '').str.replace(\"nan\", 'None')\n",
    "lr_df.to_csv(f'{output_dir}/blast_results_v4/longread_plasmid_hits_scuffed.tsv', sep='\\t', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "lr_df['pf32_hit'].fillna('None')\n",
    "lr_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "sr_df[((sr_df['pf32_hit'] != '') & (sr_df['pf32_hit'] != sr_df['wp_hit']))].to_csv(f'{output_dir}/shortread_discrepancies.tsv', sep='\\t', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "lr_df[(lr_df['pf32_hit'] != lr_df['wp_hit']) & (lr_df['pf32_hit'] != '')].to_csv(f'{output_dir}/longread_discrepancies.tsv', sep='\\t', index=False)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
